<!DOCTYPE html>
<!-- Created by pdf2htmlEX (https://github.com/coolwanglu/pdf2htmlex) -->
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8"/>

<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>
<link rel="stylesheet" href="base.min.css"/>
<link rel="stylesheet" href="fancy.min.css"/>
<link rel="stylesheet" href="main.css"/>
<script src="compatibility.min.js"></script>
<script src="theViewer.min.js"></script>
<script>
try{
theViewer.defaultViewer = new theViewer.Viewer({});
}catch(e){}
</script>
<title></title>
</head>
<body>
<div id="sidebar">
<div id="outline">
</div>
</div>
<div id="page-container">
<div id="pf1" class="pf w0 h0" data-page-no="1"><div class="pc pc1 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1.png"/><div class="t m0 x1 h1 y1 ff1 fs0 fc0 sc0 ls0 ws0">Melanoma Skin Cancer Diagnosis based </div><div class="t m0 x2 h1 y2 ff1 fs0 fc0 sc0 ls0 ws0">on </div><div class="t m0 x3 h1 y3 ff1 fs0 fc0 sc0 ls0 ws0">Dermoscopic Features </div><div class="t m0 x4 h1 y4 ff1 fs0 fc0 sc0 ls0 ws0">and </div><div class="t m0 x5 h1 y5 ff1 fs0 fc0 sc0 ls0 ws0">DNA Mutations</div><div class="t m0 x6 h2 y6 ff2 fs1 fc0 sc0 ls0 ws0">Group No. 45 - <span class="_ _0"></span><span class="ff1">Team United</span></div><div class="t m0 x7 h3 y7 ff2 fs1 fc0 sc0 ls0 ws0">Supervisor: Dr. (Ms.) G. Upeksha Ganegoda</div><div class="t m0 x8 h4 y8 ff2 fs2 fc0 sc0 ls0 ws0">174058F<span class="_ _1"> </span>Jayatilake S. M. D. A. C.</div><div class="t m0 x8 h4 y9 ff2 fs2 fc0 sc0 ls0 ws0">174060E<span class="_ _1"> </span>Jayaweera J. A. A. M.</div><div class="t m0 x9 h4 ya ff2 fs2 fc0 sc0 ls0 ws0">174113V<span class="_ _1"> </span>Peiris P. D. M.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2" class="pf w0 h0" data-page-no="2"><div class="pc pc2 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2.png"/><div class="t m0 xa h5 yb ff1 fs3 fc1 sc0 ls0 ws0">Outline</div><div class="t m0 xb h6 yc ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Introduction</span></div><div class="t m0 xb h6 yd ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Motivation to Propose a Substitute for the  Existing Clinical Approaches</span></div><div class="t m0 xb h6 ye ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Comparison of Other’s Work </span></div><div class="t m0 xb h6 yf ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Aim &amp; Objectives</span></div><div class="t m0 xb h6 y10 ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Project Approach</span></div><div class="t m0 xb h6 y11 ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Solution in Detail</span></div><div class="t m0 xb h6 y12 ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Implementation</span></div><div class="t m0 xb h6 y13 ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Evaluation</span></div><div class="t m0 xb h6 y14 ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Conclusion</span></div><div class="t m0 xb h6 y15 ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Further Work</span></div><div class="t m0 xb h6 y16 ff2 fs4 fc2 sc0 ls0 ws0">•<span class="_ _2"> </span><span class="fs5">Demonstration</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3" class="pf w0 h0" data-page-no="3"><div class="pc pc3 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3.png"/><div class="t m0 xc h5 y17 ff1 fs3 fc0 sc0 ls0 ws0">Introduction</div><div class="t m0 xd h7 y18 ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Skin cancer is most common among Caucasians and is rapidly increasing year by year. </span></div><div class="t m0 xd h7 y19 ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">There are three main skin cancer types namely,</span></div><div class="t m0 xe h7 y1a ff3 fs6 fc0 sc0 ls0 ws0">■<span class="_ _3"> </span><span class="ff2">Basal Cell Carcinoma </span></div><div class="t m0 xe h7 y1b ff3 fs6 fc0 sc0 ls0 ws0">■<span class="_ _3"> </span><span class="ff2">Squamous Cell Carcinoma </span></div><div class="t m0 xe h7 y1c ff3 fs6 fc0 sc0 ls0 ws0">■<span class="_ _3"> </span><span class="ff2">Melanoma (Superficial Melanoma, Lentigo Maligna, Acral Lentiginous Melanoma, Nodular Melanoma)</span></div><div class="t m0 xd h7 y1d ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Melanoma is the most dangerous skin cancer type and is the world’s 19</span></div><div class="t m0 xf h8 y1e ff2 fs7 fc0 sc0 ls0 ws0">th</div><div class="t m0 x10 h7 y1d ff2 fs6 fc0 sc0 ls0 ws0"> most common cancer.</div><div class="t m0 xd h7 y1f ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Based <span class="_ _4"> </span>on <span class="_ _5"> </span>how <span class="_ _4"> </span>deep <span class="_ _4"> </span>the <span class="_ _5"> </span>Melanoma <span class="_ _4"> </span>tumour <span class="_ _5"> </span>spreads <span class="_ _6"> </span>and <span class="_ _5"> </span>how <span class="_ _4"> </span>high <span class="_ _5"> </span>the <span class="_ _5"> </span>lump <span class="_ _5"> </span>on <span class="_ _5"> </span>the <span class="_ _5"> </span>surface <span class="_ _4"> </span>is, <span class="_ _5"> </span>Melanoma <span class="_ _4"> </span>has <span class="_ _4"> </span>several <span class="_ _6"> </span>stages </span></div><div class="t m0 x11 h7 y20 ff2 fs6 fc0 sc0 ls0 ws0">according to its severity. </div><div class="t m0 xd h7 y21 ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Melanoma <span class="_ _4"> </span>can <span class="_ _4"> </span>occur <span class="_ _5"> </span>due <span class="_ _5"> </span>to <span class="_ _5"> </span>many <span class="_ _4"> </span>reasons. <span class="_ _6"> </span>One <span class="_ _5"> </span>is <span class="_ _5"> </span>when <span class="_ _5"> </span>DNA <span class="_ _4"> </span>is <span class="_ _5"> </span>damaged <span class="_ _4"> </span>from <span class="_ _7"> </span>UV <span class="_ _5"> </span>radiation <span class="_ _7"> </span>burns <span class="_ _5"> </span>on <span class="_ _5"> </span>skin <span class="_ _5"> </span>(causes <span class="_ _4"> </span>somatic </span></div><div class="t m0 x11 h7 y22 ff2 fs6 fc0 sc0 ls0 ws0">mutations)</div><div class="t m0 xd h7 y23 ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Most skin cancers can be cured when diagnosed early and therefore it is crucial to diagnose at the initial stage.</span></div><div class="t m0 xd h7 y24 ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">The <span class="_ _8"></span>cost <span class="_ _8"></span>involved <span class="_ _9"></span>in <span class="_ _8"></span>screening <span class="_ _a"></span>cancers <span class="_ _8"></span>by <span class="_ _b"></span>a <span class="_ _8"></span>dermatologist <span class="_ _a"></span>is <span class="_ _8"></span>high <span class="_ _c"></span>and <span class="_ _8"></span>only <span class="_ _8"></span>a <span class="_ _8"></span>dermatologist <span class="_ _b"></span>with <span class="_ _8"></span>a <span class="_ _c"></span>comprehensive knowledge and </span></div><div class="t m0 x11 h7 y25 ff2 fs6 fc0 sc0 ls0 ws0">years of practice can detect features of Melanoma from dermoscopic images at a considerable accuracy.</div><div class="t m0 xd h7 y26 ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Genetic <span class="_ _a"></span>or <span class="_ _b"></span>inherited <span class="_ _b"></span>conditions <span class="_ _8"></span>that <span class="_ _b"></span>causes <span class="_ _b"></span>Melanoma <span class="_ _a"></span>cannot <span class="_ _8"></span>be <span class="_ _b"></span>captured by <span class="_ _a"></span>a <span class="_ _8"></span>dermoscopy and <span class="_ _8"></span>sometimes <span class="_ _a"></span>it <span class="_ _8"></span>takes time <span class="_ _b"></span>to <span class="_ _b"></span>show </span></div><div class="t m0 x11 h7 y27 ff2 fs6 fc0 sc0 ls0 ws0">features of Melanoma from outside on the skin.</div><div class="t m0 xd h7 y28 ff3 fs6 fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Therefore two approaches are being considered in this research to detect Melanoma.</span></div><div class="t m0 xe h7 y29 ff3 fs6 fc0 sc0 ls0 ws0">■<span class="_ _3"> </span><span class="ff2">Extracting the features from a dermoscopic image of a lesion and analysing them</span></div><div class="t m0 xe h7 y2a ff3 fs6 fc0 sc0 ls0 ws0">■<span class="_ _3"> </span><span class="ff2">Analysing <span class="_ _a"></span>the <span class="_ _b"></span>DNA <span class="_ _a"></span>sequences <span class="_ _a"></span>up <span class="_ _b"></span>to <span class="_ _b"></span>their <span class="_ _b"></span>protein primary <span class="_ _a"></span>structure to <span class="_ _b"></span>identify <span class="_ _b"></span>Melanoma inducing <span class="_ _b"></span>mutations <span class="_ _b"></span>and <span class="_ _a"></span>their <span class="_ _b"></span>types </span></div><div class="t m0 x12 h7 y2b ff2 fs6 fc0 sc0 ls0 ws0">in genes</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf4" class="pf w0 h0" data-page-no="4"><div class="pc pc4 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg4.png"/><div class="t m0 x13 h1 y2c ff1 fs0 fc0 sc0 ls0 ws0">Motivation to Propose a Substitute for the  </div><div class="t m0 x13 h1 y2d ff1 fs0 fc0 sc0 ls0 ws0">Existing Clinical Approaches</div><div class="t m0 x14 h9 y2e ff1 fs8 fc0 sc0 ls0 ws0">Motivation for Diagnosing Melanoma using Dermoscopic Feature Analysis</div><div class="t m0 x15 ha y2f ff3 fs9 fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Difficulty in detecting all the relevant dermoscopic features from skin lesions manually.</span></div><div class="t m0 x15 ha y30 ff3 fs9 fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Based <span class="_ _a"></span>on <span class="_ _b"></span>the <span class="_ _b"></span>dermoscopic <span class="_ _8"></span>features <span class="_ _9"></span>identified <span class="_ _9"></span>it <span class="_ _b"></span>is <span class="_ _b"></span>challenging <span class="_ _a"></span>to <span class="_ _8"></span>detect <span class="_ _a"></span>the <span class="_ _b"></span>combinations </span></div><div class="t m0 x16 ha y31 ff2 fs9 fc0 sc0 ls0 ws0">of these salient features that have a higher possibility of being diagnosed with Melanoma.</div><div class="t m0 x15 hb y32 ff3 fs8 fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2 fs9">Studies <span class="_ _5"> </span>have <span class="_ _4"> </span>shown <span class="_ _5"> </span>that <span class="_ _e"> </span>a <span class="_ _e"> </span>Dermoscopy <span class="_ _5"> </span>may <span class="_ _e"> </span>even <span class="_ _4"> </span>lower <span class="_ _6"> </span>the <span class="_ _e"> </span>diagnostic <span class="_ _f"> </span>accuracy <span class="_ _4"> </span>when </span></div><div class="t m0 x16 ha y33 ff2 fs9 fc0 sc0 ls0 ws0">being performed by <span class="_ _9"></span>inexperienced <span class="_ _9"></span>dermatologists and <span class="_ _a"></span>therefore <span class="_ _10"></span>a dermoscopy is <span class="_ _a"></span>required </div><div class="t m0 x16 ha y34 ff2 fs9 fc0 sc0 ls0 ws0">to be performed by a dermatologist with a load of experience on skin lesions. </div><div class="t m0 x15 ha y35 ff3 fs9 fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Even when an <span class="_ _a"></span>expert dermatologist uses dermoscopy for diagnosis <span class="_ _a"></span>the <span class="_ _a"></span>accuracy <span class="_ _9"></span>level <span class="_ _9"></span>of it </span></div><div class="t m0 x16 ha y36 ff2 fs9 fc0 sc0 ls0 ws0">is estimated to be between 75-84%.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf5" class="pf w0 h0" data-page-no="5"><div class="pc pc5 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg5.png"/><div class="t m0 x17 h9 y37 ff1 fs8 fc0 sc0 ls0 ws0">Diagnosing Melanoma using DNA Sequence Analysis</div><div class="t m0 x18 hc y38 ff3 fsa fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Present way of <span class="_ _a"></span>getting a <span class="_ _a"></span>biopsy <span class="_ _a"></span>to <span class="_ _a"></span>detect cancer <span class="_ _a"></span>needs to be <span class="_ _a"></span>taken <span class="_ _11"></span>from the location that is <span class="_ _a"></span>close </span></div><div class="t m0 x19 hc y39 ff2 fsa fc0 sc0 ls0 ws0">to the symptoms detected area.</div><div class="t m0 x18 hc y3a ff3 fsa fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">A <span class="_ _4"> </span>biopsy <span class="_ _6"> </span>takes <span class="_ _6"></span>about <span class="_ _6"> </span>7-10 <span class="_ _8"></span>days <span class="_ _7"> </span>to <span class="_ _4"> </span>get <span class="_ _7"> </span>the <span class="_ _7"> </span>results <span class="_ _6"></span>but <span class="_ _7"> </span>a <span class="_ _4"> </span>DNA <span class="_ _7"> </span>sequence <span class="_ _7"> </span>can <span class="_ _4"> </span>be <span class="_ _7"> </span>taken <span class="_ _6"></span>within <span class="_ _7"> </span>24 </span></div><div class="t m0 x19 hc y3b ff2 fsa fc0 sc0 ls0 ws0">hours and the process of obtaining the samples to test is also simple.</div><div class="t m0 x18 hc y3c ff3 fsa fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">When <span class="_ _b"></span>the <span class="_ _b"></span>disease is <span class="_ _b"></span>passed <span class="_ _a"></span>to <span class="_ _a"></span>the <span class="_ _b"></span>offspring <span class="_ _b"></span>genetically, <span class="_ _a"></span>then <span class="_ _a"></span>the <span class="_ _b"></span>mutations <span class="_ _a"></span>could be <span class="_ _b"></span>identified <span class="_ _9"></span>in </span></div><div class="t m0 x19 hc y3d ff2 fsa fc0 sc0 ls0 ws0">any cell not limiting to just the skin cells.</div><div class="t m0 x18 hc y3e ff3 fsa fc0 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">As the molecular <span class="_ _10"></span>structure <span class="_ _9"></span>of the sample <span class="_ _9"></span>taken <span class="_ _9"></span>is extensively <span class="_ _12"></span>checked <span class="_ _10"></span>in a biopsy the cost involved </span></div><div class="t m0 x19 hc y3f ff2 fsa fc0 sc0 ls0 ws0">is <span class="_ _8"></span>also <span class="_ _8"></span>higher <span class="_ _c"></span>than <span class="_ _8"></span>a <span class="_ _c"></span>genetic <span class="_ _8"></span>test <span class="_ _8"></span>that <span class="_ _8"></span>provides the <span class="_ _c"></span>DNA <span class="_ _8"></span>sequence. <span class="_ _8"></span>But <span class="_ _c"></span>the <span class="_ _8"></span>biopsy <span class="_ _8"></span>results <span class="_ _a"></span>have <span class="_ _b"></span>a </div><div class="t m0 x19 hc y40 ff2 fsa fc0 sc0 ls0 ws0">higher possibility of having more accurate results.</div><div class="t m0 x1a h5 y41 ff1 fs3 fc0 sc0 ls0 ws0">Motivation (cont.)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf6" class="pf w0 h0" data-page-no="6"><div class="pc pc6 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg6.png"/><div class="t m0 x1b h5 y42 ff1 fs3 fc0 sc0 ls0 ws0">Comparison of <span class="_ _10"></span>Other’s Work</div><div class="t m0 x1c hc y43 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Research <span class="_ _a"></span>on <span class="_ _c"></span>Melanoma <span class="_ _8"></span>detection <span class="_ _8"></span>using <span class="_ _c"></span>image <span class="_ _c"></span>processing <span class="_ _a"></span>has <span class="_ _c"></span>been <span class="_ _c"></span>performed <span class="_ _b"></span>mainly <span class="_ _c"></span>giving <span class="_ _c"></span>concern <span class="_ _c"></span>to </span></div><div class="t m0 xb hc y44 ff2 fsa fc0 sc0 ls0 ws0">the ABCD features.</div><div class="t m0 x1c hc y45 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">There <span class="_ _5"> </span>are <span class="_ _5"> </span>also <span class="_ _e"> </span>other <span class="_ _e"> </span>research <span class="_ _6"> </span>projects <span class="_ _6"> </span>in <span class="_ _f"> </span>which <span class="_ _f"> </span>they <span class="_ _e"> </span>have <span class="_ _4"> </span>not <span class="_ _f"> </span>specifically <span class="_ _7"> </span>extracted <span class="_ _6"> </span>the <span class="_ _e"> </span>dermoscopic </span></div><div class="t m0 xb hc y46 ff2 fsa fc0 sc0 ls0 ws0">features <span class="_ _a"></span>but <span class="_ _6"></span>have <span class="_ _8"></span>performed <span class="_ _8"></span>extensive <span class="_ _8"></span>preprocessing <span class="_ _b"></span>and <span class="_ _6"></span>then <span class="_ _c"></span>used <span class="_ _13"></span>advanced <span class="_ _b"></span>classification algorithms </div><div class="t m0 xb hc y47 ff2 fsa fc0 sc0 ls0 ws0">on the lesions to diagnose Melanoma. </div><div class="t m0 x1c hc y48 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">In <span class="_ _8"></span>some <span class="_ _8"></span>research <span class="_ _9"></span>projects <span class="_ _12"></span>the <span class="_ _c"></span>focus <span class="_ _a"></span>is <span class="_ _c"></span>given <span class="_ _b"></span>towards <span class="_ _9"></span>detection <span class="_ _b"></span>of <span class="_ _8"></span>a <span class="_ _8"></span>single <span class="_ _a"></span>dermoscopic <span class="_ _8"></span>feature detection </span></div><div class="t m0 xb hc y49 ff2 fsa fc0 sc0 ls0 ws0">so that such features could be used to diagnose different skin diseases. </div><div class="t m0 x1c hc y4a ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">There <span class="_ _b"></span>are <span class="_ _b"></span>other <span class="_ _c"></span>various <span class="_ _b"></span>research as <span class="_ _8"></span>well <span class="_ _b"></span>that <span class="_ _c"></span>were <span class="_ _a"></span>performed <span class="_ _a"></span>to <span class="_ _c"></span>increase the <span class="_ _c"></span>efficiency <span class="_ _c"></span>of <span class="_ _8"></span>the <span class="_ _c"></span>Melanoma </span></div><div class="t m0 xb hc y4b ff2 fsa fc0 sc0 ls0 ws0">detection <span class="_ _c"></span>process <span class="_ _8"></span>with <span class="_ _6"> </span>lower <span class="_ _b"></span>resources <span class="_ _a"></span>and <span class="_ _6"> </span>to <span class="_ _13"></span>provide <span class="_ _b"></span>more <span class="_ _c"></span>mobility <span class="_ _13"></span>to <span class="_ _13"></span>the <span class="_ _6"></span>solution <span class="_ _13"></span>such <span class="_ _13"></span>as <span class="_ _6"></span>by <span class="_ _8"></span>using <span class="_ _6"> </span>a </div><div class="t m0 xb hc y4c ff2 fsa fc0 sc0 ls0 ws0">mobile application through which the images to be analysed are being captured without a dermoscopy.  </div><div class="t m0 x1d hd y4d ff1 fsa fc0 sc0 ls0 ws0">In <span class="_ _b"></span>our <span class="_ _8"></span>research <span class="_ _12"></span>the <span class="_ _8"></span>focus <span class="_ _a"></span>is <span class="_ _8"></span>being <span class="_ _b"></span>given <span class="_ _b"></span>towards <span class="_ _9"></span>detecting <span class="_ _a"></span>as <span class="_ _8"></span>many <span class="_ _8"></span>as <span class="_ _8"></span>possible salient <span class="_ _8"></span>dermoscopic <span class="_ _b"></span>features </div><div class="t m0 x1d hd y4e ff1 fsa fc0 sc0 ls0 ws0">in <span class="_ _b"></span>addition <span class="_ _b"></span>to <span class="_ _b"></span>the <span class="_ _8"></span>ABCD <span class="_ _a"></span>features from the <span class="_ _b"></span>lesion <span class="_ _a"></span>so <span class="_ _b"></span>that <span class="_ _8"></span>the <span class="_ _b"></span>accuracy <span class="_ _a"></span>of <span class="_ _b"></span>the <span class="_ _8"></span>diagnosis <span class="_ _a"></span>process <span class="_ _a"></span>will <span class="_ _b"></span>be <span class="_ _b"></span>higher. </div><div class="t m0 x1d hd y4f ff1 fsa fc0 sc0 ls0 ws0">Also <span class="_ _6"> </span>we <span class="_ _6"></span>want <span class="_ _13"></span>to <span class="_ _6"> </span>explore <span class="_ _b"></span>how <span class="_ _13"></span>various <span class="_ _6"> </span>classification <span class="_ _8"></span>approaches <span class="_ _c"></span>work <span class="_ _13"></span>in <span class="_ _7"> </span>this <span class="_ _6"> </span>domain <span class="_ _6"> </span>based <span class="_ _6"> </span>on <span class="_ _6"> </span>the <span class="_ _7"> </span>extracted </div><div class="t m0 x1d hd y50 ff1 fsa fc0 sc0 ls0 ws0">features so that the best approach could be adapted in the Melanoma diagnosis process.  </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf7" class="pf w0 h0" data-page-no="7"><div class="pc pc7 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg7.png"/><div class="t m0 x13 hc y51 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Many <span class="_ _5"> </span>researchers <span class="_ _c"></span>have <span class="_ _6"> </span>used <span class="_ _4"> </span>various <span class="_ _4"> </span>alignment <span class="_ _4"> </span>techniques <span class="_ _4"> </span>to <span class="_ _4"> </span>find <span class="_ _7"> </span>mutations <span class="_ _4"> </span>in <span class="_ _5"> </span>genes, <span class="_ _4"> </span>both <span class="_ _5"> </span>wet <span class="_ _4"> </span>and <span class="_ _4"> </span>dry </span></div><div class="t m0 x1e hc y52 ff2 fsa fc0 sc0 ls0 ws0">sequencing <span class="_ _8"></span>techniques <span class="_ _8"></span>such <span class="_ _8"></span>as <span class="_ _8"></span>pyrosequencing, micoarraying, <span class="_ _a"></span>to <span class="_ _8"></span>Next <span class="_ _b"></span>Generation Sequencing, <span class="_ _c"></span>Needleman-</div><div class="t m0 x1e hc y53 ff2 fsa fc0 sc0 ls0 ws0">Wunsch, <span class="_ _a"></span>Smith-Waterman <span class="_ _0"></span>etc. <span class="_ _b"></span>Since <span class="_ _c"></span>in <span class="_ _8"></span>this <span class="_ _c"></span>research <span class="_ _9"></span>we <span class="_ _b"></span>focus <span class="_ _8"></span>on <span class="_ _8"></span>specific genes <span class="_ _8"></span>that <span class="_ _8"></span>affect <span class="_ _a"></span>Melanoma, <span class="_ _b"></span>Smith </div><div class="t m0 x1e hc y54 ff2 fsa fc0 sc0 ls0 ws0">Waterman algorithm is chosen to proceed with.</div><div class="t m0 x13 hc y55 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">There <span class="_ _a"></span>are <span class="_ _b"></span>a <span class="_ _8"></span>few <span class="_ _a"></span>research <span class="_ _9"></span>that <span class="_ _8"></span>focus <span class="_ _b"></span>on <span class="_ _8"></span>finding <span class="_ _a"></span>the <span class="_ _8"></span>similarity <span class="_ _a"></span>of <span class="_ _8"></span>a <span class="_ _8"></span>given <span class="_ _b"></span>DNA <span class="_ _8"></span>sequence <span class="_ _8"></span>using <span class="_ _c"></span>fuzzy <span class="_ _8"></span>concepts. </span></div><div class="t m0 x1e hc y56 ff2 fsa fc0 sc0 ls0 ws0">Concept <span class="_ _a"></span>papers <span class="_ _8"></span>have <span class="_ _a"></span>included <span class="_ _b"></span>that <span class="_ _8"></span>it <span class="_ _c"></span>is <span class="_ _8"></span>wise <span class="_ _c"></span>to <span class="_ _8"></span>treat the <span class="_ _8"></span>subject’s DNA <span class="_ _8"></span>sequence <span class="_ _c"></span>as <span class="_ _8"></span>a <span class="_ _8"></span>fuzzy <span class="_ _8"></span>sequence <span class="_ _c"></span>while </div><div class="t m0 x1e hc y57 ff2 fsa fc0 sc0 ls0 ws0">considering <span class="_ _a"></span>the reference <span class="_ _12"></span>gene sequence <span class="_ _a"></span>as <span class="_ _a"></span>the crisp <span class="_ _a"></span>sequence. <span class="_ _a"></span>There are <span class="_ _9"></span>almost less/no <span class="_ _9"></span>research <span class="_ _12"></span>work that </div><div class="t m0 x1e hc y58 ff2 fsa fc0 sc0 ls0 ws0">focus on deriving a new sequence with less fragmentation.</div><div class="t m0 x13 hc y59 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Existing research <span class="_ _10"></span>work <span class="_ _a"></span>also highlights <span class="_ _a"></span>that <span class="_ _a"></span>age <span class="_ _a"></span>plays a prominent role <span class="_ _12"></span>in <span class="_ _a"></span>increasing mutations in <span class="_ _a"></span>genes <span class="_ _a"></span>when </span></div><div class="t m0 x1e hc y5a ff2 fsa fc0 sc0 ls0 ws0">a <span class="_ _14"> </span>patient <span class="_ _14"> </span>is <span class="_ _14"> </span>becoming <span class="_ _15"> </span>older. <span class="_ _5"> </span>This <span class="_ _16"> </span>fact <span class="_ _15"> </span>is <span class="_ _14"> </span>taken <span class="_ _17"> </span>into <span class="_ _14"> </span>consideration <span class="_ _17"> </span>when <span class="_ _14"> </span>developing <span class="_ _17"> </span>the <span class="_ _15"> </span>algorithm <span class="_ _15"> </span>to </div><div class="t m0 x1e hc y5b ff2 fsa fc0 sc0 ls0 ws0">defragment the entered DNA sequences </div><div class="t m0 x1d hd y5c ff1 fsa fc0 sc0 ls0 ws0">In <span class="_ _b"></span>our <span class="_ _8"></span>research <span class="_ _12"></span>the <span class="_ _8"></span>focus <span class="_ _a"></span>is <span class="_ _8"></span>being <span class="_ _b"></span>given <span class="_ _b"></span>in <span class="_ _8"></span>this <span class="_ _b"></span>module <span class="_ _a"></span>to <span class="_ _8"></span>identify <span class="_ _b"></span>the <span class="_ _8"></span>similarity <span class="_ _a"></span>of <span class="_ _b"></span>the <span class="_ _8"></span>entered DNA <span class="_ _8"></span>sequence </div><div class="t m0 x1d hd y5d ff1 fsa fc0 sc0 ls0 ws0">and <span class="_ _13"></span>to <span class="_ _13"></span>find <span class="_ _8"></span>whether <span class="_ _6"></span>the <span class="_ _13"></span>input <span class="_ _13"></span>is <span class="_ _13"></span>overly <span class="_ _8"></span>fragmented, <span class="_ _8"></span>and <span class="_ _13"></span>if <span class="_ _13"></span>it <span class="_ _13"></span>is, <span class="_ _13"></span>propose <span class="_ _8"></span>a <span class="_ _13"></span>sequence <span class="_ _6"></span>with <span class="_ _c"></span>less <span class="_ _c"></span>fragmentation </div><div class="t m0 x1d hd y5e ff1 fsa fc0 sc0 ls0 ws0">and then go into finding the mutations available. </div><div class="t m0 x1b h5 y42 ff1 fs3 fc0 sc0 ls0 ws0">Comparison of <span class="_ _10"></span>Other’s Work (cont.)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf8" class="pf w0 h0" data-page-no="8"><div class="pc pc8 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg8.png"/><div class="t m0 x1f hc y5f ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">In the early development state of DNA/RNA, many researchers have been found models to turn DNA/RNA </span></div><div class="t m0 x20 hc y60 ff2 fsa fc0 sc0 ls0 ws0">into respective amino acid sequences or primary protein structures. Almost all models are based on the </div><div class="t m0 x20 hc y61 ff2 fsa fc0 sc0 ls0 ws0">three-state context model. The genetic codon table has been developed to convert DNA/RNA to amino acid </div><div class="t m0 x20 hc y62 ff2 fsa fc0 sc0 ls0 ws0">sequences using early studies such as the three-state context model, prediction of protein-coding regions </div><div class="t m0 x20 hc y63 ff2 fsa fc0 sc0 ls0 ws0">using signal processing etc.</div><div class="t m0 x1f hc y64 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">There are few kinds of research about Melanoma detection using protein secondary structure based on </span></div><div class="t m0 x20 hc y65 ff2 fsa fc0 sc0 ls0 ws0">calculating thermodynamic free energy or called it free energy. But, it is costly in both time and computer </div><div class="t m0 x20 hc y66 ff2 fsa fc0 sc0 ls0 ws0">resources. The reliability issue exists since the secondary protein structures used to calculate free energy are </div><div class="t m0 x20 hc y67 ff2 fsa fc0 sc0 ls0 ws0">predicted from the primary structures.  </div><div class="t m0 x1f hc y68 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Some research works about vectorising DNA/RNA or amino acid sequences, such as count distribution, <span class="_ _12"></span>pre-</span></div><div class="t m0 x20 hc y69 ff2 fsa fc0 sc0 ls0 ws0">defined specific<span class="_ _18"></span> kernels, sequence embedding using NLP techniques, etc. </div><div class="t m0 x1b hd y6a ff1 fsa fc0 sc0 ls0 ws0">Our <span class="_ _19"> </span>research <span class="_ _17"> </span>focuses <span class="_ _19"> </span>on <span class="_ _19"> </span>Melanoma <span class="_ _1a"> </span>detection <span class="_ _1a"> </span>based <span class="_ _1a"> </span>on <span class="_ _19"> </span>primary <span class="_ _19"> </span>protein <span class="_ _16"> </span>structure <span class="_ _16"> </span>by <span class="_ _1a"> </span>designing <span class="_ _19"> </span>and </div><div class="t m0 x1b hd y6b ff1 fsa fc0 sc0 ls0 ws0">developing <span class="_ _b"></span>a <span class="_ _13"></span>supervised <span class="_ _c"></span>learning <span class="_ _8"></span>model <span class="_ _6"></span>that <span class="_ _c"></span>takes <span class="_ _8"></span>the <span class="_ _6"></span>key <span class="_ _8"></span>input <span class="_ _13"></span>as <span class="_ _13"></span>a <span class="_ _13"></span>vectorised <span class="_ _b"></span>amino <span class="_ _6"></span>acid <span class="_ _c"></span>sequence <span class="_ _6"> </span>in <span class="_ _13"></span>an </div><div class="t m0 x1b hd y6c ff1 fsa fc0 sc0 ls0 ws0">efficiently <span class="_ _1b"> </span>and <span class="_ _f"> </span>effectively <span class="_ _e"> </span>way <span class="_ _f"> </span>to <span class="_ _f"> </span>overcome <span class="_ _5"> </span>high <span class="_ _f"> </span>time <span class="_ _1b"> </span>and <span class="_ _1b"> </span>computational <span class="_ _f"> </span>costs <span class="_ _1b"> </span>that <span class="_ _1b"> </span>exists <span class="_ _e"> </span>in <span class="_ _1b"> </span>other’s <span class="_ _1b"> </span>work </div><div class="t m0 x1b hd y6d ff1 fsa fc0 sc0 ls0 ws0">related to Melanoma detection. </div><div class="t m0 x1b h5 y42 ff1 fs3 fc0 sc0 ls0 ws0">Comparison of <span class="_ _10"></span>Other’s Work (cont.)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf9" class="pf w0 h0" data-page-no="9"><div class="pc pc9 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg9.png"/><div class="t m0 x1d h5 y6e ff1 fs3 fc0 sc0 ls0 ws0">Research Aim and Objectives</div><div class="t m0 x1d he y6f ff1 fs9 fc0 sc0 ls0 ws0">Aim</div><div class="t m0 x1d hf y70 ff2 fsb fc0 sc0 ls0 ws0">This <span class="_ _8"></span>research aims <span class="_ _8"></span>to <span class="_ _8"></span>develop a <span class="_ _8"></span>system <span class="_ _c"></span>to <span class="_ _8"></span>identify <span class="_ _8"></span>the <span class="_ _8"></span>risk <span class="_ _8"></span>of <span class="_ _c"></span>being <span class="_ _8"></span>diagnosed <span class="_ _8"></span>from <span class="_ _a"></span>Melanoma <span class="_ _b"></span>by <span class="_ _8"></span>examining the <span class="_ _b"></span>dermoscopic </div><div class="t m0 x1d hf y71 ff2 fsb fc0 sc0 ls0 ws0">images <span class="_ _13"> </span>of <span class="_ _6"> </span>the <span class="_ _13"> </span>skin <span class="_ _6"> </span>lesion <span class="_ _c"></span>using <span class="_ _13"> </span>computer <span class="_ _6"> </span>vision <span class="_ _6"> </span>and <span class="_ _13"> </span>classification <span class="_ _8"></span>techniques <span class="_ _13"></span>or <span class="_ _13"> </span>by <span class="_ _13"></span>analysing <span class="_ _6"> </span>the <span class="_ _13"> </span>mutations <span class="_ _6"> </span>occurred <span class="_ _c"></span>in <span class="_ _13"></span>the </div><div class="t m0 x1d hf y72 ff2 fsb fc0 sc0 ls0 ws0">gene <span class="_ _13"></span>DNA <span class="_ _13"></span>sequences <span class="_ _13"></span>of <span class="_ _13"> </span>a <span class="_ _13"> </span>person <span class="_ _6"> </span>along <span class="_ _c"></span>with <span class="_ _13"> </span>the <span class="_ _13"> </span>changes <span class="_ _6"> </span>in <span class="_ _13"> </span>the <span class="_ _13"> </span>protein <span class="_ _c"></span>primary <span class="_ _13"> </span>structure <span class="_ _c"></span>due <span class="_ _13"> </span>to <span class="_ _6"> </span>the <span class="_ _13"></span>mutation(s) <span class="_ _13"> </span>in <span class="_ _13"> </span>the <span class="_ _13"> </span>DNA </div><div class="t m0 x1d hf y73 ff2 fsb fc0 sc0 ls0 ws0">sequence.</div><div class="t m0 x1d he y74 ff1 fs9 fc0 sc0 ls0 ws0">Objectives</div><div class="t m0 x21 hf y75 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Critically review the literature on the existing skin disease diagnosis approaches that are being used to detect Melanoma.</span></div><div class="t m0 x21 hf y76 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Obtain <span class="_ _7"> </span>background <span class="_ _13"> </span>knowledge <span class="_ _c"></span>required <span class="_ _c"></span>in <span class="_ _7"> </span>various <span class="_ _7"> </span>technologies, <span class="_ _6"> </span>techniques <span class="_ _6"> </span>and <span class="_ _4"> </span>methodologies <span class="_ _6"> </span>that <span class="_ _4"> </span>could <span class="_ _6"> </span>be <span class="_ _7"> </span>used <span class="_ _7"> </span>to </span></div><div class="t m0 x22 hf y77 ff2 fsb fc0 sc0 ls0 ws0">increase the efficiency and the effectiveness of the proposed solution.</div><div class="t m0 x21 hf y78 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Extract advanced salient features from dermoscopic images and analyse the extracted features to diagnose Melanoma.</span></div><div class="t m0 x21 hf y79 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Training a classification model to diagnose Melanoma based on the extracted dermoscopic features. </span></div><div class="t m0 x21 hf y7a ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Analyse gene sequences and detect DNA mutations that have occurred along with the type of the mutation.</span></div><div class="t m0 x21 hf y7b ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Turn mutated DNA sequences into primary protein structures or amino acid sequences included in the mutated gene. </span></div><div class="t m0 x21 hf y7c ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Vectorising <span class="_ _6"> </span>amino  acid  sequences  and  building  a  supervised <span class="_ _e"> </span>learning <span class="_ _5"> </span>model  can  identify  Melanoma  and  provide <span class="_ _4"> </span>a </span></div><div class="t m0 x22 hf y7d ff2 fsb fc0 sc0 ls0 ws0">descriptive  report <span class="_ _e"> </span>by <span class="_ _e"> </span>designing  and <span class="_ _1b"> </span>developing <span class="_ _5"> </span>machine  learning <span class="_ _e"> </span>or  neural  network <span class="_ _e"> </span>architectural <span class="_ _4"> </span>solutions <span class="_ _1b"> </span>such <span class="_ _1b"> </span>as </div><div class="t m0 x22 hf y7e ff2 fsb fc0 sc0 ls0 ws0">Logistic Regression, KNN, LSTM, RNN or CNN.</div><div class="t m0 x21 hf y7f ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Design <span class="_ _c"></span>and <span class="_ _c"></span>develop <span class="_ _a"></span>a <span class="_ _c"></span>prototype <span class="_ _8"></span>that <span class="_ _13"></span>could <span class="_ _8"></span>assist <span class="_ _c"></span>in <span class="_ _c"></span>diagnosing <span class="_ _13"> </span>Melanoma <span class="_ _8"></span>at <span class="_ _13"></span>its <span class="_ _c"></span>initial <span class="_ _c"></span>stages <span class="_ _c"></span>so <span class="_ _c"></span>that <span class="_ _c"></span>proper <span class="_ _c"></span>treatment </span></div><div class="t m0 x22 hf y80 ff2 fsb fc0 sc0 ls0 ws0">could be started immediately for the patient.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfa" class="pf w0 h0" data-page-no="a"><div class="pc pca w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bga.png"/><div class="t m0 x1d h5 y81 ff1 fs3 fc1 sc0 ls0 ws0">Project Approach</div><div class="t m0 x23 h10 y82 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Two <span class="_ _17"> </span>different <span class="_ _15"> </span>approaches <span class="_ _14"> </span>are <span class="_ _14"> </span>being <span class="_ _1a"> </span>used <span class="_ _1a"> </span>to </span></div><div class="t m0 x24 h10 y83 ff2 fsc fc3 sc0 ls0 ws0">diagnose <span class="_ _16"> </span>Melanoma <span class="_ _16"> </span>namely <span class="_ _1a"> </span>via <span class="_ _1a"> </span>dermoscopic </div><div class="t m0 x24 h10 y84 ff2 fsc fc3 sc0 ls0 ws0">images and DNA sequences. </div><div class="t m0 x23 h10 y85 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">In <span class="_ _1c"> </span>the <span class="_ _1c"> </span>first <span class="_ _1d"> </span>approach <span class="_ _1e"> </span>image <span class="_ _1c"> </span>processing </span></div><div class="t m0 x24 h10 y86 ff2 fsc fc3 sc0 ls0 ws0">techniques <span class="_ _15"> </span>are <span class="_ _15"> </span>being <span class="_ _16"> </span>used <span class="_ _14"> </span>to <span class="_ _16"> </span>extract <span class="_ _17"> </span>salient </div><div class="t m0 x24 h10 y87 ff2 fsc fc3 sc0 ls0 ws0">dermoscopic <span class="_ _1f"> </span>features <span class="_ _20"> </span>when <span class="_ _1f"> </span>diagnosing </div><div class="t m0 x24 h10 y88 ff2 fsc fc3 sc0 ls0 ws0">Melanoma <span class="_ _19"> </span>using <span class="_ _21"> </span>dermoscopy <span class="_ _19"> </span>images <span class="_ _21"> </span>and <span class="_ _21"> </span>a </div><div class="t m0 x24 h10 y89 ff2 fsc fc3 sc0 ls0 ws0">classification <span class="_ _8"></span>model <span class="_ _4"> </span>will <span class="_ _7"> </span>be <span class="_ _4"> </span>trained <span class="_ _c"></span>to <span class="_ _4"> </span>diagnose </div><div class="t m0 x24 h10 y8a ff2 fsc fc3 sc0 ls0 ws0">Melanoma using the extracted features. </div><div class="t m0 x23 h10 y8b ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The <span class="_ _a"></span>next approach <span class="_ _12"></span>uses <span class="_ _a"></span>the DNA <span class="_ _a"></span>sequences and </span></div><div class="t m0 x24 h10 y8c ff2 fsc fc3 sc0 ls0 ws0">analyzes <span class="_ _8"></span>its <span class="_ _13"></span>mutations <span class="_ _13"></span>(if <span class="_ _c"></span>any) <span class="_ _13"></span>to <span class="_ _13"></span>determine <span class="_ _c"></span>how </div><div class="t m0 x24 h10 y8d ff2 fsc fc3 sc0 ls0 ws0">it <span class="_ _21"> </span>would <span class="_ _1a"> </span>be <span class="_ _21"> </span>a <span class="_ _21"> </span>cause <span class="_ _19"> </span>to <span class="_ _21"> </span>be <span class="_ _21"> </span>diagnosed <span class="_ _19"> </span>with </div><div class="t m0 x24 h10 y8e ff2 fsc fc3 sc0 ls0 ws0">Melanoma.</div><div class="t m0 x23 h10 y8f ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Then <span class="_ _22"> </span>the <span class="_ _22"> </span>mutated <span class="_ _23"> </span>gene <span class="_ _22"> </span>sequence <span class="_ _23"> </span>will <span class="_ _22"> </span>be </span></div><div class="t m0 x24 h10 y90 ff2 fsc fc3 sc0 ls0 ws0">converted <span class="_ _7"> </span>to <span class="_ _e"> </span>its <span class="_ _f"> </span>primary <span class="_ _e"> </span>protein <span class="_ _5"> </span>structure <span class="_ _4"> </span>and </div><div class="t m0 x24 h10 y91 ff2 fsc fc3 sc0 ls0 ws0">then transformed to its vector representation. </div><div class="t m0 x23 h10 y92 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Using <span class="_ _5"> </span>the <span class="_ _5"> </span>mutations <span class="_ _4"> </span>detected <span class="_ _7"> </span>and <span class="_ _5"> </span>the <span class="_ _4"> </span>primary </span></div><div class="t m0 x24 h10 y93 ff2 fsc fc3 sc0 ls0 ws0">protein <span class="_ _4"> </span>structure <span class="_ _5"> </span>as <span class="_ _f"> </span>key <span class="_ _5"> </span>inputs, <span class="_ _e"> </span>the <span class="_ _e"> </span>Melanoma </div><div class="t m0 x24 h10 y94 ff2 fsc fc3 sc0 ls0 ws0">diagnosis will be performed.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfb" class="pf w0 h0" data-page-no="b"><div class="pc pcb w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bgb.png"/><div class="t m0 x1b h5 y95 ff1 fs3 fc0 sc0 ls0 ws0">Project Approach (cont.)<span class="_ _24"></span> </div><div class="t m0 x22 h11 y96 ff4 fsc fc0 sc0 ls0 ws0">Output Format of the Melanoma Diagnosis Result </div><div class="t m0 x25 h11 y97 ff4 fsc fc0 sc0 ls0 ws0">via DNA Sequence</div><div class="t m0 x26 h11 y96 ff4 fsc fc0 sc0 ls0 ws0">Output Format of the Melanoma Diagnosis Result </div><div class="t m0 x27 h11 y97 ff4 fsc fc0 sc0 ls0 ws0">via Dermoscopic Image</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfc" class="pf w0 h0" data-page-no="c"><div class="pc pcc w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bgc.png"/><div class="t m0 xc h5 y98 ff1 fs3 fc0 sc0 ls0 ws0">Why use Two Approaches?</div><div class="t m0 xa h10 y99 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Detecting <span class="_ _6"> </span>the <span class="_ _7"> </span>dermoscopic <span class="_ _7"> </span>features <span class="_ _c"></span>that <span class="_ _7"> </span>exists <span class="_ _6"> </span>in <span class="_ _4"> </span>a <span class="_ _7"> </span>skin <span class="_ _7"> </span>lesion <span class="_ _6"> </span>using <span class="_ _7"> </span>image <span class="_ _7"> </span>processing <span class="_ _13"> </span>techniques <span class="_ _6"> </span>to <span class="_ _7"> </span>diagnose </span></div><div class="t m0 x28 h10 y9a ff2 fsc fc0 sc0 ls0 ws0">Melanoma <span class="_ _5"> </span>is <span class="_ _e"> </span>the <span class="_ _e"> </span>most <span class="_ _e"> </span>common <span class="_ _e"> </span>procedure <span class="_ _13"> </span>followed <span class="_ _6"> </span>clinically <span class="_ _5"> </span>to <span class="_ _e"> </span>detect <span class="_ _5"> </span>Melanoma <span class="_ _5"> </span>and <span class="_ _e"> </span>therefore <span class="_ _13"> </span>that <span class="_ _e"> </span>is <span class="_ _f"> </span>being </div><div class="t m0 x28 h10 y9b ff2 fsc fc0 sc0 ls0 ws0">adapted in  this research.</div><div class="t m0 xa h10 y9c ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">But <span class="_ _b"></span>in <span class="_ _b"></span>this <span class="_ _b"></span>research <span class="_ _12"></span>the <span class="_ _b"></span>analysis <span class="_ _a"></span>on <span class="_ _b"></span>dermoscopic <span class="_ _b"></span>features <span class="_ _12"></span>is <span class="_ _b"></span>being <span class="_ _b"></span>performed extensively so <span class="_ _a"></span>that <span class="_ _b"></span>no <span class="_ _b"></span>salient <span class="_ _b"></span>feature </span></div><div class="t m0 x28 h10 y9d ff2 fsc fc0 sc0 ls0 ws0">is <span class="_ _25"> </span>being <span class="_ _25"> </span>missed  out <span class="_ _25"> </span>and <span class="_ _25"> </span>therefore <span class="_ _5"> </span>the <span class="_ _25"> </span>combination <span class="_ _25"> </span>of <span class="_ _25"> </span>these <span class="_ _25"> </span>features <span class="_ _e"> </span>could  be <span class="_ _25"> </span>used  optimally <span class="_ _25"> </span>to <span class="_ _25"> </span>diagnose </div><div class="t m0 x28 h10 y9e ff2 fsc fc0 sc0 ls0 ws0">Melanoma at a higher accuracy when compared with the existing clinical approach. </div><div class="t m0 xa h10 y9f ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The <span class="_ _a"></span>major <span class="_ _9"></span>reason <span class="_ _9"></span>to use <span class="_ _a"></span>another approach <span class="_ _12"></span>like <span class="_ _9"></span>analysing the DNA <span class="_ _a"></span>sequence was to detect Melanoma <span class="_ _9"></span>at <span class="_ _a"></span>its earliest </span></div><div class="t m0 x28 h10 ya0 ff2 fsc fc0 sc0 ls0 ws0">stages so that it could be treated and cured. </div><div class="t m0 xa h10 ya1 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">In <span class="_ _a"></span>situations where Melanoma <span class="_ _9"></span>is <span class="_ _a"></span>being <span class="_ _a"></span>genetically passed to the <span class="_ _a"></span>offspring <span class="_ _a"></span>or even in other <span class="_ _a"></span>scenarios in <span class="_ _a"></span>which there </span></div><div class="t m0 x28 h10 ya2 ff2 fsc fc0 sc0 ls0 ws0">are no dermoscopic features that could be identified at the initial stage.</div><div class="t m0 xa h10 ya3 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">But <span class="_ _e"> </span>in <span class="_ _5"> </span>these <span class="_ _5"> </span>scenarios <span class="_ _5"> </span>the <span class="_ _5"> </span>diagnosis <span class="_ _5"> </span>can <span class="_ _5"> </span>be <span class="_ _5"> </span>performed <span class="_ _4"> </span>via <span class="_ _5"> </span>the <span class="_ _5"> </span>DNA <span class="_ _5"> </span>sequence <span class="_ _5"> </span>since <span class="_ _5"> </span>DNA <span class="_ _5"> </span>mutations <span class="_ _5"> </span>could <span class="_ _4"> </span>be </span></div><div class="t m0 x28 h10 ya4 ff2 fsc fc0 sc0 ls0 ws0">identified which results in unexpected behavior or inactivity of the protein functionality.</div><div class="t m0 xa h10 ya5 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Still <span class="_ _c"></span>for <span class="_ _8"></span>patients <span class="_ _8"></span>who <span class="_ _c"></span>have <span class="_ _a"></span>found <span class="_ _b"></span>lesions <span class="_ _b"></span>on <span class="_ _c"></span>their <span class="_ _8"></span>skin <span class="_ _c"></span>could <span class="_ _b"></span>use <span class="_ _8"></span>the <span class="_ _c"></span>approach <span class="_ _a"></span>to <span class="_ _c"></span>use <span class="_ _c"></span>the <span class="_ _8"></span>image <span class="_ _c"></span>processing <span class="_ _a"></span>which </span></div><div class="t m0 x28 h10 ya6 ff2 fsc fc0 sc0 ls0 ws0">detects the dermoscopic features on the lesion from which the diagnosis is performed. </div><div class="t m0 xa h10 ya7 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This approach <span class="_ _12"></span>is faster and also cost effective <span class="_ _10"></span>when compared <span class="_ _12"></span>with the <span class="_ _9"></span>DNA <span class="_ _a"></span>analysis while <span class="_ _12"></span>it provides <span class="_ _10"></span>accurate <span class="_ _9"></span>and </span></div><div class="t m0 x28 h10 ya8 ff2 fsc fc0 sc0 ls0 ws0">reliable results as well.  </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfd" class="pf w0 h0" data-page-no="d"><div class="pc pcd w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bgd.png"/><div class="t m0 x1d h5 ya9 ff1 fs3 fc0 sc0 ls0 ws0">Tech Stack and Datasets Used</div><div class="t m0 x1d he yaa ff1 fs9 fc0 sc0 ls0 ws0">Technologies</div><div class="t m0 x29 h10 yab ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Python Language</span></div><div class="t m0 x29 h10 yac ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Python Libraries</span></div><div class="t m0 x29 h10 yad ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Flask<span class="_ _9"></span> Framework</span></div><div class="t m0 x29 h10 yae ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">TensorFlow </span></div><div class="t m0 x1d h12 yaf ff1 fsc fc0 sc0 ls0 ws0">Datasets</div><div class="t m0 x1d h10 yb0 ff2 fsc fc0 sc0 ls0 ws0">The following datasets have been found up to now,</div><div class="t m0 x29 h10 yb1 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Genomic <span class="_ _25"> </span>profiles <span class="_ _5"> </span>of <span class="_ _15"> </span>ultraviolet-induced <span class="_ _4"> </span>6-4PP <span class="_ _15"> </span>susceptibilities <span class="_ _25"> </span>and <span class="_ _17"> </span>potential <span class="_ _17"> </span>new <span class="_ _25"> </span>roles  in <span class="_ _17"> </span>Melanoma </span></div><div class="t m0 x12 h10 yb2 ff2 fsc fc0 sc0 ls0 ws0">development </div><div class="t m0 x29 h10 yb3 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Cancer Browser (COSMIC) </span></div><div class="t m0 x29 h10 yb4 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Skin Cancer MNIST:10000 Dataset from Kaggle</span></div><div class="t m0 x29 h10 yb5 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Melanoma Detection Datasets from Kaggle</span></div><div class="t m0 x25 h10 yb6 ff5 fsc fc0 sc0 ls0 ws0">❖<span class="_ _26"> </span><span class="ff2">OpenCV<span class="_ _27"> </span></span>❖<span class="_ _26"> </span><span class="ff2">Scikit-learn</span></div><div class="t m0 x25 h10 yb7 ff5 fsc fc0 sc0 ls0 ws0">❖<span class="_ _26"> </span><span class="ff2">Numpy<span class="_ _28"> </span></span>❖<span class="_ _26"> </span><span class="ff2">Scikit-image</span></div><div class="t m0 x25 h10 yb8 ff5 fsc fc0 sc0 ls0 ws0">❖<span class="_ _26"> </span><span class="ff2">SciPy<span class="_ _29"> </span></span>❖<span class="_ _26"> </span><span class="ff2">Pandas</span></div><div class="t m0 x25 h10 yb9 ff5 fsc fc0 sc0 ls0 ws0">❖<span class="_ _26"> </span><span class="ff2">Matplotlib<span class="_ _2a"> </span></span>❖<span class="_ _26"> </span><span class="ff2">Keras</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfe" class="pf w0 h0" data-page-no="e"><div class="pc pce w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bge.png"/><div class="t m0 x1d h5 yba ff1 fs3 fc0 sc0 ls0 ws0">Solution in Detail</div><div class="t m0 x2a h9 ybb ff1 fs8 fc0 sc0 ls0 ws0">Module 1: Detection of Melanoma using Dermoscopic Feature Extraction and Analysis</div><div class="t m0 x1d hc ybc ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This <span class="_ _4"> </span>module <span class="_ _7"> </span>will <span class="_ _4"> </span>process <span class="_ _6"> </span>the <span class="_ _5"> </span>dermoscopic <span class="_ _7"> </span>images <span class="_ _4"> </span>and <span class="_ _4"> </span>extract <span class="_ _13"></span>dermoscopic <span class="_ _4"> </span>features <span class="_ _c"></span>that <span class="_ _4"> </span>shows <span class="_ _7"> </span>a <span class="_ _4"> </span>higher </span></div><div class="t m0 x2b hc ybd ff2 fsa fc0 sc0 ls0 ws0">possibility of being diagnosed with Melanoma.</div><div class="t m0 x1d hc ybe ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Works separately from the other two modules.</span></div><div class="t m0 x1d h10 ybf ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The <span class="_ _2b"> </span>images <span class="_ _2b"> </span>will <span class="_ _21"> </span>be <span class="_ _2b"> </span>pre-processed <span class="_ _16"> </span>using <span class="_ _2c"> </span>techniques <span class="_ _21"> </span>such <span class="_ _21"> </span>as <span class="_ _2c"> </span>image <span class="_ _2c"> </span>enhancements, <span class="_ _2c"> </span>noise <span class="_ _2c"> </span>removal, <span class="_ _19"> </span>lesion </span></div><div class="t m0 x2b h10 yc0 ff2 fsc fc0 sc0 ls0 ws0">segmentation, geometric transformations, etc. before extracting the dermoscopic features to be analyzed.</div><div class="t m0 x1d h10 yc1 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Removal <span class="_ _c"></span>of <span class="_ _13"></span>noise <span class="_ _13"> </span>includes <span class="_ _13"></span>removing <span class="_ _b"></span>of <span class="_ _6"> </span>skin <span class="_ _13"></span>hair, <span class="_ _9"></span>ink <span class="_ _13"> </span>patches <span class="_ _13"> </span>and <span class="_ _6"> </span>the <span class="_ _13"></span>lens <span class="_ _c"></span>of <span class="_ _13"> </span>the <span class="_ _6"> </span>dermoscope <span class="_ _13"></span>which <span class="_ _13"></span>causes <span class="_ _13"></span>errors </span></div><div class="t m0 x2b h10 yc2 ff2 fsc fc0 sc0 ls0 ws0">when trying to segment the lesion from its background.  </div><div class="t m0 x1d h10 yc3 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">In <span class="_ _a"></span>addition to the ABCD features, <span class="_ _10"></span>other dermoscopic features <span class="_ _10"></span>such as atypical pigment network, regression <span class="_ _0"></span>structure, </span></div><div class="t m0 x2b h10 yc4 ff2 fsc fc0 sc0 ls0 ws0">streaks, <span class="_ _5"> </span>blotches, <span class="_ _5"> </span>globules, <span class="_ _5"> </span>milky-red <span class="_ _4"> </span>areas, <span class="_ _5"> </span>rosettes <span class="_ _5"> </span>and  blue-white <span class="_ _e"> </span>veil <span class="_ _e"> </span>will  be <span class="_ _e"> </span>extracted <span class="_ _4"> </span>in <span class="_ _f"> </span>this <span class="_ _f"> </span>module <span class="_ _e"> </span>of <span class="_ _f"> </span>the </div><div class="t m0 x2b h10 yc5 ff2 fsc fc0 sc0 ls0 ws0">research. </div><div class="t m0 x1d h10 yc6 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Finally <span class="_ _4"> </span>a <span class="_ _4"> </span>classification <span class="_ _c"></span>approach <span class="_ _6"> </span>will <span class="_ _4"> </span>be <span class="_ _4"> </span>selected <span class="_ _c"></span>based <span class="_ _7"> </span>on <span class="_ _4"> </span>a <span class="_ _4"> </span>evaluation <span class="_ _6"> </span>that <span class="_ _4"> </span>would <span class="_ _6"> </span>work <span class="_ _7"> </span>best <span class="_ _4"> </span>with <span class="_ _4"> </span>the <span class="_ _4"> </span>extracted </span></div><div class="t m0 x2b h10 yc7 ff2 fsc fc0 sc0 ls0 ws0">dermoscopic features to diagnose Melanoma with a greater accuracy.</div><div class="t m0 x2c h13 yc8 ff6 fsb fc0 sc0 ls0 ws0">Output</div><div class="t m0 x2d h14 yc9 ff4 fsb fc0 sc0 ls0 ws0">Melanoma </div><div class="t m0 x2e h14 yca ff4 fsb fc0 sc0 ls0 ws0">diagnosis </div><div class="t m0 x2f h14 ycb ff4 fsb fc0 sc0 ls0 ws0">result with the </div><div class="t m0 x2d h14 ycc ff4 fsb fc0 sc0 ls0 ws0">confidence </div><div class="t m0 x30 h14 ycd ff4 fsb fc0 sc0 ls0 ws0">level </div><div class="t m0 x31 h15 yce ff4 fs8 fc0 sc0 ls0 ws0">Module 1</div><div class="t m0 x32 h13 ycf ff6 fsb fc0 sc0 ls0 ws0">Input </div><div class="t m0 x33 h14 yd0 ff4 fsb fc0 sc0 ls0 ws0">Dermoscopic </div><div class="t m0 x34 h14 yd1 ff4 fsb fc0 sc0 ls0 ws0">Images</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pff" class="pf w0 h0" data-page-no="f"><div class="pc pcf w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bgf.png"/><div class="t m0 xa hd yd2 ff1 fsa fc0 sc0 ls0 ws0">Module 2: Detection of Mutations in Gene DNA Sequences</div><div class="t m0 x14 hc yd3 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This <span class="_ _8"></span>module <span class="_ _a"></span>checks <span class="_ _b"></span>for <span class="_ _8"></span>mutations <span class="_ _b"></span>in <span class="_ _8"></span>the <span class="_ _8"></span>inputted <span class="_ _b"></span>DNA <span class="_ _8"></span>sequence <span class="_ _8"></span>by <span class="_ _a"></span>comparing <span class="_ _8"></span>it <span class="_ _8"></span>with <span class="_ _8"></span>a <span class="_ _8"></span>DNA <span class="_ _8"></span>sequence <span class="_ _8"></span>of </span></div><div class="t m0 x35 hc yd4 ff2 fsa fc0 sc0 ls0 ws0">Non-melanoma subjects. </div><div class="t m0 x14 hc yd5 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This <span class="_ _17"> </span>module  is  the <span class="_ _17"> </span>starting <span class="_ _25"> </span>point <span class="_ _25"> </span>of <span class="_ _17"> </span>the <span class="_ _25"> </span>process  in <span class="_ _25"> </span>calculating <span class="_ _f"> </span>the <span class="_ _25"> </span>probability  of  being <span class="_ _17"> </span>affected <span class="_ _f"> </span>by </span></div><div class="t m0 x35 hc yd6 ff2 fsa fc0 sc0 ls0 ws0">Melanoma using the DNA sequence and proteins evaluation.</div><div class="t m0 x14 hc yd7 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Most <span class="_ _b"></span>commonly <span class="_ _b"></span>mutated <span class="_ _a"></span>gene <span class="_ _b"></span>that <span class="_ _8"></span>causes <span class="_ _b"></span>Melanoma is <span class="_ _8"></span>CDKN2A <span class="_ _8"></span>germline <span class="_ _b"></span>wise. <span class="_ _8"></span>In <span class="_ _b"></span>this <span class="_ _8"></span>module we <span class="_ _b"></span>hope <span class="_ _8"></span>to </span></div><div class="t m0 x35 hc yd8 ff2 fsa fc0 sc0 ls0 ws0">analyze somatic mutations which are depending on the COSMIC dataset obtained.</div><div class="t m0 x14 hc yd9 ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The outputs from this module are passed as inputs to the Module 3.</span></div><div class="t m0 x14 hc yda ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This <span class="_ _8"></span>module <span class="_ _b"></span>also <span class="_ _b"></span>includes <span class="_ _8"></span>preprocessing of <span class="_ _8"></span>DNA <span class="_ _8"></span>sequences <span class="_ _8"></span>if <span class="_ _8"></span>the <span class="_ _8"></span>entered sequences <span class="_ _8"></span>are <span class="_ _a"></span>fragmented<span class="_ _12"></span>. <span class="_ _8"></span>It <span class="_ _8"></span>is </span></div><div class="t m0 x35 hc ydb ff2 fsa fc0 sc0 ls0 ws0">done <span class="_ _c"></span>by <span class="_ _c"></span>a <span class="_ _c"></span>newly <span class="_ _c"></span>developed algorithm <span class="_ _8"></span>that <span class="_ _c"></span>identifies <span class="_ _8"></span>the <span class="_ _c"></span>least <span class="_ _b"></span>mutable <span class="_ _8"></span>nucleotides <span class="_ _b"></span>and <span class="_ _13"></span>reversing <span class="_ _a"></span>them <span class="_ _13"></span>to </div><div class="t m0 x35 hc ydc ff2 fsa fc0 sc0 ls0 ws0">their corresponding reference nucleotide to generate a less fragmented sequence.</div><div class="t m0 x1d h5 ydd ff1 fs3 fc0 sc0 ls0 ws0">Solution in Detail (cont.)</div><div class="t m0 x36 h16 yde ff6 fsd fc0 sc0 ls0 ws0">Output</div><div class="t m0 x37 h17 ydf ff4 fs6 fc0 sc0 ls0 ws0">1. Reference gene </div><div class="t m0 x38 h17 ye0 ff4 fs6 fc0 sc0 ls0 ws0">DNA sequence</div><div class="t m0 x39 h17 ye1 ff4 fs6 fc0 sc0 ls0 ws0">2. Mutated gene </div><div class="t m0 x38 h17 ye2 ff4 fs6 fc0 sc0 ls0 ws0">DNA sequence </div><div class="t m0 x3a h17 ye3 ff4 fs6 fc0 sc0 ls0 ws0">3. Identified </div><div class="t m0 x38 h17 ye4 ff4 fs6 fc0 sc0 ls0 ws0">mutations and </div><div class="t m0 x3b h17 ye5 ff4 fs6 fc0 sc0 ls0 ws0">positions</div><div class="t m0 x3c h15 ye6 ff4 fs8 fc0 sc0 ls0 ws0">Module 2</div><div class="t m0 x3d h18 ye7 ff6 fsc fc0 sc0 ls0 ws0">Input</div><div class="t m0 x3e h19 ye8 ff4 fse fc0 sc0 ls0 ws0">Gene DNA </div><div class="t m0 x3f h19 ye9 ff4 fse fc0 sc0 ls0 ws0">sequence of a </div><div class="t m0 x40 h19 yea ff4 fse fc0 sc0 ls0 ws0">subject, age</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf10" class="pf w0 h0" data-page-no="10"><div class="pc pc10 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg10.png"/><div class="t m0 x1a hd yeb ff1 fsa fc0 sc0 ls0 ws0">Module 3: Detection of Melanoma based on Protein Primary Structure </div><div class="t m0 x41 hc yec ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This <span class="_ _e"> </span>module <span class="_ _4"> </span>aims <span class="_ _e"> </span>to <span class="_ _5"> </span>design <span class="_ _e"> </span>and <span class="_ _e"> </span>develop <span class="_ _6"> </span>supervised <span class="_ _5"> </span>learning <span class="_ _4"> </span>model <span class="_ _5"> </span>architecture <span class="_ _6"> </span>to <span class="_ _e"> </span>classify <span class="_ _4"> </span>Melanoma </span></div><div class="t m0 x42 hc yed ff2 fsa fc0 sc0 ls0 ws0">using the key input as primary protein structure.</div><div class="t m0 x41 hc yee ff3 fsa fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This module can be divided into four steps.</span></div><div class="t m0 x41 hc yef ff2 fsa fc0 sc0 ls0 ws0">1.<span class="_ _3"> </span>Translate <span class="_ _18"></span>original <span class="_ _8"></span>and <span class="_ _c"></span>mutated <span class="_ _a"></span>DNA <span class="_ _c"></span>sequences <span class="_ _8"></span>into <span class="_ _8"></span>protein <span class="_ _a"></span>sequences <span class="_ _8"></span>using <span class="_ _c"></span>a <span class="_ _8"></span>three-state context <span class="_ _a"></span>model, </div><div class="t m0 x42 hc yf0 ff2 fsa fc0 sc0 ls0 ws0">genetic codon tables, etc.</div><div class="t m0 x41 hc yf1 ff2 fsa fc0 sc0 ls0 ws0">2.<span class="_ _1d"> </span>Pre-processing, feature engineering and feature selection based on the COSMIC dataset. </div><div class="t m0 x41 hc yf2 ff2 fsa fc0 sc0 ls0 ws0">3.<span class="_ _1d"> </span>Vector representation of protein primary structure or amino acid sequences.</div><div class="t m0 x41 hc yf3 ff2 fsa fc0 sc0 ls0 ws0">4.<span class="_ _1"> </span>Design <span class="_ _13"></span>and <span class="_ _13"></span>develop <span class="_ _a"></span>generalized <span class="_ _b"></span>supervised <span class="_ _c"></span>learning <span class="_ _8"></span>model <span class="_ _13"></span>to <span class="_ _c"></span>classify <span class="_ _c"></span>Melanoma <span class="_ _8"></span>using <span class="_ _13"></span>machine <span class="_ _13"></span>learning </div><div class="t m0 x42 hc yf4 ff2 fsa fc0 sc0 ls0 ws0">or neural network<span class="_ _12"></span> classification algorithms with a reliable accuracy.   </div><div class="t m0 x1d h5 yf5 ff1 fs3 fc0 sc0 ls0 ws0">Solution in Detail (cont.)</div><div class="t m0 x43 h1a yf6 ff6 fs9 fc0 sc0 ls0 ws0">Output</div><div class="t m0 x44 h14 yf7 ff4 fsb fc0 sc0 ls0 ws0">Melanoma </div><div class="t m0 x43 h14 yf8 ff4 fsb fc0 sc0 ls0 ws0">diagnosis </div><div class="t m0 x45 h14 yf9 ff4 fsb fc0 sc0 ls0 ws0">result with the </div><div class="t m0 x44 h14 yfa ff4 fsb fc0 sc0 ls0 ws0">confidence </div><div class="t m0 x2c h14 yfb ff4 fsb fc0 sc0 ls0 ws0">level </div><div class="t m0 x46 h15 yfc ff4 fs8 fc0 sc0 ls0 ws0">Module 3</div><div class="t m0 x47 h1b yfd ff6 fs6 fc0 sc0 ls0 ws0">Input</div><div class="t m0 x40 h17 yfe ff4 fs6 fc0 sc0 ls0 ws0">1. Original DNA </div><div class="t m0 x48 h17 yff ff4 fs6 fc0 sc0 ls0 ws0">Sequence </div><div class="t m0 x49 h17 y100 ff4 fs6 fc0 sc0 ls0 ws0">2. Mutated DNA </div><div class="t m0 x48 h17 y101 ff4 fs6 fc0 sc0 ls0 ws0">Sequence</div><div class="t m0 x3e h17 y102 ff4 fs6 fc0 sc0 ls0 ws0">3. Mutation<span class="_ _9"></span>s</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf11" class="pf w0 h0" data-page-no="11"><div class="pc pc11 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg11.png"/><div class="t m0 x4a h5 y103 ff1 fs3 fc1 sc0 ls0 ws0">Detection of Melanoma using </div><div class="t m0 x4b h5 y104 ff1 fs3 fc1 sc0 ls0 ws0">Dermoscopic Feature Extraction and </div><div class="t m0 x4c h5 y105 ff1 fs3 fc1 sc0 ls0 ws0">Analysis </div><div class="t m0 x4d h1 y106 ff1 fs0 fc1 sc0 ls0 ws0">Module 1 - Implementation</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf12" class="pf w0 h0" data-page-no="12"><div class="pc pc12 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg12.png"/><div class="t m0 x1d h5 y107 ff1 fs3 fc1 sc0 ls0 ws0">Pre-processing Stages</div><div class="t m0 x1d ha y108 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">There <span class="_ _4"> </span>are <span class="_ _4"> </span>four <span class="_ _5"> </span>main <span class="_ _5"> </span>stages <span class="_ _f"> </span>in <span class="_ _5"> </span>the <span class="_ _e"> </span>pre-processing <span class="_ _7"> </span>phase <span class="_ _e"> </span>which <span class="_ _e"> </span>are <span class="_ _4"> </span>being <span class="_ _e"> </span>performed <span class="_ _4"> </span>in <span class="_ _e"> </span>the <span class="_ _e"> </span>below </span></div><div class="t m0 x4e ha y109 ff2 fs9 fc3 sc0 ls0 ws0">mentioned order.</div><div class="t m0 x29 ha y10a ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Hair Removal</span></div><div class="t m0 x29 ha y10b ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Ink Patch Removal</span></div><div class="t m0 x1d ha y10c ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">Hair, <span class="_ _10"></span>ink <span class="_ _8"></span>patches <span class="_ _c"></span>and <span class="_ _c"></span>lens <span class="_ _b"></span>removal are <span class="_ _a"></span>crucial <span class="_ _c"></span>to <span class="_ _c"></span>be <span class="_ _c"></span>performed <span class="_ _a"></span>at <span class="_ _c"></span>the <span class="_ _c"></span>pre-processing stage <span class="_ _c"></span>as <span class="_ _c"></span>when </span></div><div class="t m0 x4e ha y10d ff2 fs9 fc3 sc0 ls0 ws0">they <span class="_ _b"></span>existed <span class="_ _a"></span>the <span class="_ _8"></span>algorithms <span class="_ _a"></span>were not <span class="_ _b"></span>able <span class="_ _b"></span>to <span class="_ _8"></span>properly <span class="_ _a"></span>perform <span class="_ _a"></span>contour <span class="_ _8"></span>detection <span class="_ _b"></span>which <span class="_ _8"></span>is <span class="_ _b"></span>required <span class="_ _9"></span>to </div><div class="t m0 x4e ha y10e ff2 fs9 fc3 sc0 ls0 ws0">segment the lesion from the dermoscopic image.</div><div class="t m0 x1d ha y10f ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">The <span class="_ _4"> </span>reason <span class="_ _6"> </span>to <span class="_ _5"> </span>perform <span class="_ _4"> </span>lesion <span class="_ _4"> </span>segmentation <span class="_ _4"> </span>was <span class="_ _4"> </span>to <span class="_ _4"> </span>make <span class="_ _6"> </span>sure <span class="_ _7"> </span>that <span class="_ _5"> </span>all <span class="_ _5"> </span>the <span class="_ _5"> </span>features <span class="_ _13"></span>that <span class="_ _5"> </span>are <span class="_ _6"> </span>being </span></div><div class="t m0 x4e ha y110 ff2 fs9 fc3 sc0 ls0 ws0">extracted <span class="_ _12"></span>are from the <span class="_ _b"></span>skin <span class="_ _b"></span>lesion <span class="_ _a"></span>and <span class="_ _b"></span>not <span class="_ _b"></span>from any <span class="_ _b"></span>noise <span class="_ _b"></span>which <span class="_ _b"></span>might <span class="_ _b"></span>exist <span class="_ _a"></span>in <span class="_ _b"></span>the <span class="_ _b"></span>background <span class="_ _12"></span>of <span class="_ _b"></span>the </div><div class="t m0 x4e ha y111 ff2 fs9 fc3 sc0 ls0 ws0">skin lesion.</div><div class="t m0 x1d ha y112 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">When <span class="_ _d"> </span>performing <span class="_ _3"> </span>lesion <span class="_ _d"> </span>segmentation <span class="_ _d"> </span>gaussian <span class="_ _3"> </span>blur, <span class="_ _1d"> </span>thresholding <span class="_ _2e"> </span>and <span class="_ _3"> </span>morphological </span></div><div class="t m0 x4e ha y113 ff2 fs9 fc3 sc0 ls0 ws0">transformations <span class="_ _16"> </span>such <span class="_ _2f"> </span>techniques <span class="_ _19"> </span>were <span class="_ _14"> </span>required <span class="_ _15"> </span>to <span class="_ _19"> </span>obtain <span class="_ _2f"> </span>the <span class="_ _2f"> </span>most <span class="_ _2f"> </span>appropriate <span class="_ _1a"> </span>contour <span class="_ _2f"> </span>that </div><div class="t m0 x4e ha y114 ff2 fs9 fc3 sc0 ls0 ws0">captures the lesion border accurately.</div><div class="t m0 x4f ha y115 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Lens Frame Removal</span></div><div class="t m0 x4f ha y116 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Skin Lesion Extraction</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf13" class="pf w0 h0" data-page-no="13"><div class="pc pc13 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg13.png"/><div class="t m0 x13 h5 y117 ff1 fs3 fc1 sc0 ls0 ws0">Hair Removal</div><div class="t m0 x50 h1c y118 ff2 fsf fc3 sc0 ls0 ws0">The importance of performing hair removal in the pre-processing stage is to detect the </div><div class="t m0 x50 h1c y119 ff2 fsf fc3 sc0 ls0 ws0">lesion border more accurately.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf14" class="pf w0 h0" data-page-no="14"><div class="pc pc14 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg14.png"/><div class="t m0 x1f h5 y11a ff1 fs3 fc1 sc0 ls0 ws0">Ink Patches Removal</div><div class="t m0 x51 h1c y11b ff2 fsf fc3 sc0 ls0 ws0">As <span class="_ _5"> </span>the <span class="_ _e"> </span>colour <span class="_ _4"> </span>features <span class="_ _13"></span>are <span class="_ _4"> </span>being <span class="_ _5"> </span>highly <span class="_ _e"> </span>considered <span class="_ _7"></span>in <span class="_ _5"> </span>the <span class="_ _e"> </span>feature <span class="_ _13"></span>extraction <span class="_ _6"></span>stage, <span class="_ _5"> </span>it <span class="_ _e"> </span>is </div><div class="t m0 x51 h1c y11c ff2 fsf fc3 sc0 ls0 ws0">vital <span class="_ _a"></span>to remove <span class="_ _12"></span>the ink <span class="_ _a"></span>patches from <span class="_ _9"></span>the <span class="_ _a"></span>image. <span class="_ _a"></span>It <span class="_ _a"></span>also <span class="_ _a"></span>makes <span class="_ _12"></span>the <span class="_ _a"></span>lesion <span class="_ _11"></span>extraction <span class="_ _30"></span>process </div><div class="t m0 x51 h1c y11d ff2 fsf fc3 sc0 ls0 ws0">more accurate.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf15" class="pf w0 h0" data-page-no="15"><div class="pc pc15 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg15.png"/><div class="t m0 x13 h5 y11e ff1 fs3 fc1 sc0 ls0 ws0">Lens Frame Removal</div><div class="t m0 x52 h1c y11f ff3 fsf fc3 sc0 ls0 ws0">●<span class="_ _31"> </span><span class="ff2">In <span class="_ _32"> </span>some <span class="_ _32"> </span>dermoscopic </span></div><div class="t m0 x53 h1c y120 ff2 fsf fc3 sc0 ls0 ws0">images, <span class="_ _1"> </span>the <span class="_ _33"> </span>frame <span class="_ _34"> </span>of <span class="_ _1"> </span>the </div><div class="t m0 x53 h1c y121 ff2 fsf fc3 sc0 ls0 ws0">dermoscope  appears  in  the </div><div class="t m0 x53 h1c y122 ff2 fsf fc3 sc0 ls0 ws0">image <span class="_ _35"> </span>which <span class="_ _1b"> </span>makes <span class="_ _e"> </span>the <span class="_ _1b"> </span>skin </div><div class="t m0 x53 h1c y123 ff2 fsf fc3 sc0 ls0 ws0">lesion extraction inaccurate. </div><div class="t m0 x52 h1c y124 ff3 fsf fc3 sc0 ls0 ws0">●<span class="_ _31"> </span><span class="ff2">Therefore if <span class="_ _c"></span>the <span class="_ _c"></span>lens <span class="_ _8"></span>frame <span class="_ _b"></span>of </span></div><div class="t m0 x53 h1c y125 ff2 fsf fc3 sc0 ls0 ws0">the <span class="_ _36"> </span>dermoscope <span class="_ _36"> </span>appear <span class="_ _36"> </span>in </div><div class="t m0 x53 h1c y126 ff2 fsf fc3 sc0 ls0 ws0">the <span class="_ _22"> </span>image <span class="_ _22"> </span>it <span class="_ _22"> </span>needs <span class="_ _37"> </span>to <span class="_ _22"> </span>be </div><div class="t m0 x53 h1c y127 ff2 fsf fc3 sc0 ls0 ws0">cropped <span class="_ _b"></span>so <span class="_ _c"></span>that <span class="_ _13"></span>the <span class="_ _13"></span>effect <span class="_ _8"></span>of </div><div class="t m0 x53 h1c y128 ff2 fsf fc3 sc0 ls0 ws0">the <span class="_ _16"> </span>lens  frame <span class="_ _15"> </span>is <span class="_ _16"> </span>minimized </div><div class="t m0 x53 h1c y129 ff2 fsf fc3 sc0 ls0 ws0">for <span class="_ _1a"> </span>the <span class="_ _2f"> </span>lesion <span class="_ _1a"> </span>segmentation </div><div class="t m0 x53 h1c y12a ff2 fsf fc3 sc0 ls0 ws0">process.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf16" class="pf w0 h0" data-page-no="16"><div class="pc pc16 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg16.png"/><div class="t m0 xc h5 y12b ff1 fs3 fc1 sc0 ls0 ws0">Skin Lesion Extraction</div><div class="t m0 x54 h1c y12c ff2 fsf fc3 sc0 ls0 ws0">As <span class="_ _6"></span>the <span class="_ _6"></span>background <span class="_ _a"></span>of <span class="_ _6"></span>the <span class="_ _6"></span>dermoscopic <span class="_ _6"></span>image <span class="_ _7"></span>can <span class="_ _6"></span>contain <span class="_ _6"></span>unnecessary <span class="_ _c"></span>noise <span class="_ _7"></span>that <span class="_ _6"></span>would </div><div class="t m0 x54 h1c y12d ff2 fsf fc3 sc0 ls0 ws0">affect the <span class="_ _b"></span>accuracy of <span class="_ _b"></span>the <span class="_ _b"></span>feature <span class="_ _9"></span>extraction <span class="_ _12"></span>process, the <span class="_ _8"></span>skin <span class="_ _b"></span>lesion <span class="_ _a"></span>needs to <span class="_ _b"></span>be <span class="_ _8"></span>extracted </div><div class="t m0 x54 h1c y12e ff2 fsf fc3 sc0 ls0 ws0">from the image and passed to the feature extraction stage.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf17" class="pf w0 h0" data-page-no="17"><div class="pc pc17 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg17.png"/><div class="t m0 xc h1d y12f ff1 fs10 fc1 sc0 ls0 ws0">Feature Extraction</div><div class="t m0 x1d ha y130 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">There are 12 dermoscopic features that are being extracted in the feature extraction phase.</span></div><div class="t m0 x29 ha y131 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Asymmetry</span></div><div class="t m0 x29 ha y132 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Border Irregularity</span></div><div class="t m0 x29 ha y133 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Colour Variation</span></div><div class="t m0 x29 ha y134 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Diameter</span></div><div class="t m0 x29 ha y135 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Atypical Pigment Network</span></div><div class="t m0 x29 ha y136 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Irregular Streaks</span></div><div class="t m0 x1d ha y137 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">When <span class="_ _c"></span>giving <span class="_ _c"></span>concern <span class="_ _c"></span>to <span class="_ _c"></span>the <span class="_ _13"></span>number <span class="_ _c"></span>of <span class="_ _c"></span>features <span class="_ _a"></span>to <span class="_ _c"></span>be <span class="_ _13"></span>extracted, if <span class="_ _c"></span>the <span class="_ _c"></span>number <span class="_ _c"></span>of <span class="_ _13"></span>features <span class="_ _a"></span>extracted </span></div><div class="t m0 x4e ha y138 ff2 fs9 fc3 sc0 ls0 ws0">are <span class="_ _f"> </span>minimum  it  will  not <span class="_ _35"> </span>have <span class="_ _f"> </span>enough  information <span class="_ _1b"> </span>to  diagnose  Melanoma <span class="_ _1b"> </span>and  if  the  number <span class="_ _35"> </span>of </div><div class="t m0 x4e ha y139 ff2 fs9 fc3 sc0 ls0 ws0">features extracted is very high it might lead to overfitting as well when using a classifier.</div><div class="t m0 x1d ha y13a ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">It <span class="_ _8"></span>had <span class="_ _c"></span>been <span class="_ _8"></span>observed <span class="_ _b"></span>that <span class="_ _c"></span>5-20 <span class="_ _a"></span>number <span class="_ _c"></span>of <span class="_ _c"></span>features extracted would <span class="_ _a"></span>deliver <span class="_ _8"></span>acceptable <span class="_ _8"></span>classification </span></div><div class="t m0 x4e ha y13b ff2 fs9 fc3 sc0 ls0 ws0">results. <span class="_ _c"></span>Therefore, these <span class="_ _6"> </span>12 <span class="_ _6"></span>features <span class="_ _b"></span>that <span class="_ _6"></span>are <span class="_ _8"></span>being <span class="_ _6"></span>extracted <span class="_ _a"></span>in <span class="_ _6"></span>this <span class="_ _6"></span>stage <span class="_ _13"></span>for <span class="_ _13"></span>classification <span class="_ _8"></span>assist <span class="_ _6"></span>in </div><div class="t m0 x4e ha y13c ff2 fs9 fc3 sc0 ls0 ws0">obtaining more accurate and reliable results.  </div><div class="t m0 x55 ha y13d ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Regression Structure</span></div><div class="t m0 x55 ha y13e ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Blue-white Veil</span></div><div class="t m0 x55 ha y13f ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Milky-red Areas</span></div><div class="t m0 x55 ha y140 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Globules</span></div><div class="t m0 x55 ha y141 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Blotches</span></div><div class="t m0 x55 ha y142 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Rosettes</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf18" class="pf w0 h0" data-page-no="18"><div class="pc pc18 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg18.png"/><div class="t m0 x1b h5 y143 ff1 fs3 fc1 sc0 ls0 ws0">Asymmetry</div><div class="t m0 x56 ha y144 ff2 fs9 fc3 sc0 ls0 ws0">Asymmetry <span class="_ _17"> </span>means  that  the <span class="_ _17"> </span>skin </div><div class="t m0 x56 ha y145 ff2 fs9 fc3 sc0 ls0 ws0">lesion <span class="_ _19"> </span>is <span class="_ _19"> </span>not <span class="_ _21"> </span>symmetric <span class="_ _19"> </span>in <span class="_ _2f"> </span>any </div><div class="t m0 x56 ha y146 ff2 fs9 fc3 sc0 ls0 ws0">considered <span class="_ _36"> </span>axis. <span class="_ _22"> </span>Therefore <span class="_ _2b"> </span>the </div><div class="t m0 x56 h4 y147 ff2 fs2 fc3 sc0 ls0 ws0">top, <span class="_ _38"> </span>bottom, <span class="_ _38"> </span>left, <span class="_ _39"> </span>right <span class="_ _38"> </span>and </div><div class="t m0 x56 h4 y148 ff2 fs2 fc3 sc0 ls0 ws0">center <span class="_ _3a"> </span>coordinates <span class="_ _3b"> </span>were </div><div class="t m0 x56 h4 y149 ff2 fs2 fc3 sc0 ls0 ws0">obtained <span class="_ _3c"> </span>from <span class="_ _3c"> </span>which <span class="_ _3d"> </span>the </div><div class="t m0 x56 h4 y14a ff2 fs2 fc3 sc0 ls0 ws0">distance <span class="_ _13"></span>ratios <span class="_ _c"></span>were <span class="_ _8"></span>calculated </div><div class="t m0 x56 h4 y14b ff2 fs2 fc3 sc0 ls0 ws0">to detect asymmetry <span class="_ _a"></span>of <span class="_ _a"></span>the <span class="_ _a"></span>skin </div><div class="t m0 x56 h4 y14c ff2 fs2 fc3 sc0 ls0 ws0">lesion. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf19" class="pf w0 h0" data-page-no="19"><div class="pc pc19 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg19.png"/><div class="t m0 xc h5 y14d ff1 fs3 fc1 sc0 ls0 ws0">Border Irregularity</div><div class="t m0 x57 ha y14e ff2 fs9 fc3 sc0 ls0 ws0">Border irregularity is observed when the skin lesion seems to have abrupt endings which are not regular </div><div class="t m0 x57 ha y14f ff2 fs9 fc3 sc0 ls0 ws0">and ragged, notched, or blurred borders are visible. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf1a" class="pf w0 h0" data-page-no="1a"><div class="pc pc1a w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1a.png"/><div class="t m0 x58 h5 y150 ff1 fs3 fc1 sc0 ls0 ws0">Colour Variation</div><div class="t m0 x59 ha y151 ff2 fs9 fc3 sc0 ls0 ws0">The <span class="_ _17"> </span>colour  variation  of <span class="_ _17"> </span>a <span class="_ _17"> </span>skin <span class="_ _17"> </span>lesion  can <span class="_ _15"> </span>be <span class="_ _17"> </span>detected </div><div class="t m0 x59 ha y152 ff2 fs9 fc3 sc0 ls0 ws0">when <span class="_ _8"></span>multiple <span class="_ _b"></span>colours <span class="_ _b"></span>are <span class="_ _a"></span>observed <span class="_ _b"></span>such <span class="_ _8"></span>as <span class="_ _8"></span>white, <span class="_ _8"></span>brown, </div><div class="t m0 x59 ha y153 ff2 fs9 fc3 sc0 ls0 ws0">black, <span class="_ _b"></span>red, and <span class="_ _c"></span>blue. <span class="_ _8"></span>Normally <span class="_ _c"></span>more <span class="_ _a"></span>than <span class="_ _c"></span>two <span class="_ _8"></span>colours <span class="_ _b"></span>can </div><div class="t m0 x59 ha y154 ff2 fs9 fc3 sc0 ls0 ws0">be detected in a Melanoma skin lesion.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf1b" class="pf w0 h0" data-page-no="1b"><div class="pc pc1b w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1b.png"/><div class="t m0 xc h5 y155 ff1 fs3 fc1 sc0 ls0 ws0">Diameter</div><div class="t m0 x5a ha y156 ff2 fs9 fc3 sc0 ls0 ws0">The <span class="_ _8"></span>diameter <span class="_ _8"></span>feature is <span class="_ _8"></span>identified when <span class="_ _8"></span>the <span class="_ _8"></span>highest <span class="_ _8"></span>diameter <span class="_ _c"></span>to <span class="_ _8"></span>any <span class="_ _8"></span>considered <span class="_ _a"></span>axis <span class="_ _8"></span>of <span class="_ _8"></span>the <span class="_ _8"></span>area <span class="_ _a"></span>that <span class="_ _8"></span>the </div><div class="t m0 x5a ha y157 ff2 fs9 fc3 sc0 ls0 ws0">skin lesion had spread on the surface is greater than 6 millimetres.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf1c" class="pf w0 h0" data-page-no="1c"><div class="pc pc1c w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1c.png"/><div class="t m0 xc h5 y158 ff1 fs3 fc1 sc0 ls0 ws0">Atypical Pigment Network</div><div class="t m0 x50 ha y159 ff2 fs9 fc3 sc0 ls0 ws0">Atypical <span class="_ _4"> </span>pigment <span class="_ _7"> </span>networks <span class="_ _7"> </span>are <span class="_ _13"></span>observed <span class="_ _6"></span>when <span class="_ _7"> </span>a <span class="_ _4"> </span>black, <span class="_ _13"></span>brown <span class="_ _c"></span>or <span class="_ _4"> </span>grey <span class="_ _13"></span>asymmetrically <span class="_ _4"> </span>distributed <span class="_ _7"> </span>network </div><div class="t m0 x50 ha y15a ff2 fs9 fc3 sc0 ls0 ws0">like structure that spreads on the skin lesion which creates irregular holes and thick lines on the skin lesion.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf1d" class="pf w0 h0" data-page-no="1d"><div class="pc pc1d w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1d.png"/><div class="t m0 x1a h5 y15b ff1 fs3 fc1 sc0 ls0 ws0">Irregular Streaks</div><div class="t m0 x50 ha y15c ff2 fs9 fc3 sc0 ls0 ws0">Irregular steaks are identified as Bulbous and finger-like projections that could be seen spread irregularly </div><div class="t m0 x50 ha y15d ff2 fs9 fc3 sc0 ls0 ws0">at the edge of the skin lesion that may vary from tan to black in colour. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf1e" class="pf w0 h0" data-page-no="1e"><div class="pc pc1e w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1e.png"/><div class="t m0 x1b h5 y15e ff1 fs3 fc1 sc0 ls0 ws0">Regression Structure</div><div class="t m0 x50 ha y15f ff2 fs9 fc3 sc0 ls0 ws0">Regression structures are observed as white scar-like depigmentation on the clinically flat part of the skin </div><div class="t m0 x50 ha y160 ff2 fs9 fc3 sc0 ls0 ws0">lesion. In some cases this area may seem with a blue peppering effect as well.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf1f" class="pf w0 h0" data-page-no="1f"><div class="pc pc1f w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1f.png"/><div class="t m0 x23 h5 y161 ff1 fs3 fc1 sc0 ls0 ws0">Blue-white Veil</div><div class="t m0 x5b ha y162 ff2 fs9 fc3 sc0 ls0 ws0">Blue-white <span class="_ _35"> </span>veils <span class="_ _1b"> </span>are <span class="_ _f"> </span>irregular </div><div class="t m0 x5b ha y163 ff2 fs9 fc3 sc0 ls0 ws0">and <span class="_ _2b"> </span>structureless <span class="_ _21"> </span>areas <span class="_ _21"> </span>on </div><div class="t m0 x5b ha y164 ff2 fs9 fc3 sc0 ls0 ws0">the <span class="_ _13"></span>clinically <span class="_ _13"></span>elevated <span class="_ _a"></span>part <span class="_ _13"></span>of </div><div class="t m0 x5b ha y165 ff2 fs9 fc3 sc0 ls0 ws0">the <span class="_ _19"> </span>skin <span class="_ _19"> </span>lesion <span class="_ _16"> </span>with <span class="_ _19"> </span>a <span class="_ _19"> </span>blue </div><div class="t m0 x5b ha y166 ff2 fs9 fc3 sc0 ls0 ws0">pigmentation <span class="_ _3e"> </span>and <span class="_ _3e"> </span>an </div><div class="t m0 x5b ha y167 ff2 fs9 fc3 sc0 ls0 ws0">overlying <span class="_ _5"> </span>white <span class="_ _1b"> </span>ground-glass </div><div class="t m0 x5b ha y168 ff2 fs9 fc3 sc0 ls0 ws0">like <span class="_ _21"> </span>effect <span class="_ _21"> </span>which <span class="_ _3f"> </span>may <span class="_ _2b"> </span>even </div><div class="t m0 x5b ha y169 ff2 fs9 fc3 sc0 ls0 ws0">result <span class="_ _15"> </span>in <span class="_ _16"> </span>the <span class="_ _16"> </span>colour <span class="_ _15"> </span>of <span class="_ _16"> </span>that </div><div class="t m0 x5b ha y16a ff2 fs9 fc3 sc0 ls0 ws0">area <span class="_ _6"></span>appear <span class="_ _4"> </span>to <span class="_ _5"> </span>be <span class="_ _5"> </span>blue-grey </div><div class="t m0 x5b ha y16b ff2 fs9 fc3 sc0 ls0 ws0">as <span class="_ _39"> </span>well. <span class="_ _39"> </span>This <span class="_ _39"> </span>is <span class="_ _39"> </span>a <span class="_ _39"> </span>colour </div><div class="t m0 x5b ha y16c ff2 fs9 fc3 sc0 ls0 ws0">feature <span class="_ _40"> </span>that <span class="_ _41"> </span>is <span class="_ _41"> </span>being </div><div class="t m0 x5b ha y16d ff2 fs9 fc3 sc0 ls0 ws0">extracted from the lesion. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf20" class="pf w0 h0" data-page-no="20"><div class="pc pc20 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg20.png"/><div class="t m0 x59 h5 y158 ff1 fs3 fc1 sc0 ls0 ws0">Milky-red Areas</div><div class="t m0 x50 ha y16e ff2 fs9 fc3 sc0 ls0 ws0">Milky-red <span class="_ _b"></span>areas <span class="_ _8"></span>can <span class="_ _13"></span>be <span class="_ _13"></span>identified <span class="_ _b"></span>as <span class="_ _13"></span>red <span class="_ _8"></span>or <span class="_ _13"></span>pink <span class="_ _13"></span>structureless <span class="_ _b"></span>areas <span class="_ _8"></span>on <span class="_ _13"></span>the <span class="_ _13"></span>skin <span class="_ _13"></span>lesion <span class="_ _c"></span>with <span class="_ _6"></span>a <span class="_ _13"></span>red <span class="_ _b"></span>vascular </div><div class="t m0 x50 ha y16f ff2 fs9 fc3 sc0 ls0 ws0">blush <span class="_ _b"></span>on <span class="_ _b"></span>it. <span class="_ _b"></span>Two <span class="_ _10"></span>shades <span class="_ _8"></span>of <span class="_ _b"></span>red <span class="_ _9"></span>needs to <span class="_ _b"></span>be <span class="_ _b"></span>observed <span class="_ _a"></span>when <span class="_ _a"></span>extracting this <span class="_ _a"></span>feature <span class="_ _9"></span>and <span class="_ _b"></span> <span class="_ _b"></span>therefore <span class="_ _30"></span>this <span class="_ _8"></span>is <span class="_ _b"></span>also </div><div class="t m0 x50 ha y170 ff2 fs9 fc3 sc0 ls0 ws0">another colour feature that is being extracted from the skin lesion. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf21" class="pf w0 h0" data-page-no="21"><div class="pc pc21 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg21.png"/><div class="t m0 xc h5 y171 ff1 fs3 fc1 sc0 ls0 ws0">Globules</div><div class="t m0 x50 ha y172 ff2 fs9 fc3 sc0 ls0 ws0">Globules <span class="_ _12"></span>appear on a skin <span class="_ _a"></span>lesion <span class="_ _9"></span>as <span class="_ _a"></span>black <span class="_ _9"></span>or <span class="_ _a"></span>brown <span class="_ _30"></span>in colour, <span class="_ _42"></span>round <span class="_ _12"></span>to <span class="_ _a"></span>oval <span class="_ _12"></span>in <span class="_ _a"></span>shape and <span class="_ _a"></span>may be in <span class="_ _a"></span>different </div><div class="t m0 x50 ha y173 ff2 fs9 fc3 sc0 ls0 ws0">sizes which are being spread irregularly on the skin lesion. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf22" class="pf w0 h0" data-page-no="22"><div class="pc pc22 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg22.png"/><div class="t m0 x59 h5 y158 ff1 fs3 fc1 sc0 ls0 ws0">Blotches</div><div class="t m0 x58 ha y174 ff2 fs9 fc3 sc0 ls0 ws0">Blotches <span class="_ _b"></span>can <span class="_ _c"></span>be <span class="_ _c"></span>detected <span class="_ _b"></span>on <span class="_ _8"></span>a <span class="_ _c"></span>skin <span class="_ _c"></span>lesion <span class="_ _b"></span>as <span class="_ _c"></span> <span class="_ _8"></span>structureless <span class="_ _a"></span>areas <span class="_ _a"></span>in <span class="_ _c"></span>black, <span class="_ _b"></span>brown or <span class="_ _c"></span>sometimes <span class="_ _c"></span>even <span class="_ _a"></span>grey </div><div class="t m0 x58 ha y175 ff2 fs9 fc3 sc0 ls0 ws0">in colour which shows an asymmetric distribution on the skin lesion and ends abruptly.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf23" class="pf w0 h0" data-page-no="23"><div class="pc pc23 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg23.png"/><div class="t m0 x2a h1d y176 ff1 fs10 fc1 sc0 ls0 ws0">Rosettes</div><div class="t m0 x50 ha y177 ff2 fs9 fc3 sc0 ls0 ws0">Four <span class="_ _c"></span>white <span class="_ _8"></span>colour <span class="_ _8"></span>dots <span class="_ _c"></span>that <span class="_ _c"></span>are <span class="_ _b"></span>oriented <span class="_ _8"></span>in <span class="_ _c"></span>the <span class="_ _c"></span>same <span class="_ _c"></span>angle <span class="_ _b"></span>and <span class="_ _c"></span>arranged <span class="_ _a"></span>as <span class="_ _c"></span>a <span class="_ _c"></span>square <span class="_ _a"></span>resembling <span class="_ _8"></span>a <span class="_ _c"></span>four-</div><div class="t m0 x50 ha y178 ff2 fs9 fc3 sc0 ls0 ws0">leaf <span class="_ _9"></span>clover <span class="_ _12"></span>on <span class="_ _a"></span>the <span class="_ _a"></span>skin <span class="_ _a"></span>lesion. <span class="_ _a"></span>When <span class="_ _a"></span>the <span class="_ _a"></span>lesion is <span class="_ _a"></span>captured <span class="_ _9"></span>under <span class="_ _a"></span>polarized <span class="_ _9"></span>light, <span class="_ _a"></span>then <span class="_ _a"></span>rosettes appear <span class="_ _a"></span>more </div><div class="t m0 x50 ha y179 ff2 fs9 fc3 sc0 ls0 ws0">clearer on the lesion. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf24" class="pf w0 h0" data-page-no="24"><div class="pc pc24 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg24.png"/><div class="t m0 x1b h5 y17a ff1 fs3 fc1 sc0 ls0 ws0">Classification</div><div class="t m0 x1d ha y17b ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">In <span class="_ _c"></span>the <span class="_ _13"></span>classification <span class="_ _a"></span>process <span class="_ _8"></span>there <span class="_ _8"></span>were <span class="_ _a"></span>8 <span class="_ _13"></span>different <span class="_ _b"></span>classifiers <span class="_ _a"></span>that <span class="_ _13"></span>were <span class="_ _a"></span>evaluated <span class="_ _b"></span>to <span class="_ _13"></span>obtain <span class="_ _13"></span>the <span class="_ _13"></span>best </span></div><div class="t m0 x4e ha y17c ff2 fs9 fc3 sc0 ls0 ws0">suited <span class="_ _8"></span>classifier <span class="_ _a"></span>for <span class="_ _8"></span>the <span class="_ _c"></span>diagnosis <span class="_ _c"></span>of <span class="_ _c"></span>Melanoma <span class="_ _8"></span>based <span class="_ _8"></span>on <span class="_ _c"></span>the <span class="_ _c"></span>obtained <span class="_ _8"></span>feature <span class="_ _a"></span>combinations. <span class="_ _c"></span>These </div><div class="t m0 x4e ha y17d ff2 fs9 fc3 sc0 ls0 ws0">classifiers were,</div><div class="t m0 x29 ha y17e ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Support Vector Machine</span></div><div class="t m0 x29 ha y17f ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Decision Trees</span></div><div class="t m0 x29 ha y180 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Logistic Regression</span></div><div class="t m0 x29 ha y181 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">K-Nearest Neighbour</span></div><div class="t m0 x1d ha y182 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">The <span class="_ _7"> </span>Voting-based <span class="_ _b"></span>Ensemble <span class="_ _6"> </span>Method <span class="_ _4"> </span>Classification <span class="_ _c"></span>was <span class="_ _7"> </span>also <span class="_ _4"> </span>evaluated <span class="_ _c"></span>in <span class="_ _4"> </span>this <span class="_ _7"> </span>process <span class="_ _6"> </span>which <span class="_ _4"> </span>was <span class="_ _6"> </span>a </span></div><div class="t m0 x4e ha y183 ff2 fs9 fc3 sc0 ls0 ws0">combination <span class="_ _b"></span>of <span class="_ _8"></span>all <span class="_ _b"></span>the <span class="_ _8"></span>aforementioned <span class="_ _9"></span>classifiers <span class="_ _9"></span>except <span class="_ _9"></span>for <span class="_ _a"></span>the <span class="_ _8"></span>Naive <span class="_ _a"></span>Bayes <span class="_ _a"></span>as <span class="_ _b"></span>it <span class="_ _8"></span>had <span class="_ _b"></span>comparatively </div><div class="t m0 x4e ha y184 ff2 fs9 fc3 sc0 ls0 ws0">very poor evaluation results.</div><div class="t m0 x1d ha y185 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">The <span class="_ _b"></span>reason to <span class="_ _b"></span>use <span class="_ _b"></span>these <span class="_ _8"></span>classifiers <span class="_ _9"></span>was <span class="_ _b"></span>due <span class="_ _b"></span>to <span class="_ _8"></span>their <span class="_ _b"></span>extensive applications <span class="_ _8"></span>and <span class="_ _8"></span>the <span class="_ _b"></span>promising <span class="_ _a"></span>results </span></div><div class="t m0 x4e ha y186 ff2 fs9 fc3 sc0 ls0 ws0">delivered in the previous research projects in the considered domain. </div><div class="t m0 x5c ha y187 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Naive Bayes</span></div><div class="t m0 x5c ha y188 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Artificial Neural Networks</span></div><div class="t m0 x5c ha y189 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Adaboost</span></div><div class="t m0 x5c ha y18a ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _2d"> </span><span class="ff2">Gradient Boosting</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf25" class="pf w0 h0" data-page-no="25"><div class="pc pc25 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg25.png"/><div class="t m0 x5d h1e y18b ff1 fs11 fc1 sc0 ls0 ws0">Rest Api Design Documentation</div><div class="t m0 xd h12 y18c ff1 fsc fc3 sc0 ls0 ws0">API Endpoint<span class="ff2">: <span class="fc4">https://melanoma-detection-tool-api.herokuapp.com/predict-melanoma<span class="_ _12"></span>/dermoscopic-images</span></span></div><div class="t m0 xd h12 y18d ff1 fsc fc3 sc0 ls0 ws0">Method<span class="ff2">: POST</span></div><div class="t m0 xd h12 y18e ff1 fsc fc3 sc0 ls0 ws0">Request body format<span class="ff2">: Form-data</span></div><div class="t m0 xd h12 y18f ff1 fsc fc3 sc0 ls0 ws0">Required keys in request body<span class="_ _0"></span><span class="ff2">: image</span></div><div class="t m0 xd h10 y190 ff2 fsc fc3 sc0 ls0 ws0">Example request                                                                                   Example Response </div><div class="t m0 xd h10 y191 ff2 fsc fc3 sc0 ls0 ws0">  <span class="_ _43"> </span><span class="fc2">{ </span></div><div class="t m0 xd h10 y192 ff2 fsc fc2 sc0 ls0 ws0">        <span class="_ _44"> </span>&quot;image&quot; : ISIC_0024554.jpg</div><div class="t m0 xd h10 y193 ff2 fsc fc2 sc0 ls0 ws0">   <span class="_ _45"> </span>} </div><div class="t m0 xd h12 y194 ff1 fsc fc3 sc0 ls0 ws0">Response Status</div><div class="t m0 xe h10 y195 ff2 fsc fc3 sc0 ls0 ws0">200 - Request processed successfully</div><div class="t m0 xe h10 y196 ff2 fsc fc3 sc0 ls0 ws0">400 - Bad Request</div><div class="t m0 x5e h10 y197 ff2 fsc fc5 sc0 ls0 ws0">[</div><div class="t m0 x5e h10 y198 ff2 fsc fc5 sc0 ls0 ws0">   &quot;Positive&quot;,</div><div class="t m0 x5e h10 y199 ff2 fsc fc5 sc0 ls0 ws0">   90.53,</div><div class="t m0 x5e h10 y19a ff2 fsc fc5 sc0 ls0 ws0">   [ 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0 ]</div><div class="t m0 x5e h10 y19b ff2 fsc fc5 sc0 ls0 ws0">]</div><a class="l" href="https://melanoma-detection-tool-api.herokuapp.com/predict-melanoma/dermoscopic-images"><div class="d m1" style="border-style:none;position:absolute;left:165.215607px;bottom:410.813599px;width:436.269501px;height:15.845703px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="https://melanoma-detection-tool-api.herokuapp.com/predict-melanoma/dermoscopic-images"><div class="d m1" style="border-style:none;position:absolute;left:691.465576px;bottom:410.813599px;width:3.889648px;height:15.845703px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="https://melanoma-detection-tool-api.herokuapp.com/predict-melanoma/dermoscopic-images"><div class="d m1" style="border-style:none;position:absolute;left:698.090576px;bottom:410.813599px;width:129.322266px;height:15.845703px;background-color:rgba(255,255,255,0.000001);"></div></a></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf26" class="pf w0 h0" data-page-no="26"><div class="pc pc26 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg26.png"/><div class="t m0 x5f h5 y19c ff1 fs3 fc1 sc0 ls0 ws0">Detection of Mutations in </div><div class="t m0 x7 h5 y19d ff1 fs3 fc1 sc0 ls0 ws0">Gene DNA Sequences </div><div class="t m0 x4d h1 y19e ff1 fs0 fc1 sc0 ls0 ws0">Module 2 - Implementation</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf27" class="pf w0 h0" data-page-no="27"><div class="pc pc27 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg27.png"/><div class="t m0 x1d h5 y19f ff1 fs3 fc1 sc0 ls0 ws0">Overview</div><div class="t m0 x21 hf y1a0 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">The <span class="_ _25"> </span>subject’s <span class="_ _35"> </span>input <span class="_ _17"> </span>DNA <span class="_ _17"> </span>is <span class="_ _17"> </span>checked  for <span class="_ _25"> </span>fragmentation <span class="_ _25"> </span>by </span></div><div class="t m0 x22 hf y1a1 ff2 fsb fc3 sc0 ls0 ws0">finding the similarity percentage using fuzzy concepts. </div><div class="t m0 x21 hf y1a2 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">If <span class="_ _1a"> </span>the <span class="_ _1a"> </span>similarity <span class="_ _16"> </span>ratio <span class="_ _16"> </span>is <span class="_ _19"> </span>within <span class="_ _1a"> </span>the <span class="_ _1a"> </span>similarity <span class="_ _16"> </span>ratio <span class="_ _16"> </span>for <span class="_ _1a"> </span>a </span></div><div class="t m0 x22 hf y1a3 ff2 fsb fc3 sc0 ls0 ws0">particular <span class="_ _16"> </span>gene, <span class="_ _16"> </span>it <span class="_ _16"> </span>is <span class="_ _1a"> </span>passed <span class="_ _16"> </span>on <span class="_ _1a"> </span>to <span class="_ _16"> </span>the <span class="_ _1a"> </span>Smith-Waterman </div><div class="t m0 x22 hf y1a4 ff2 fsb fc3 sc0 ls0 ws0">algorithm to detect the mutations.</div><div class="t m0 x21 hf y1a5 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Similarity <span class="_ _1a"> </span>percentage <span class="_ _19"> </span>threshold <span class="_ _14"> </span>for <span class="_ _19"> </span>a <span class="_ _19"> </span>particular <span class="_ _19"> </span>gene <span class="_ _19"> </span>is </span></div><div class="t m0 x22 hf y1a6 ff2 fsb fc3 sc0 ls0 ws0">calculated <span class="_ _5"> </span>by <span class="_ _e"> </span>considering <span class="_ _1b"> </span>the  maximum  rate <span class="_ _e"> </span>of  mutations </div><div class="t m0 x22 hf y1a7 ff2 fsb fc3 sc0 ls0 ws0">allowed <span class="_ _9"></span>per <span class="_ _8"></span>gene, <span class="_ _b"></span>and <span class="_ _8"></span>also <span class="_ _b"></span>the <span class="_ _8"></span>patient’s <span class="_ _b"></span>age <span class="_ _8"></span>(according <span class="_ _a"></span>to <span class="_ _8"></span>lit. </div><div class="t m0 x22 hf y1a8 ff2 fsb fc3 sc0 ls0 ws0">review)</div><div class="t m0 x21 hf y1a9 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">The <span class="_ _a"></span>defragmentation is <span class="_ _b"></span>algorithm is <span class="_ _b"></span>then <span class="_ _a"></span>used <span class="_ _a"></span>to <span class="_ _a"></span>defragment </span></div><div class="t m0 x22 hf y1aa ff2 fsb fc3 sc0 ls0 ws0">the sequence, if any.</div><div class="t m0 x21 hf y1ab ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">When <span class="_ _b"></span>the <span class="_ _8"></span>similarity <span class="_ _a"></span>is <span class="_ _8"></span>within <span class="_ _8"></span>the <span class="_ _b"></span>threshold, the <span class="_ _8"></span>sequences <span class="_ _8"></span>are </span></div><div class="t m0 x22 hf y1ac ff2 fsb fc3 sc0 ls0 ws0">aligned <span class="_ _a"></span>using <span class="_ _8"></span>the <span class="_ _b"></span>local <span class="_ _a"></span>alignment <span class="_ _8"></span>Smith-Waterman <span class="_ _0"></span>algorithm, </div><div class="t m0 x22 hf y1ad ff2 fsb fc3 sc0 ls0 ws0">and <span class="_ _c"></span>the <span class="_ _13"></span>mutations <span class="_ _c"></span>are <span class="_ _c"></span>identified. <span class="_ _a"></span>These <span class="_ _c"></span>are <span class="_ _8"></span>passed <span class="_ _c"></span>onto <span class="_ _13"> </span>the </div><div class="t m0 x22 hf y1ae ff2 fsb fc3 sc0 ls0 ws0">module 3 as inputs.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf28" class="pf w0 h0" data-page-no="28"><div class="pc pc28 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg28.png"/><div class="t m0 xa h5 y1af ff1 fs3 fc1 sc0 ls0 ws0">Deriving the Maximum Fragmentation </div><div class="t m0 xa h5 y1b0 ff1 fs3 fc1 sc0 ls0 ws0">Threshold</div><div class="t m0 x20 hf y1b1 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">According <span class="_ _12"></span>to <span class="_ _a"></span>the literature <span class="_ _30"></span>review <span class="_ _30"></span>conducted, the maximum number of mutations <span class="_ _a"></span>that can be <span class="_ _a"></span>present <span class="_ _9"></span>in a healthy and </span></div><div class="t m0 x60 hf y1b2 ff2 fsb fc3 sc0 ls0 ws0">young human gene is 2.8 x 10</div><div class="t m0 x48 h1f y1b3 ff2 fs12 fc3 sc0 ls0 ws0">-7 </div><div class="t m0 x47 hf y1b2 ff2 fsb fc3 sc0 ls0 ws0"> per base pair (Nachman et al, 2000)</div><div class="t m0 x20 hf y1b4 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">To <span class="_ _c"></span>ease <span class="_ _5"> </span>the  calculation <span class="_ _4"> </span>during <span class="_ _e"> </span>the <span class="_ _e"> </span>implementation, <span class="_ _5"> </span>the <span class="_ _e"> </span>base  pair <span class="_ _e"> </span>calculation <span class="_ _4"> </span>has  been <span class="_ _5"> </span>converted <span class="_ _7"> </span>to  nucleotide </span></div><div class="t m0 x60 hf y1b5 ff2 fsb fc3 sc0 ls0 ws0">calculation</div><div class="t m0 x20 hf y1b6 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">The <span class="_ _b"></span>mutation <span class="_ _c"></span>similarity <span class="_ _b"></span>threshold obtained <span class="_ _8"></span>is <span class="_ _8"></span>.then <span class="_ _8"></span>further <span class="_ _8"></span>improved by <span class="_ _a"></span>taking <span class="_ _c"></span>into <span class="_ _8"></span>account <span class="_ _8"></span>the <span class="_ _8"></span>patient’s <span class="_ _8"></span>age, <span class="_ _c"></span>as <span class="_ _8"></span>the </span></div><div class="t m0 x60 hf y1b7 ff2 fsb fc3 sc0 ls0 ws0">age has a prominent effect on the mutation rate of a human gene. (Milholland et al, 2015)</div><div class="t m0 x20 hf y1b8 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Using these as parameters, fuzzy logic concepts has been used to derive the mutation fragmentation threshold.</span></div><div class="t m0 x20 hf y1b9 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">There <span class="_ _9"></span>were <span class="_ _9"></span>3 <span class="_ _a"></span>libraries evaluated <span class="_ _9"></span>for the <span class="_ _a"></span>calculation of the <span class="_ _a"></span>mutation <span class="_ _a"></span>threshold <span class="_ _9"></span>and <span class="_ _a"></span>an <span class="_ _a"></span>ensemble of those <span class="_ _a"></span>libraries have </span></div><div class="t m0 x60 hf y1ba ff2 fsb fc3 sc0 ls0 ws0">been used in this module, which is be discussed in evaluation section further</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf29" class="pf w0 h0" data-page-no="29"><div class="pc pc29 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg29.png"/><div class="t m0 xa h5 y1af ff1 fs3 fc1 sc0 ls0 ws0">Defragmentation Algorithm</div><div class="t m0 x61 hf y1bb ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">This newly developed algorithm can be described using the following pseudo code</span></div><div class="t m0 x61 hf y1bc ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">The mutation probabilities are derived from the following table</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2a" class="pf w0 h0" data-page-no="2a"><div class="pc pc2a w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2a.png"/><div class="t m0 xa h5 y1af ff1 fs3 fc1 sc0 ls0 ws0">Smith Waterman Algorithm to detect </div><div class="t m0 xa h5 y1b0 ff1 fs3 fc1 sc0 ls0 ws0">Mutations</div><div class="t m0 x61 hf y1bd ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">After <span class="_ _c"></span>considering <span class="_ _6"> </span>many <span class="_ _13"> </span>mutation <span class="_ _13"> </span>detection <span class="_ _13"></span>algorithms <span class="_ _c"></span>available <span class="_ _a"></span>in <span class="_ _13"></span>the <span class="_ _13"></span>bioinformatics <span class="_ _13"></span>domain, <span class="_ _13"> </span>the <span class="_ _13"></span>Smith <span class="_ _13"></span>Waterman </span></div><div class="t m0 x62 hf y1be ff2 fsb fc3 sc0 ls0 ws0">algorithm <span class="_ _7"> </span>was <span class="_ _7"> </span>chosen <span class="_ _4"> </span>to <span class="_ _7"> </span>proceed <span class="_ _13"></span>with <span class="_ _7"> </span>as <span class="_ _4"> </span>in <span class="_ _7"> </span>this <span class="_ _4"> </span>research <span class="_ _c"></span>we <span class="_ _6"> </span>focus <span class="_ _7"> </span>on <span class="_ _4"> </span>genes, <span class="_ _4"> </span>which <span class="_ _7"> </span>are <span class="_ _6"> </span>available <span class="_ _8"></span>as <span class="_ _4"> </span>parts <span class="_ _4"> </span>in <span class="_ _4"> </span>the </div><div class="t m0 x62 hf y1bf ff2 fsb fc3 sc0 ls0 ws0">human DNA.</div><div class="t m0 x61 hf y1c0 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Possible <span class="_ _6"> </span>mutations <span class="_ _5"> </span>that <span class="_ _4"> </span>can <span class="_ _4"> </span>be <span class="_ _4"> </span>identified <span class="_ _6"> </span>include, <span class="_ _4"> </span>insertions, <span class="_ _4"> </span>deletions, <span class="_ _7"> </span>substitutions. <span class="_ _5"> </span>However, <span class="_ _9"></span>in <span class="_ _4"> </span>this <span class="_ _4"> </span>research <span class="_ _13"> </span>we </span></div><div class="t m0 x62 hf y1c1 ff2 fsb fc3 sc0 ls0 ws0">focus on substitution mutations to reduce the research complexity</div><div class="t m0 x61 hf y1c2 ff3 fsb fc3 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">The algorithm in consisted of three steps including the initialization, matrix fill and the traceback step.</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2b" class="pf w0 h0" data-page-no="2b"><div class="pc pc2b w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2b.png"/><div class="t m0 x1f h5 y19c ff1 fs3 fc1 sc0 ls0 ws0">Detection of Melanoma based on Protein </div><div class="t m0 x63 h5 y19d ff1 fs3 fc1 sc0 ls0 ws0">Primary Structure </div><div class="t m0 x4d h1 y19e ff1 fs0 fc1 sc0 ls0 ws0">Module 3 - Implementation</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2c" class="pf w0 h0" data-page-no="2c"><div class="pc pc2c w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2c.png"/><div class="t m0 x64 h1e y1c3 ff1 fs11 fc1 sc0 ls0 ws0">High-Level Architecture Diagram of Proposed Solution </div><div class="t m0 x5d ha y1c4 ff4 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">As <span class="_ _a"></span>a <span class="_ _b"></span>first <span class="_ _9"></span>step, <span class="_ _b"></span>the <span class="_ _b"></span>mutated <span class="_ _a"></span>DNA <span class="_ _a"></span>sequence </span></div><div class="t m0 x4e ha y1c5 ff2 fs9 fc3 sc0 ls0 ws0">will <span class="_ _1b"> </span>be <span class="_ _35"> </span>passed <span class="_ _1b"> </span>through <span class="_ _e"> </span>the <span class="_ _1b"> </span>transcription </div><div class="t m0 x4e ha y1c6 ff2 fs9 fc3 sc0 ls0 ws0">and <span class="_ _19"> </span>translation <span class="_ _16"> </span>process <span class="_ _1a"> </span>to <span class="_ _2f"> </span>retrieve <span class="_ _15"> </span>the </div><div class="t m0 x4e ha y1c7 ff2 fs9 fc3 sc0 ls0 ws0">primary <span class="_ _35"> </span>protein <span class="_ _1b"> </span>structure <span class="_ _f"> </span>or  amino <span class="_ _1b"> </span>acid </div><div class="t m0 x4e ha y1c8 ff2 fs9 fc3 sc0 ls0 ws0">sequence. </div><div class="t m0 x5d ha y1c9 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Designing <span class="_ _19"> </span>and <span class="_ _19"> </span>developing <span class="_ _15"> </span>efficient <span class="_ _19"> </span>and </span></div><div class="t m0 x4e ha y1ca ff2 fs9 fc3 sc0 ls0 ws0">effective <span class="_ _2c"> </span>methods <span class="_ _36"> </span>to <span class="_ _3f"> </span>represent <span class="_ _21"> </span>protein </div><div class="t m0 x4e ha y1cb ff2 fs9 fc3 sc0 ls0 ws0">primary <span class="_ _46"> </span>structure <span class="_ _47"> </span>as <span class="_ _46"> </span>numerical </div><div class="t m0 x4e ha y1cc ff2 fs9 fc3 sc0 ls0 ws0">representation <span class="_ _47"> </span>or <span class="_ _46"> </span>called <span class="_ _47"> </span>vector </div><div class="t m0 x4e ha y1cd ff2 fs9 fc3 sc0 ls0 ws0">representation. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2d" class="pf w0 h0" data-page-no="2d"><div class="pc pc2d w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2d.png"/><div class="t m0 x1 ha y1ce ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Performing <span class="_ _13"></span>pre-processing, <span class="_ _c"></span>feature <span class="_ _c"></span>engineering <span class="_ _7"> </span>to <span class="_ _4"> </span>select <span class="_ _13"></span>features <span class="_ _c"></span>both <span class="_ _4"> </span>categorical <span class="_ _7"> </span>and </span></div><div class="t m0 x65 ha y1cf ff2 fs9 fc3 sc0 ls0 ws0">numerical <span class="_ _b"></span>which <span class="_ _b"></span>supports <span class="_ _8"></span>for <span class="_ _a"></span>melanoma prediction as <span class="_ _b"></span>helper <span class="_ _b"></span>features <span class="_ _9"></span>along <span class="_ _a"></span>with <span class="_ _b"></span>protein </div><div class="t m0 x65 ha y1d0 ff2 fs9 fc3 sc0 ls0 ws0">primary structure.</div><div class="t m0 x1 ha y1d1 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Applying relevant label encoding for categorical features.</span></div><div class="t m0 x1 ha y1d2 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Resolving class imbalance.</span></div><div class="t m0 x1 ha y1d3 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Selecting <span class="_ _13"></span>machine <span class="_ _4"> </span>learning <span class="_ _13"></span>or <span class="_ _4"> </span>neural <span class="_ _13"></span>network <span class="_ _7"> </span>algorithms <span class="_ _7"> </span>to <span class="_ _4"> </span>train <span class="_ _6"></span>generalized <span class="_ _c"></span>model <span class="_ _4"> </span>to </span></div><div class="t m0 x65 ha y1d4 ff2 fs9 fc3 sc0 ls0 ws0">classify melanoma using cross-validation. </div><div class="t m0 x1 ha y1d5 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Optimizing <span class="_ _8"></span>models <span class="_ _c"></span>with <span class="_ _8"></span>randomized grid <span class="_ _8"></span>search to <span class="_ _c"></span>fine-tune hyperparameters <span class="_ _b"></span>for <span class="_ _b"></span>better </span></div><div class="t m0 x65 ha y1d6 ff2 fs9 fc3 sc0 ls0 ws0">performance in each individual models. </div><div class="t m0 x1 ha y1d7 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Selecting best classifier by evaluating all designed and developed models based on </span></div><div class="t m0 x65 ha y1d8 ff2 fs9 fc3 sc0 ls0 ws0">precision, recall, f1-score as evaluation criteria. </div><div class="t m0 x28 h1e y1d9 ff1 fs11 fc1 sc0 ls0 ws0">High-Level Architecture Diagram of Proposed Solution (cont.)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2e" class="pf w0 h0" data-page-no="2e"><div class="pc pc2e w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2e.png"/><div class="t m0 x13 h1e y1da ff1 fs11 fc1 sc0 ls0 ws0">Transcription and Translation<span class="_ _48"></span><span class="fc3"> Process<span class="_ _49"> </span> <span class="_ _4a"> </span> <span class="_ _4a"> </span> </span></div><div class="t m0 x41 hc y1db ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">DNA <span class="_ _a"></span>is made up <span class="_ _a"></span>of <span class="_ _a"></span>a linear string of <span class="_ _a"></span>nucleotides, </span></div><div class="t m0 x42 hc y1dc ff2 fsa fc3 sc0 ls0 ws0">or bases, which <span class="_ _a"></span>are <span class="_ _9"></span>abbreviated <span class="_ _30"></span>as A, T, <span class="_ _30"></span>C, <span class="_ _a"></span>and G </div><div class="t m0 x42 hc y1dd ff2 fsa fc3 sc0 ls0 ws0">in their chemical identities. </div><div class="t m0 x41 hc y1de ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">It <span class="_ _1"> </span>includes <span class="_ _33"> </span>making <span class="_ _1"> </span>an <span class="_ _33"> </span>RNA <span class="_ _33"> </span>molecule <span class="_ _39"> </span>by </span></div><div class="t m0 x42 hc y1df ff2 fsa fc3 sc0 ls0 ws0">duplicating <span class="_ _5"> </span>the <span class="_ _5"> </span>DNA <span class="_ _e"> </span>sequence <span class="_ _5"> </span>of <span class="_ _5"> </span>a <span class="_ _e"> </span>gene. <span class="_ _5"> </span>RNA </div><div class="t m0 x42 hc y1e0 ff2 fsa fc3 sc0 ls0 ws0">polymerases <span class="_ _a"></span>are <span class="_ _b"></span>enzymes <span class="_ _c"></span>that <span class="_ _8"></span>bind <span class="_ _c"></span>nucleotides </div><div class="t m0 x42 hc y1e1 ff2 fsa fc3 sc0 ls0 ws0">together  to <span class="_ _25"> </span>generate <span class="_ _1b"> </span>an  RNA <span class="_ _25"> </span>strand  (using  a </div><div class="t m0 x42 hc y1e2 ff2 fsa fc3 sc0 ls0 ws0">DNA <span class="_ _1a"> </span>strand <span class="_ _14"> </span>as <span class="_ _16"> </span>a <span class="_ _1a"> </span>template). <span class="_ _14"> </span>After <span class="_ _1a"> </span>generating </div><div class="t m0 x42 hc y1e3 ff2 fsa fc3 sc0 ls0 ws0">RNA, <span class="_ _5"> </span>the <span class="_ _4"> </span>amino <span class="_ _4"> </span>acid <span class="_ _5"> </span>sequence <span class="_ _4"> </span>can <span class="_ _5"> </span>be <span class="_ _4"> </span>derived </div><div class="t m0 x42 hc y1e4 ff2 fsa fc3 sc0 ls0 ws0">from the RNA.</div><div class="t m0 x41 hc y1e5 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">In <span class="_ _4"> </span>the <span class="_ _7"> </span>translation <span class="_ _8"></span>process, <span class="_ _6"> </span>three-letter <span class="_ _8"></span>codes <span class="_ _7"> </span>in </span></div><div class="t m0 x42 hc y1e6 ff2 fsa fc3 sc0 ls0 ws0">DNA/RNA <span class="_ _c"></span>are <span class="_ _b"></span>decoded <span class="_ _a"></span>as <span class="_ _c"></span>one <span class="_ _c"></span>amino <span class="_ _8"></span>acid. <span class="_ _c"></span>That </div><div class="t m0 x42 hc y1e7 ff2 fsa fc3 sc0 ls0 ws0">is called a codon. </div><div class="t m0 x41 hc y1e8 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">AUG <span class="_ _c"></span>is <span class="_ _6"></span>the <span class="_ _13"></span>start <span class="_ _13"></span>codon <span class="_ _6"></span>and <span class="_ _13"></span>UAA, <span class="_ _c"></span>UAG, <span class="_ _8"></span>UGA <span class="_ _6"> </span>are </span></div><div class="t m0 x42 hc y1e9 ff2 fsa fc3 sc0 ls0 ws0">the stop codons. </div><div class="t m0 x66 hc y1ea ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">So, <span class="_ _16"> </span>the <span class="_ _16"> </span>translation <span class="_ _25"> </span>process <span class="_ _15"> </span>starts <span class="_ _16"> </span>with <span class="_ _16"> </span>the <span class="_ _16"> </span>start </span></div><div class="t m0 x67 hc y1eb ff2 fsa fc3 sc0 ls0 ws0">codon and ends by meeting any stop codon. </div><div class="t m0 x66 hc y1ec ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">About <span class="_ _5"> </span>500 <span class="_ _5"> </span>amino <span class="_ _5"> </span>acids <span class="_ _5"> </span>are <span class="_ _4"> </span>known <span class="_ _4"> </span>at <span class="_ _5"> </span>present, <span class="_ _7"> </span>but </span></div><div class="t m0 x67 hc y1ed ff2 fsa fc3 sc0 ls0 ws0">only 20 appear in our genetic code. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2f" class="pf w0 h0" data-page-no="2f"><div class="pc pc2f w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2f.png"/><div class="t m0 xc h1e y1ee ff1 fs11 fc1 sc0 ls0 ws0">Selected Genes and Mutation Types </div><div class="t m0 x1 hc y1ef ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">Substitution - Missense </span></div><div class="t m0 x1 hc y1f0 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">Substitution - Coding Silent </span></div><div class="t m0 x1 hc y1f1 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">Substitution - Nonsense</span></div><div class="t m0 x1 hc y1f2 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">Deletion - In Frame </span></div><div class="t m0 x1 hc y1f3 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">Deletion - Frameshift </span></div><div class="t m0 x1 hc y1f4 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">Insertion - In Frame </span></div><div class="t m0 x1 hc y1f5 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">Insertion - Frameshift </span></div><div class="t m0 x24 h20 y1f6 ff1 fs5 fc2 sc0 ls0 ws0">Mutation Types</div><div class="t m0 x3a h20 y1f7 ff1 fs5 fc2 sc0 ls0 ws0">Selected Genes</div><div class="t m0 x68 hc y1f8 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">BRAF</span></div><div class="t m0 x68 hc y1f9 ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">MUC16</span></div><div class="t m0 x68 hc y1fa ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">NRAS</span></div><div class="t m0 x68 hc y1fb ff5 fsa fc3 sc0 ls0 ws0">❖<span class="_ _22"> </span><span class="ff2">LRP1B</span></div><div class="t m0 x61 hc y1fc ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Substitution <span class="_ _8"></span>is <span class="_ _8"></span>the <span class="_ _8"></span>most <span class="_ _b"></span>frequent <span class="_ _a"></span>mutation <span class="_ _8"></span>type <span class="_ _b"></span>and <span class="_ _8"></span> <span class="_ _8"></span>insertion <span class="_ _8"></span>and <span class="_ _8"></span>deletion <span class="_ _a"></span>mutations <span class="_ _8"></span>are <span class="_ _a"></span>extremely </span></div><div class="t m0 x62 hc y1fd ff2 fsa fc3 sc0 ls0 ws0">rare.</div><div class="t m0 x61 hc y1fe ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Coding <span class="_ _13"></span>Silent <span class="_ _13"></span>has <span class="_ _6"> </span>no <span class="_ _7"> </span>effect <span class="_ _13"></span>on <span class="_ _6"> </span>the <span class="_ _7"> </span>primary <span class="_ _6"> </span>structure <span class="_ _13"></span>of <span class="_ _7"> </span>proteins <span class="_ _13"></span>because <span class="_ _13"></span>the <span class="_ _7"> </span>mutation <span class="_ _6"> </span>in <span class="_ _7"> </span>the <span class="_ _7"> </span>DNA </span></div><div class="t m0 x62 hc y1ff ff2 fsa fc3 sc0 ls0 ws0">codon results in the same real amino-acid.</div><div class="t m0 x61 hc y200 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Substitution - Missense and Substitution - Nonsense were selected mutation types for the analysis. </span></div><div class="t m0 x61 hc y201 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Genes were selected for the analysis based on the frequency of distribution on COSMIC dataset. </span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf30" class="pf w0 h0" data-page-no="30"><div class="pc pc30 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg30.png"/><div class="t m0 x1f h1e y1ee ff1 fs11 fc1 sc0 ls0 ws0">Vectorizing the Amino Acid Sequence (Protein Primary Structure)</div><div class="t m0 x69 h21 y202 ff2 fs5 fc3 sc0 ls0 ws0">In this module, primarily two types of vectorization approaches for protein </div><div class="t m0 x69 h21 y203 ff2 fs5 fc3 sc0 ls0 ws0">primary structure representation are designed and developed, as follows:</div><div class="t m0 x11 h20 y204 ff3 fs5 fc3 sc0 ls0 ws0">●<span class="_ _2e"> </span><span class="ff1">Vectorizing Method 01<span class="_ _4b"></span><span class="ff2"> - Individual Amino-acid Count Distribution</span></span></div><div class="t m2 x6a h22 y205 ff7 fs13 fc3 sc0 ls0 ws0">Vectorizing without preserving amino-acids relations</div><div class="t m0 x6b h20 y206 ff3 fs8 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff1 fs5">Vectorizing Method 02<span class="_ _4c"></span><span class="ff2"> - Pairwise Amino-acid Count Distribution<span class="_ _a"></span><span class="fs8"> </span></span></span></div><div class="t m2 x6a h22 y207 ff7 fs13 fc3 sc0 ls0 ws0">Vectorizing with preserving amino-acids relations between each amino-acid </div><div class="t m2 x6a h22 y208 ff7 fs13 fc3 sc0 ls0 ws0">pair</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf31" class="pf w0 h0" data-page-no="31"><div class="pc pc31 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg31.png"/><div class="t m0 x1b h1e y209 ff1 fs11 fc1 sc0 ls0 ws0">Vectorizing Method 01 - Individual Amino-acid Count Distribution</div><div class="t m0 x12 he y20a ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff1">Original Amino-Acid Distribution Count of BRAF</span></div><div class="t m0 x12 he y20b ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff1">Mutated Amino-Acid Distribution Count of BRAF</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf32" class="pf w0 h0" data-page-no="32"><div class="pc pc32 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg32.png"/><div class="t m0 x1b h1e y209 ff1 fs11 fc1 sc0 ls0 ws0">Vectorizing Method 01 - <span class="_ _4d"></span><span class="ff2">Individual Amino-acid Count Distribution </span></div><div class="t m0 x1b h23 y20c ff2 fs11 fc1 sc0 ls0 ws0">(cont.) </div><div class="t m0 x15 he y20d ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff1">Difference (Changes) of Amino-Acid Distribution Count of BRAF</span></div><div class="t m0 x1 ha y20e ff2 fs9 fc3 sc0 ls0 ws0">In this method, </div><div class="t m0 x6c ha y20f ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">There are a total of twenty (20) features accountable to all classification models from </span></div><div class="t m0 x6d ha y210 ff2 fs9 fc3 sc0 ls0 ws0">protein primary structure. </div><div class="t m0 x6c ha y211 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Because this method simply considers changes at the protein primary structure level, it </span></div><div class="t m0 x6d ha y212 ff2 fs9 fc3 sc0 ls0 ws0">is simple to learn for all classification models.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf33" class="pf w0 h0" data-page-no="33"><div class="pc pc33 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg33.png"/><div class="t m0 x1b h1e y209 ff1 fs11 fc1 sc0 ls0 ws0">Vectorizing Method 02 - Pairwise Amino-acid Count Distribution</div><div class="t m0 x6e h9 y213 ff3 fs8 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff1">Example Vector Representation </span></div><div class="t m0 x69 hc y214 ff2 fsa fc3 sc0 ls0 ws0">In this method, </div><div class="t m0 x6f hc y215 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Amino-acid sequence will be read from left to right order. </span></div><div class="t m0 x6f hc y216 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Final vector has a size of 400, resulting in 20 x 20 amino-acids. </span></div><div class="t m0 x6f hc y217 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">From the altered DNA, calculates the total number of pairs in the protein primary structure.</span></div><div class="t m0 x6f hc y218 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Do not takes only changes as previous method (Vectorization Method 01) . Because, there are </span></div><div class="t m0 x70 hc y219 ff2 fsa fc3 sc0 ls0 ws0">total four hundred (400) features and it will tend to contains too many zeros in the input buffer. </div><div class="t m0 x70 hc y21a ff2 fsa fc3 sc0 ls0 ws0">This will have an impact on the inability to learn for classification models. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf34" class="pf w0 h0" data-page-no="34"><div class="pc pc34 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg34.png"/><div class="t m0 x41 h1e y21b ff1 fs11 fc1 sc0 ls0 ws0">Reason for Using Count Distribution Vectorizing Technique</div><div class="t m0 x24 h21 y21c ff2 fs5 fc3 sc0 ls0 ws0">Biochemists have discovered that each form of protein includes the same </div><div class="t m0 x24 h21 y21d ff2 fs5 fc3 sc0 ls0 ws0">amount of total amino acids (often referred to as residues) in the same </div><div class="t m0 x24 h21 y21e ff2 fs5 fc3 sc0 ls0 ws0">proportion.</div><div class="t m0 x24 h21 y21f ff2 fs5 fc3 sc0 ls0 ws0">As an example,</div><div class="t m0 x71 h20 y220 ff1 fs5 fc3 sc0 ls0 ws0">Insulin = 30 Glycines + 44 Alanies + 5 Tyrosines + 14 Glutamines + ....</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf35" class="pf w0 h0" data-page-no="35"><div class="pc pc35 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg35.png"/><div class="t m0 x41 h1e y221 ff1 fs11 fc1 sc0 ls0 ws0">Feature Selection </div><div class="t m0 x17 h10 y222 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The <span class="_ _4e"> </span>entropy-based <span class="_ _19"> </span>calculation <span class="_ _21"> </span>technique <span class="_ _2c"> </span>was <span class="_ _3f"> </span>used </span></div><div class="t m0 x22 h10 y223 ff2 fsc fc3 sc0 ls0 ws0">which <span class="_ _5"> </span>is <span class="_ _e"> </span>Extra <span class="_ _7"> </span>Decision <span class="_ _4"> </span>Tree <span class="_ _b"></span>Classifier <span class="_ _6"> </span>to <span class="_ _e"> </span>identify <span class="_ _5"> </span>most </div><div class="t m0 x22 h10 y224 ff2 fsc fc3 sc0 ls0 ws0">prominent features for melanoma detection. </div><div class="t m0 x17 h10 y225 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Information <span class="_ _7"> </span>gain <span class="_ _7"> </span>was <span class="_ _4"> </span>the <span class="_ _7"> </span>measurement <span class="_ _13"></span>that <span class="_ _7"> </span>was <span class="_ _7"> </span>taken </span></div><div class="t m0 x22 h10 y226 ff2 fsc fc3 sc0 ls0 ws0">for <span class="_ _b"></span>feature importance <span class="_ _a"></span>when <span class="_ _8"></span>building <span class="_ _a"></span>the <span class="_ _8"></span>Extra Decision </div><div class="t m0 x22 h10 y227 ff2 fsc fc3 sc0 ls0 ws0">Tree Classifier.</div><div class="t m0 x2f h10 y228 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Most influential features for melanoma </span></div><div class="t m0 x39 h10 y229 ff2 fsc fc3 sc0 ls0 ws0">detection </div><div class="t m0 x3a h10 y22a ff3 fsc fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Tumor Origin </span></div><div class="t m0 x3a h10 y22b ff3 fsc fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Age</span></div><div class="t m0 x2f h10 y22c ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Due to the low significance of the feature, </span></div><div class="t m0 x39 h10 y22d ff2 fsc fc3 sc0 ls0 ws0">Mutation Type was rejected.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf36" class="pf w0 h0" data-page-no="36"><div class="pc pc36 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg36.png"/><div class="t m0 x41 h1e y22e ff1 fs11 fc1 sc0 ls0 ws0">Handling Class Imbalance</div><div class="t m0 x17 h10 y22f ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Total <span class="_ _12"></span>one <span class="_ _c"></span>hundred <span class="_ _a"></span>and <span class="_ _c"></span>twenty <span class="_ _b"></span>four <span class="_ _c"></span>(124) <span class="_ _8"></span>cancer <span class="_ _c"></span>and <span class="_ _c"></span>non-</span></div><div class="t m0 x22 h10 y230 ff2 fsc fc3 sc0 ls0 ws0">cancer variations contains the target label. </div><div class="t m0 x17 h10 y231 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Labeling the target label  </span></div><div class="t m0 x72 h10 y232 ff3 fsc fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Yes (1) - Has Melanoma </span></div><div class="t m0 x72 h10 y233 ff3 fsc fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">No (0) - Not Melanoma</span></div><div class="t m0 x19 h10 y234 ff3 fsc fc3 sc0 ls0 ws0">■<span class="_ _d"> </span><span class="ff2">This <span class="_ _6"> </span>label <span class="_ _c"></span>contains <span class="_ _13"></span>123 <span class="_ _6"> </span>cancer <span class="_ _13"></span>and <span class="_ _13"> </span>non-cancer </span></div><div class="t m0 x73 h10 y235 ff2 fsc fc3 sc0 ls0 ws0">variations </div><div class="t m0 x17 h10 y236 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Melanoma <span class="_ _a"></span>cancer <span class="_ _a"></span>accounted for <span class="_ _a"></span>over 23% <span class="_ _b"></span>of <span class="_ _b"></span>data <span class="_ _b"></span>points, </span></div><div class="t m0 x22 h10 y237 ff2 fsc fc3 sc0 ls0 ws0">whereas other 123 cancer types accounted for 77%.</div><div class="t m0 x17 h10 y238 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">As <span class="_ _a"></span>a <span class="_ _a"></span>result, the dataset is <span class="_ _a"></span>unbalanced, <span class="_ _12"></span>which <span class="_ _a"></span>may <span class="_ _a"></span>result <span class="_ _9"></span>in </span></div><div class="t m0 x22 h10 y239 ff2 fsc fc3 sc0 ls0 ws0">the <span class="_ _b"></span>model <span class="_ _8"></span>being <span class="_ _8"></span>under-fitted. <span class="_ _9"></span>This <span class="_ _8"></span>problem can <span class="_ _b"></span>be <span class="_ _8"></span>solved </div><div class="t m0 x22 h10 y23a ff2 fsc fc3 sc0 ls0 ws0">by <span class="_ _4"> </span>employing <span class="_ _6"> </span>oversampling <span class="_ _7"> </span>techniques, <span class="_ _7"> </span>and <span class="_ _5"> </span>here <span class="_ _4"> </span>it <span class="_ _5"> </span>was </div><div class="t m0 x22 h10 y23b ff2 fsc fc3 sc0 ls0 ws0">addressed by employing the following two strategies:</div><div class="t m0 x72 h10 y23c ff3 fsc fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Synthetic <span class="_ _3d"> </span>Minority <span class="_ _3c"> </span>Over-sampling <span class="_ _3c"> </span>Technique </span></div><div class="t m0 x64 h10 y23d ff2 fsc fc3 sc0 ls0 ws0">(SMOTE)</div><div class="t m0 x72 h10 y23e ff3 fsc fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Adaptive <span class="_ _4f"> </span>Synthetic <span class="_ _50"> </span>Over-sampling <span class="_ _4f"> </span>Technique </span></div><div class="t m0 x64 h10 y23f ff2 fsc fc3 sc0 ls0 ws0">(ASOTE)</div><div class="t m0 x74 h10 y240 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">ASOTE <span class="_ _17"> </span>was <span class="_ _14"> </span>used, <span class="_ _15"> </span>which <span class="_ _14"> </span>is <span class="_ _14"> </span>a <span class="_ _14"> </span>statistical </span></div><div class="t m0 x75 h10 y241 ff2 fsc fc3 sc0 ls0 ws0">technique <span class="_ _e"> </span>for  oversampling <span class="_ _5"> </span>that  creates </div><div class="t m0 x75 h10 y242 ff2 fsc fc3 sc0 ls0 ws0">data points rather than duplicating data.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf37" class="pf w0 h0" data-page-no="37"><div class="pc pc37 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg37.png"/><div class="t m0 x5d h1e y243 ff1 fs11 fc1 sc0 ls0 ws0">Classification</div><div class="t m0 x24 hc y244 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">In the classification process, there were four machine learning algorithms selected for melanoma </span></div><div class="t m0 x76 hc y245 ff2 fsa fc3 sc0 ls0 ws0">classification.</div><div class="t m0 x60 hc y246 ff3 fsa fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Logistic Regression</span></div><div class="t m0 x60 hc y247 ff3 fsa fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Random Forest</span></div><div class="t m0 x60 hc y248 ff3 fsa fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">K-Nearest Neighbors (KNN)</span></div><div class="t m0 x60 hc y249 ff3 fsa fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2">Gradient Boosting </span></div><div class="t m0 x24 hc y24a ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">35,000+ data points before handling class imbalance </span></div><div class="t m0 x24 hc y24b ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">50,000+ data points after handling class imbalance </span></div><div class="t m0 x24 hc y24c ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">When employing Vectorizing Method 01, a total of twenty-four features (24) were employed to feed the </span></div><div class="t m0 x76 hc y24d ff2 fsa fc3 sc0 ls0 ws0">models.</div><div class="t m0 x24 hc y24e ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">When employing Vectorizing Method 02, a total of four hundred four features (404)  were employed to </span></div><div class="t m0 x76 hc y24f ff2 fsa fc3 sc0 ls0 ws0">feed the models.</div><div class="t m0 x24 hc y250 ff3 fsa fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Randomized search cross- validation (RSCV) was used to discover the optimal hyper parameter for each </span></div><div class="t m0 x76 hc y251 ff2 fsa fc3 sc0 ls0 ws0">method at random. After that, grid-search cross-validation was used to find the best-generalized model </div><div class="t m0 x76 hc y252 ff2 fsa fc3 sc0 ls0 ws0">for each algorithm using hyper parameter combinations obtained in the RSCV.  </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf38" class="pf w0 h0" data-page-no="38"><div class="pc pc38 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg38.png"/><div class="t m0 x5d h1e y18b ff1 fs11 fc1 sc0 ls0 ws0">Rest Api Design Documentation</div><div class="t m0 xd h12 y253 ff1 fsc fc3 sc0 ls0 ws0">API Endpoint<span class="ff2">: <span class="fc4">https://melanoma-detection-tool-api.herokuapp.com/predict-melanoma/pps</span></span></div><div class="t m0 xd h12 y254 ff1 fsc fc3 sc0 ls0 ws0">Method<span class="ff2">: POST</span></div><div class="t m0 xd h12 y255 ff1 fsc fc3 sc0 ls0 ws0">Request body format<span class="ff2">: JSON</span></div><div class="t m0 xd h12 y256 ff1 fsc fc3 sc0 ls0 ws0">Required params in request body<span class="_ _10"></span><span class="ff2">: age, gene, tumor, tier, mutated dna</span></div><div class="t m0 xd h10 y257 ff2 fsc fc3 sc0 ls0 ws0">Example request                                                                                   Example Response </div><div class="t m0 xd h10 y258 ff2 fsc fc3 sc0 ls0 ws0">  <span class="_ _43"> </span><span class="fc2">{ </span></div><div class="t m0 xd h10 y259 ff2 fsc fc2 sc0 ls0 ws0">        <span class="_ _44"> </span>&quot;age&quot; : 45, </div><div class="t m0 xd h10 y25a ff2 fsc fc2 sc0 ls0 ws0">        <span class="_ _44"> </span>&quot;gene&quot; : &quot;BRAF&quot;,</div><div class="t m0 xd h10 y25b ff2 fsc fc2 sc0 ls0 ws0">        <span class="_ _44"> </span>&quot;tumor&quot; : &quot;metastasis&quot;,</div><div class="t m0 xd h10 y25c ff2 fsc fc2 sc0 ls0 ws0">        <span class="_ _44"> </span>&quot;tier&quot; : 1,</div><div class="t m0 xd h10 y25d ff2 fsc fc2 sc0 ls0 ws0">        <span class="_ _44"> </span>&quot;mutated dna&quot; : &quot;ATGGCGGCGCTG...&quot;</div><div class="t m0 xd h10 y25e ff2 fsc fc2 sc0 ls0 ws0">   <span class="_ _45"> </span>} </div><div class="t m0 xd h12 y25f ff1 fsc fc3 sc0 ls0 ws0">Response Status</div><div class="t m0 xe h10 y260 ff2 fsc fc3 sc0 ls0 ws0">200 - Request processed successfully</div><div class="t m0 xe h10 y261 ff2 fsc fc3 sc0 ls0 ws0">400 - Bad Request</div><div class="t m0 x77 h10 y262 ff2 fsc fc6 sc0 ls0 ws0">{</div><div class="t m0 x77 h10 y263 ff2 fsc fc6 sc0 ls0 ws0">    <span class="_ _51"> </span>&quot;age&quot;: 29,</div><div class="t m0 x77 h10 y264 ff2 fsc fc6 sc0 ls0 ws0">   <span class="_ _45"> </span> &quot;gene&quot;: &quot;BRAF&quot;,</div><div class="t m0 x77 h10 y265 ff2 fsc fc6 sc0 ls0 ws0">    <span class="_ _51"> </span>&quot;pps&quot;: [ 0, 0, 0, 0, 0, 0, 0, -1,</div><div class="t m0 x77 h10 y266 ff2 fsc fc6 sc0 ls0 ws0">     <span class="_ _52"> </span>                 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0],</div><div class="t m0 x77 h10 y267 ff2 fsc fc6 sc0 ls0 ws0">    <span class="_ _51"> </span>&quot;probability&quot;: 0.8882274916128,</div><div class="t m0 x77 h10 y268 ff2 fsc fc6 sc0 ls0 ws0">    <span class="_ _51"> </span>&quot;tier&quot;: 1,</div><div class="t m0 x77 h10 y269 ff2 fsc fc6 sc0 ls0 ws0">    <span class="_ _51"> </span>&quot;tumor&quot;: &quot;metastasis&quot;</div><div class="t m0 x77 h10 y26a ff2 fsc fc6 sc0 ls0 ws0">}</div><a class="l" href="https://melanoma-detection-tool-api.herokuapp.com/predict-melanoma/pps"><div class="d m1" style="border-style:none;position:absolute;left:165.215607px;bottom:422.813599px;width:462.731415px;height:15.845703px;background-color:rgba(255,255,255,0.000001);"></div></a></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf39" class="pf w0 h0" data-page-no="39"><div class="pc pc39 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg39.png"/><div class="t m0 xd h1e y18b ff1 fs11 fc1 sc0 ls0 ws0">Code Repos &amp; Application Website</div><div class="t m0 x1e ha y26b ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Melanoma Diagnosis Analysis</span></div><div class="t m0 x78 ha y26c ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2 fc4">https://github.com/amesh-jayaweera/melanoma-diagnosis-analysis</span></div><div class="t m0 x1e ha y26d ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Melanoma Detection API</span></div><div class="t m0 x78 ha y26e ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2 fc4">https://github.com/amesh-jayaweera/melanoma-detection-api</span></div><div class="t m0 x1e ha y26f ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Melanoma Detection Tool </span></div><div class="t m0 x78 ha y270 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2 fc4">https://github.com/amesh-jayaweera/melanoma-detection-tool</span></div><div class="t m0 x1e ha y271 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Melanoma Detection Website </span></div><div class="t m0 x78 ha y272 ff3 fs9 fc3 sc0 ls0 ws0">○<span class="_ _d"> </span><span class="ff2 fc4">https://melanoma-detection-tool.vercel.app/</span></div><a class="l" href="https://github.com/amesh-jayaweera/melanoma-diagnosis-analysis"><div class="d m1" style="border-style:none;position:absolute;left:134.482300px;bottom:348.046082px;width:472.882812px;height:18.109375px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="https://github.com/amesh-jayaweera/melanoma-detection-api"><div class="d m1" style="border-style:none;position:absolute;left:134.482300px;bottom:271.246063px;width:433.578125px;height:18.109375px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="https://github.com/amesh-jayaweera/melanoma-detection-tool"><div class="d m1" style="border-style:none;position:absolute;left:134.482300px;bottom:194.446045px;width:438.023438px;height:18.109375px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="https://melanoma-detection-tool.vercel.app/"><div class="d m1" style="border-style:none;position:absolute;left:134.482300px;bottom:117.646057px;width:307.203125px;height:18.109375px;background-color:rgba(255,255,255,0.000001);"></div></a></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3a" class="pf w0 h0" data-page-no="3a"><div class="pc pc3a w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3a.png"/><div class="t m0 x79 h5 y273 ff1 fs3 fc1 sc0 ls0 ws0">Detection of Melanoma using </div><div class="t m0 x6b h5 y274 ff1 fs3 fc1 sc0 ls0 ws0">Dermoscopic Feature Extraction and </div><div class="t m0 x7a h5 y275 ff1 fs3 fc1 sc0 ls0 ws0">Analysis </div><div class="t m0 x7b h1 y276 ff1 fs0 fc1 sc0 ls0 ws0">Module 1 - Evaluation</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3b" class="pf w0 h0" data-page-no="3b"><div class="pc pc3b w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3b.png"/><div class="t m0 x23 h5 y277 ff1 fs3 fc1 sc0 ls0 ws0">Results and Evaluation </div><div class="t m0 x4e h10 y278 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">A <span class="_ _4"> </span>balanced <span class="_ _c"></span>dataset <span class="_ _6"> </span>was <span class="_ _6"> </span>used <span class="_ _6"> </span>for <span class="_ _7"> </span>the <span class="_ _6"> </span>evaluation <span class="_ _13"></span>with <span class="_ _7"> </span>equal <span class="_ _7"> </span>amount <span class="_ _6"> </span>of <span class="_ _4"> </span>Melanoma <span class="_ _13"></span>and <span class="_ _7"> </span>Non-melanoma </span></div><div class="t m0 x7c h10 y279 ff2 fsc fc3 sc0 ls0 ws0">skin <span class="_ _8"></span>lesions. <span class="_ _a"></span>These <span class="_ _8"></span>Non-melanoma <span class="_ _b"></span>lesions <span class="_ _b"></span>included <span class="_ _a"></span>different skin <span class="_ _8"></span>diseases <span class="_ _b"></span>so <span class="_ _c"></span>that <span class="_ _b"></span>the <span class="_ _c"></span>model <span class="_ _8"></span>will <span class="_ _8"></span>be <span class="_ _8"></span>able </div><div class="t m0 x7c h10 y27a ff2 fsc fc3 sc0 ls0 ws0">to detect Melanoma even among such a scope of skin diseases. </div><div class="t m0 x7d h24 y27b ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Actinic Keratoses</span></div><div class="t m0 x7d h24 y27c ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Basal Cell Carcinoma</span></div><div class="t m0 x7d h24 y27d ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Benign Keratosis</span></div><div class="t m0 x4e h10 y27e ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">The <span class="_ _a"></span>best <span class="_ _a"></span>accuracy <span class="_ _12"></span>was obtained with the <span class="_ _a"></span>SVM, <span class="_ _9"></span>LR, <span class="_ _a"></span>ANN, <span class="_ _a"></span>Gradient <span class="_ _9"></span>Boosting, <span class="_ _a"></span>Adaboost classifiers <span class="_ _10"></span>and <span class="_ _a"></span>when </span></div><div class="t m0 x7c h10 y27f ff2 fsc fc3 sc0 ls0 ws0">using <span class="_ _5"> </span>the <span class="_ _5"> </span>voting-based <span class="_ _7"> </span>ensemble <span class="_ _4"> </span>method <span class="_ _4"> </span>in <span class="_ _5"> </span>which <span class="_ _5"> </span>Adaboost <span class="_ _4"> </span>and <span class="_ _5"> </span>Gradient <span class="_ _4"> </span>Boosting <span class="_ _5"> </span>had <span class="_ _5"> </span>the <span class="_ _5"> </span>highest </div><div class="t m0 x7c h10 y280 ff2 fsc fc3 sc0 ls0 ws0">accuracy values. </div><div class="t m0 x4e h10 y281 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">Precision, <span class="_ _a"></span>recall <span class="_ _a"></span>and <span class="_ _c"></span>the <span class="_ _c"></span>f-measure <span class="_ _b"></span>are <span class="_ _b"></span>the <span class="_ _c"></span>three <span class="_ _a"></span>other <span class="_ _c"></span>parameters <span class="_ _b"></span>that <span class="_ _c"></span>were <span class="_ _a"></span>used <span class="_ _8"></span>to <span class="_ _c"></span>determine <span class="_ _c"></span>whether </span></div><div class="t m0 x7c h10 y282 ff2 fsc fc3 sc0 ls0 ws0">these <span class="_ _c"></span>accuracy <span class="_ _b"></span>values <span class="_ _8"></span>are <span class="_ _8"></span>reliable and <span class="_ _c"></span>to <span class="_ _c"></span>know <span class="_ _8"></span>which <span class="_ _c"></span>classifier best <span class="_ _c"></span>suits <span class="_ _c"></span>out <span class="_ _c"></span>of <span class="_ _13"></span>the <span class="_ _c"></span>four <span class="_ _8"></span>classifiers that </div><div class="t m0 x7c h10 y283 ff2 fsc fc3 sc0 ls0 ws0">had the highest accuracy. </div><div class="t m0 x4e h10 y284 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">With this condition the Adaboost and Gradient <span class="_ _12"></span>Boosting classifiers <span class="_ _10"></span>were <span class="_ _9"></span>not selected <span class="_ _30"></span>as even <span class="_ _9"></span>though their </span></div><div class="t m0 x7c h10 y285 ff2 fsc fc3 sc0 ls0 ws0">accuracy <span class="_ _8"></span>values <span class="_ _c"></span>are <span class="_ _c"></span>high <span class="_ _13"></span>the <span class="_ _6"> </span>other <span class="_ _13"></span>parameters <span class="_ _c"></span>are <span class="_ _8"></span>low <span class="_ _8"></span>which <span class="_ _6"> </span>raises <span class="_ _8"></span>a <span class="_ _6"> </span>question <span class="_ _13"></span>on <span class="_ _13"> </span>the <span class="_ _6"> </span>reliability <span class="_ _8"></span>of <span class="_ _13"> </span>the </div><div class="t m0 x7c h10 y286 ff2 fsc fc3 sc0 ls0 ws0">accuracy results they obtained.</div><div class="t m0 x39 h24 y287 ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Melanocytic Nevi</span></div><div class="t m0 x39 h24 y288 ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Dermatofibroma</span></div><div class="t m0 x39 h24 y289 ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Vascular Skin Lesions</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3c" class="pf w0 h0" data-page-no="3c"><div class="pc pc3c w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3c.png"/><div class="t m0 x5a h5 y28a ff1 fs3 fc1 sc0 ls0 ws0">Results and Evaluation Contd.</div><div class="t m0 x7e h25 y28b ff1 fs2 fc7 sc0 ls0 ws0">Support Vector </div><div class="t m0 x7e h25 y28c ff1 fs2 fc7 sc0 ls0 ws0">Machine</div><div class="t m0 x7f ha y28d ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 90%</div><div class="t m0 x7f ha y28e ff2 fs9 fc3 sc0 ls0 ws0">Precision : 90%</div><div class="t m0 x7f ha y28f ff2 fs9 fc3 sc0 ls0 ws0">Recall : 91%</div><div class="t m0 x7f ha y290 ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 90% </div><div class="t m0 x80 ha y291 ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 91%</div><div class="t m0 x80 ha y292 ff2 fs9 fc3 sc0 ls0 ws0">Precision : 90%</div><div class="t m0 x80 ha y293 ff2 fs9 fc3 sc0 ls0 ws0">Recall : 92%</div><div class="t m0 x80 ha y294 ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 91% </div><div class="t m0 x81 h25 y295 ff1 fs2 fc7 sc0 ls0 ws0">Logistic </div><div class="t m0 x81 h25 y296 ff1 fs2 fc7 sc0 ls0 ws0">Regression</div><div class="t m0 x82 ha y297 ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 90%</div><div class="t m0 x82 ha y298 ff2 fs9 fc3 sc0 ls0 ws0">Precision : 93%</div><div class="t m0 x82 ha y299 ff2 fs9 fc3 sc0 ls0 ws0">Recall : 88%</div><div class="t m0 x82 ha y29a ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 90% </div><div class="t m0 x83 h25 y29b ff1 fs2 fc7 sc0 ls0 ws0">K-Nearest </div><div class="t m0 x83 h25 y29c ff1 fs2 fc7 sc0 ls0 ws0">Neighbour</div><div class="t m0 x4d h25 y29d ff1 fs2 fc7 sc0 ls0 ws0">Artificial Neural </div><div class="t m0 x4d h25 y29e ff1 fs2 fc7 sc0 ls0 ws0">Network</div><div class="t m0 x4d ha y29f ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 91%</div><div class="t m0 x4d ha y2a0 ff2 fs9 fc3 sc0 ls0 ws0">Precision : 90%</div><div class="t m0 x4d ha y2a1 ff2 fs9 fc3 sc0 ls0 ws0">Recall : 93%</div><div class="t m0 x4d ha y2a2 ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 91% </div><div class="t m0 x3d ha y2a3 ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 93%</div><div class="t m0 x3d ha y2a4 ff2 fs9 fc3 sc0 ls0 ws0">Precision : 88%</div><div class="t m0 x3d ha y2a5 ff2 fs9 fc3 sc0 ls0 ws0">Recall : 85%</div><div class="t m0 x3d ha y2a6 ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 86% </div><div class="t m0 x84 h25 y2a7 ff1 fs2 fc7 sc0 ls0 ws0">Gradient </div><div class="t m0 x84 h25 y2a8 ff1 fs2 fc7 sc0 ls0 ws0">Boosting</div><div class="t m0 x5a ha y2a9 ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 92%</div><div class="t m0 x5a ha y2aa ff2 fs9 fc3 sc0 ls0 ws0">Precision : 90%</div><div class="t m0 x5a ha y2ab ff2 fs9 fc3 sc0 ls0 ws0">Recall : 92%</div><div class="t m0 x5a ha y2ac ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 91% </div><div class="t m0 x5a h25 y2ad ff1 fs2 fc7 sc0 ls0 ws0">Voting-based </div><div class="t m0 x5a h25 y2ae ff1 fs2 fc7 sc0 ls0 ws0">Ensemble Method</div><div class="t m0 x18 ha y2af ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 94%</div><div class="t m0 x18 ha y2b0 ff2 fs9 fc3 sc0 ls0 ws0">Precision : 88%</div><div class="t m0 x18 ha y2b1 ff2 fs9 fc3 sc0 ls0 ws0">Recall : 87%</div><div class="t m0 x18 ha y2b2 ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 87% </div><div class="t m0 x18 h25 y2b3 ff1 fs2 fc7 sc0 ls0 ws0">Adaboost</div><div class="t m0 x85 ha y2b4 ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 88%</div><div class="t m0 x85 ha y2b5 ff2 fs9 fc3 sc0 ls0 ws0">Precision : 93%</div><div class="t m0 x85 ha y2b6 ff2 fs9 fc3 sc0 ls0 ws0">Recall : 83%</div><div class="t m0 x85 ha y2b7 ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 88% </div><div class="t m0 x85 h25 y2b8 ff1 fs2 fc7 sc0 ls0 ws0">Decision Trees</div><div class="t m0 x86 ha y2b9 ff2 fs9 fc3 sc0 ls0 ws0">Accuracy : 88%</div><div class="t m0 x86 ha y2ba ff2 fs9 fc3 sc0 ls0 ws0">Precision : 94%</div><div class="t m0 x86 ha y2bb ff2 fs9 fc3 sc0 ls0 ws0">Recall : 81%</div><div class="t m0 x86 ha y2bc ff2 fs9 fc3 sc0 ls0 ws0">F-measure : 87% </div><div class="t m0 x87 h25 y2b3 ff1 fs2 fc7 sc0 ls0 ws0">Naive Bayes</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3d" class="pf w0 h0" data-page-no="3d"><div class="pc pc3d w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3d.png"/><div class="t m0 x57 h5 y2bd ff1 fs3 fc1 sc0 ls0 ws0">Results and Evaluation Contd. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3e" class="pf w0 h0" data-page-no="3e"><div class="pc pc3e w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3e.png"/><div class="t m0 x23 h5 y2be ff1 fs3 fc1 sc0 ls0 ws0">Results and Evaluation Contd. </div><div class="t m0 x4e h10 y2bf ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">When <span class="_ _6"> </span>considering <span class="_ _c"></span>precision, <span class="_ _8"></span>recall <span class="_ _8"></span>and <span class="_ _13"> </span>f-measure <span class="_ _8"></span>along <span class="_ _13"></span>with <span class="_ _13"> </span>the <span class="_ _6"> </span>accuracy, <span class="_ _8"></span>the <span class="_ _6"> </span>best <span class="_ _13"> </span>suited <span class="_ _13"></span>classifier <span class="_ _a"></span>for </span></div><div class="t m0 x7c h10 y2c0 ff2 fsc fc3 sc0 ls0 ws0">the  Melanoma <span class="_ _e"> </span>diagnosis  process <span class="_ _5"> </span>based  on <span class="_ _f"> </span>the  dermoscopic <span class="_ _f"> </span>features <span class="_ _5"> </span>extracted <span class="_ _4"> </span>is  the <span class="_ _f"> </span>voting-based </div><div class="t m0 x7c h10 y2c1 ff2 fsc fc3 sc0 ls0 ws0">ensemble method as it has overall higher evaluation results. </div><div class="t m0 x4e h10 y2c2 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">ANN <span class="_ _8"></span>can <span class="_ _a"></span>be <span class="_ _b"></span>identified <span class="_ _12"></span>as <span class="_ _b"></span>the <span class="_ _a"></span>second <span class="_ _a"></span>best <span class="_ _a"></span>suited <span class="_ _a"></span>classier for <span class="_ _a"></span>the <span class="_ _b"></span>purpose <span class="_ _a"></span>where LR <span class="_ _b"></span>and <span class="_ _b"></span> <span class="_ _b"></span>SVM shows <span class="_ _a"></span>quite </span></div><div class="t m0 x7c h10 y2c3 ff2 fsc fc3 sc0 ls0 ws0">similar evaluation results which makes them take the third place. </div><div class="t m0 x4e h10 y2c4 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">The <span class="_ _a"></span>least <span class="_ _11"></span>suited classifiers <span class="_ _10"></span>for the <span class="_ _a"></span>purpose <span class="_ _a"></span>based on the <span class="_ _a"></span>evaluation <span class="_ _9"></span>results were <span class="_ _12"></span>the <span class="_ _a"></span>NB <span class="_ _a"></span>and <span class="_ _a"></span>DT <span class="_ _9"></span>classifiers. </span></div><div class="t m0 x7c h10 y2c5 ff2 fsc fc3 sc0 ls0 ws0">Therefore <span class="_ _9"></span>these <span class="_ _8"></span>two <span class="_ _8"></span>classifiers <span class="_ _9"></span>were <span class="_ _a"></span>not <span class="_ _8"></span>used <span class="_ _8"></span>in <span class="_ _8"></span>the <span class="_ _8"></span>voting-based <span class="_ _a"></span>ensemble <span class="_ _a"></span>classification <span class="_ _9"></span>method <span class="_ _8"></span>for <span class="_ _8"></span>the </div><div class="t m0 x7c h10 y2c6 ff2 fsc fc3 sc0 ls0 ws0">voting process.</div><div class="t m0 x4e h10 y2c7 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">When <span class="_ _b"></span>comparing <span class="_ _b"></span>the <span class="_ _b"></span>evaluation results obtained using <span class="_ _b"></span>the <span class="_ _b"></span>voting-based ensemble classification <span class="_ _12"></span>with <span class="_ _a"></span>the </span></div><div class="t m0 x7c h10 y2c8 ff2 fsc fc3 sc0 ls0 ws0">other <span class="_ _a"></span>research <span class="_ _12"></span>projects <span class="_ _12"></span>in <span class="_ _b"></span>the <span class="_ _b"></span>domain, <span class="_ _b"></span>the <span class="_ _b"></span>results were a <span class="_ _a"></span>standout <span class="_ _b"></span>not <span class="_ _b"></span>just <span class="_ _b"></span>as <span class="_ _b"></span>it <span class="_ _b"></span>had <span class="_ _b"></span>better <span class="_ _b"></span>accuracy, but </div><div class="t m0 x7c h10 y2c9 ff2 fsc fc3 sc0 ls0 ws0">also the precision, recall and f-measure values were also high.</div><div class="t m0 x4e h10 y2ca ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">Also <span class="_ _c"></span>as <span class="_ _c"></span>the <span class="_ _c"></span>solution <span class="_ _8"></span>follows the <span class="_ _c"></span>clinical <span class="_ _8"></span>process, <span class="_ _a"></span>it <span class="_ _c"></span>specifically points <span class="_ _8"></span>the <span class="_ _8"></span>dermoscopic <span class="_ _8"></span>features <span class="_ _a"></span>identified </span></div><div class="t m0 x7c h10 y2cb ff2 fsc fc3 sc0 ls0 ws0">from the skin <span class="_ _a"></span>lesion and therefore <span class="_ _30"></span>the <span class="_ _a"></span>output <span class="_ _a"></span>also <span class="_ _a"></span>assist <span class="_ _a"></span>dermatologists <span class="_ _9"></span>to <span class="_ _b"></span>get <span class="_ _a"></span>a <span class="_ _a"></span>better <span class="_ _a"></span>idea on the <span class="_ _a"></span>lesion </div><div class="t m0 x7c h10 y2cc ff2 fsc fc3 sc0 ls0 ws0">and explore the other risks that might occur due to those dermoscopic features. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3f" class="pf w0 h0" data-page-no="3f"><div class="pc pc3f w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3f.png"/><div class="t m0 x5f h5 y19c ff1 fs3 fc1 sc0 ls0 ws0">Detection of Mutations in </div><div class="t m0 x7 h5 y19d ff1 fs3 fc1 sc0 ls0 ws0">Gene DNA Sequences </div><div class="t m0 x7b h1 y19e ff1 fs0 fc1 sc0 ls0 ws0">Module 2 - Evaluation</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf40" class="pf w0 h0" data-page-no="40"><div class="pc pc40 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg40.png"/><div class="t m0 x57 h5 y2bd ff1 fs3 fc1 sc0 ls0 ws0">Evaluation of the Smith Waterman </div><div class="t m0 x57 h5 y2cd ff1 fs3 fc1 sc0 ls0 ws0">Implementation</div><div class="t m0 xc h10 y2ce ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">To <span class="_ _10"></span>evaluate <span class="_ _9"></span>the <span class="_ _a"></span>efficiency of <span class="_ _a"></span>the <span class="_ _a"></span>algorithm using <span class="_ _a"></span>execution <span class="_ _10"></span>time, <span class="_ _a"></span>several <span class="_ _12"></span>sequences were <span class="_ _9"></span>aligned and <span class="_ _a"></span>mutations <span class="_ _a"></span>were </span></div><div class="t m0 x1e h10 y2cf ff2 fsc fc3 sc0 ls0 ws0">found while recording the execution time. </div><div class="t m0 xc h10 y2d0 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">it <span class="_ _7"> </span>was <span class="_ _6"> </span>clearly <span class="_ _c"></span>seen <span class="_ _13"></span>that <span class="_ _6"> </span>the <span class="_ _6"> </span>implementation <span class="_ _13"></span>done <span class="_ _6"> </span>using <span class="_ _6"> </span>Python <span class="_ _6"> </span>for <span class="_ _6"> </span>the <span class="_ _6"> </span>Smith <span class="_ _6"> </span>Waterman <span class="_ _b"></span>Algorithm <span class="_ _13"></span>has <span class="_ _7"> </span>more <span class="_ _13"></span>than </span></div><div class="t m0 x1e h10 y2d1 ff2 fsc fc3 sc0 ls0 ws0">enough <span class="_ _b"></span>efficiency <span class="_ _b"></span>to <span class="_ _8"></span>be <span class="_ _b"></span>used <span class="_ _a"></span>on <span class="_ _8"></span>a <span class="_ _8"></span>clinical <span class="_ _a"></span>application <span class="_ _b"></span>as <span class="_ _8"></span>it <span class="_ _b"></span>has <span class="_ _8"></span>taken less <span class="_ _a"></span>than <span class="_ _b"></span>a <span class="_ _8"></span>second <span class="_ _a"></span>of <span class="_ _8"></span>execution <span class="_ _12"></span>time <span class="_ _b"></span>even <span class="_ _a"></span>for </div><div class="t m0 x1e h10 y2d2 ff2 fsc fc3 sc0 ls0 ws0">sequences with length close 45000 nucleotides</div><div class="t m0 x2a h24 y2d3 ff3 fsf fc3 sc0 ls0 ws0">●<span class="_ _31"> </span><span class="ff2 fsc">Algorithm implementation efficiency:</span></div><div class="t m0 x88 h1a y2d4 ff6 fs9 fc3 sc0 ls0 ws0">Sequence length<span class="_ _53"> </span>No. of Mutations<span class="_ _54"> </span>Execution Time (seconds)</div><div class="t m0 x89 h26 y2d5 ff4 fs9 fc3 sc0 ls0 ws0">8<span class="_ _55"> </span>5<span class="_ _56"> </span>0.505</div><div class="t m0 x8a h26 y2d6 ff4 fs9 fc3 sc0 ls0 ws0">69<span class="_ _57"> </span>69<span class="_ _58"> </span>0.688</div><div class="t m0 x8a h26 y2d7 ff4 fs9 fc3 sc0 ls0 ws0">69<span class="_ _59"> </span>0<span class="_ _56"> </span>0.004</div><div class="t m0 x80 h26 y2d8 ff4 fs9 fc3 sc0 ls0 ws0">570<span class="_ _5a"> </span>14<span class="_ _58"> </span>0.036</div><div class="t m0 x8b h26 y2d9 ff4 fs9 fc3 sc0 ls0 ws0">2301<span class="_ _5a"> </span>1<span class="_ _56"> </span>0.531</div><div class="t m0 x8b h26 y2da ff4 fs9 fc3 sc0 ls0 ws0">2301<span class="_ _5a"> </span>1<span class="_ _56"> </span>0.028</div><div class="t m0 x8b h26 y2db ff4 fs9 fc3 sc0 ls0 ws0">2301<span class="_ _5b"> </span>1735<span class="_ _5c"> </span>0.866</div><div class="t m0 x8b h26 y2dc ff4 fs9 fc3 sc0 ls0 ws0">2301<span class="_ _5d"> </span>42<span class="_ _58"> </span>0.038</div><div class="t m0 x46 h26 y2dd ff4 fs9 fc3 sc0 ls0 ws0">43524<span class="_ _5e"> </span>1212<span class="_ _5c"> </span>0.107</div><div class="t m0 x46 h26 y2de ff4 fs9 fc3 sc0 ls0 ws0">43524<span class="_ _5b"> </span>920<span class="_ _5f"> </span>0.056</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf41" class="pf w0 h0" data-page-no="41"><div class="pc pc41 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg41.png"/><div class="t m0 xc h10 y2df ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _2d"> </span><span class="ff2">The <span class="_ _8"></span>accuracy was <span class="_ _a"></span>evaluated <span class="_ _11"></span>by <span class="_ _a"></span>passing <span class="_ _a"></span>a <span class="_ _8"></span>100% <span class="_ _b"></span>similar sequence <span class="_ _b"></span>and <span class="_ _b"></span>100% <span class="_ _b"></span>dissimilar sequence <span class="_ _b"></span>to <span class="_ _b"></span>check <span class="_ _a"></span>for <span class="_ _b"></span>how <span class="_ _a"></span>the </span></div><div class="t m0 x1e h10 y2e0 ff2 fsc fc3 sc0 ls0 ws0">algorithm <span class="_ _6"> </span>behaves <span class="_ _13"></span>at <span class="_ _4"> </span>the <span class="_ _7"> </span>extremes, <span class="_ _13"></span>in <span class="_ _7"> </span>where <span class="_ _6"> </span>the <span class="_ _4"> </span>algorithm <span class="_ _6"> </span>behaved <span class="_ _c"></span>with <span class="_ _7"> </span>100% <span class="_ _4"> </span>accuracy <span class="_ _13"></span>even <span class="_ _13"> </span>after <span class="_ _4"> </span>changing <span class="_ _7"> </span>the </div><div class="t m0 x1e h10 y2e1 ff2 fsc fc3 sc0 ls0 ws0">sequence <span class="_ _13"></span>length. <span class="_ _13"></span>Next, <span class="_ _13"></span>by <span class="_ _c"></span>passing <span class="_ _6"> </span>a <span class="_ _6"> </span>known <span class="_ _13"></span>mutated <span class="_ _13"></span>sequence <span class="_ _13"></span>and <span class="_ _6"> </span>cross-checking <span class="_ _b"></span>the <span class="_ _6"> </span>output, <span class="_ _6"> </span>it <span class="_ _13"> </span>was <span class="_ _6"> </span>clearly <span class="_ _8"></span>seen </div><div class="t m0 x1e h10 y2e2 ff2 fsc fc3 sc0 ls0 ws0">that the algorithm does work as expected.</div><div class="t m0 x23 h10 y2e3 ff8 fs6 fc3 sc0 ls0 ws0">●<span class="_ _1f"> </span><span class="ff2 fsc">Algorithm implementation accuracy</span></div><div class="t m0 x5a h27 y2e4 ff1 fs14 fc1 sc0 ls0 ws0">Evaluation of the Smith Waterman </div><div class="t m0 x5a h27 y2e5 ff1 fs14 fc1 sc0 ls0 ws0">Implementation (cont.)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf42" class="pf w0 h0" data-page-no="42"><div class="pc pc42 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg42.png"/><div class="t m0 x23 h10 y2e6 ff8 fs6 fc3 sc0 ls0 ws0">●<span class="_ _1f"> </span><span class="ff2 fsc">Three libraries have <span class="_ _a"></span>been <span class="_ _a"></span>evaluated for <span class="_ _a"></span>the <span class="_ _8"></span>calculation of <span class="_ _b"></span>the <span class="_ _b"></span>mutation <span class="_ _8"></span>threshold, <span class="_ _9"></span>so <span class="_ _8"></span>that <span class="_ _8"></span>the <span class="_ _b"></span>derived <span class="_ _a"></span>similarity <span class="_ _a"></span>and </span></div><div class="t m0 x1e h10 y2e7 ff2 fsc fc3 sc0 ls0 ws0">difference rates vouch for each other’s accuracy. </div><div class="t m0 x11 h24 y2e8 ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">FuzzyWuzzy with FuzzySearch - calculates similarity</span></div><div class="t m0 x11 h24 y2e9 ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Strisimpy - calculates difference</span></div><div class="t m0 x11 h24 y2ea ff3 fsf fc3 sc0 ls0 ws0">○<span class="_ _31"> </span><span class="ff2 fsc">Rapid Fuzz -  calculates similarity</span></div><div class="t m0 x23 h10 y2eb ff8 fs6 fc3 sc0 ls0 ws0">●<span class="_ _1f"> </span><span class="ff2 fsc">It <span class="_ _c"></span>is <span class="_ _8"></span>observed <span class="_ _a"></span>that <span class="_ _c"></span>this <span class="_ _8"></span>error <span class="_ _b"></span>rate <span class="_ _a"></span>is <span class="_ _c"></span>less <span class="_ _b"></span>than <span class="_ _8"></span>1% <span class="_ _c"></span>for <span class="_ _8"></span>&gt;98% <span class="_ _8"></span>of <span class="_ _c"></span>the <span class="_ _8"></span>instances <span class="_ _8"></span>and <span class="_ _8"></span>therefore this <span class="_ _8"></span>error <span class="_ _a"></span>rate <span class="_ _b"></span>is <span class="_ _c"></span>accepted </span></div><div class="t m0 x1e h10 y2ec ff2 fsc fc3 sc0 ls0 ws0">while conducting this research. </div><div class="t m0 xc h27 y2ed ff1 fs14 fc1 sc0 ls0 ws0">Evaluation of the Defragmentation Algorithm</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf43" class="pf w0 h0" data-page-no="43"><div class="pc pc43 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg43.png"/><div class="t m0 x23 h10 y2ee ff8 fs6 fc3 sc0 ls0 ws0">●<span class="_ _1f"> </span><span class="ff2 fsc">However, <span class="_ _10"></span>it <span class="_ _13"></span>was <span class="_ _c"></span>also <span class="_ _6"> </span>observed <span class="_ _c"></span>that <span class="_ _13"></span>even <span class="_ _c"></span>though <span class="_ _13"></span>the <span class="_ _13"> </span>Strisimpy <span class="_ _c"></span>library <span class="_ _c"></span>generates <span class="_ _8"></span>the <span class="_ _6"> </span>similarity <span class="_ _c"></span>difference, <span class="_ _b"></span>it <span class="_ _6"> </span>takes <span class="_ _8"></span>a </span></div><div class="t m0 x1e h10 y2ef ff2 fsc fc3 sc0 ls0 ws0">comparatively <span class="_ _c"></span>large <span class="_ _8"></span>time <span class="_ _4"> </span>to <span class="_ _6"> </span>derive <span class="_ _6"> </span>the <span class="_ _7"> </span>result, <span class="_ _13"></span>making <span class="_ _7"> </span>it <span class="_ _7"> </span>not <span class="_ _7"> </span>suitable <span class="_ _13"></span>to <span class="_ _7"> </span>be <span class="_ _7"> </span>used <span class="_ _6"> </span>as <span class="_ _4"> </span>the <span class="_ _6"> </span>main <span class="_ _7"> </span>similarity <span class="_ _6"> </span>measuring </div><div class="t m0 x1e h10 y2f0 ff2 fsc fc3 sc0 ls0 ws0">library for the whole process of defragmenting.</div><div class="t m0 x23 h10 y2f1 ff8 fs6 fc3 sc0 ls0 ws0">●<span class="_ _1f"> </span><span class="ff2 fsc">When <span class="_ _b"></span>using <span class="_ _b"></span>the <span class="_ _b"></span>RapidFuzz <span class="_ _b"></span>library, it <span class="_ _b"></span>was <span class="_ _a"></span>noted <span class="_ _a"></span>even though <span class="_ _b"></span>it <span class="_ _b"></span>does <span class="_ _8"></span>calculate <span class="_ _9"></span>the <span class="_ _b"></span>similarity rate <span class="_ _a"></span>well, <span class="_ _a"></span>the <span class="_ _b"></span>accuracy is </span></div><div class="t m0 x1e h10 y2f2 ff2 fsc fc3 sc0 ls0 ws0">more <span class="_ _4"> </span>towards <span class="_ _6"> </span>less <span class="_ _4"> </span>fragmented <span class="_ _7"> </span>sequences <span class="_ _5"> </span>and <span class="_ _5"> </span>it <span class="_ _e"> </span>under <span class="_ _5"> </span>fits <span class="_ _4"> </span>and <span class="_ _5"> </span>overlooks <span class="_ _6"> </span>some <span class="_ _5"> </span>fragmentations <span class="_ _4"> </span>because <span class="_ _4"> </span>of <span class="_ _e"> </span>the </div><div class="t m0 x1e h10 y2f3 ff2 fsc fc3 sc0 ls0 ws0">inaccurate similarity measuring in more fragmented situations.</div><div class="t m0 x23 h10 y2f4 ff8 fs6 fc3 sc0 ls0 ws0">●<span class="_ _1f"> </span><span class="ff2 fsc">Therefore, <span class="_ _8"></span>as <span class="_ _7"> </span>described <span class="_ _13"></span>in <span class="_ _7"> </span>the <span class="_ _7"> </span>implementation <span class="_ _13"></span>section <span class="_ _13"> </span>for <span class="_ _6"> </span>this <span class="_ _7"> </span>module, <span class="_ _6"> </span>an <span class="_ _7"> </span>ensemble <span class="_ _13"></span>of <span class="_ _7"> </span>the <span class="_ _7"> </span>libraries <span class="_ _13"></span>were <span class="_ _c"></span>used <span class="_ _6"> </span>to </span></div><div class="t m0 x1e h10 y2f5 ff2 fsc fc3 sc0 ls0 ws0">derive <span class="_ _9"></span>the similarity <span class="_ _9"></span>rates <span class="_ _12"></span>of the defragmenting <span class="_ _12"></span>sequence to obtain better results, <span class="_ _30"></span>and that it derives <span class="_ _9"></span>better accuracy </div><div class="t m0 x1e h10 y2f6 ff2 fsc fc3 sc0 ls0 ws0">than using only one library from the beginning to the end of the defragmenting process.</div><div class="t m0 x23 h10 y2f7 ff8 fs6 fc3 sc0 ls0 ws0">●<span class="_ _1f"> </span><span class="ff2 fsc">The <span class="_ _a"></span>algorithm tries its <span class="_ _a"></span>best <span class="_ _a"></span>sure <span class="_ _9"></span>not <span class="_ _a"></span>to <span class="_ _a"></span>over-fit <span class="_ _10"></span>the <span class="_ _a"></span>patient sequence and <span class="_ _a"></span>overwrite <span class="_ _9"></span>the <span class="_ _a"></span>actual mutation(s) by mistake, </span></div><div class="t m0 x1e h10 y2f8 ff2 fsc fc3 sc0 ls0 ws0">which is done by having threshold values to change the similarity measuring library from one to another. </div><div class="t m0 x5a h27 y2e4 ff1 fs14 fc1 sc0 ls0 ws0">Evaluation of the Defragmentation Algorithm </div><div class="t m0 x5a h27 y2e5 ff1 fs14 fc1 sc0 ls0 ws0">(cont.)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf44" class="pf w0 h0" data-page-no="44"><div class="pc pc44 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg44.png"/><div class="t m0 x1f h5 y2f9 ff1 fs3 fc1 sc0 ls0 ws0">Detection of Melanoma based on Protein </div><div class="t m0 x63 h5 y2fa ff1 fs3 fc1 sc0 ls0 ws0">Primary Structure </div><div class="t m0 x7b h1 y2fb ff1 fs0 fc1 sc0 ls0 ws0">Module 3 - Evaluation</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf45" class="pf w0 h0" data-page-no="45"><div class="pc pc45 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg45.png"/><div class="t m0 x59 h1e y1ee ff1 fs11 fc1 sc0 ls0 ws0">Results and Evaluation - Vectorizing Method 01</div><div class="t m0 x6e h9 y2fc ff3 fs8 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff1">F1-scores for selected classifiers </span></div><div class="t m0 x77 ha y2fd ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">50,000+ data points </span></div><div class="t m0 x77 ha y2fe ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">24 features </span></div><div class="t m0 x77 ha y2ff ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">124 cancer and non-cancer variations </span></div><div class="t m0 x77 ha y300 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Cross-validation </span></div><div class="t m0 x77 ha y301 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Hyperparameter-tuning </span></div><div class="t m0 x77 ha y302 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Best Classifier - Gradient Boosting</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf46" class="pf w0 h0" data-page-no="46"><div class="pc pc46 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg46.png"/><div class="t m0 x59 h1e y1ee ff1 fs11 fc1 sc0 ls0 ws0">Results and Evaluation - Vectorizing Method 02</div><div class="t m0 x6e h9 y2fc ff3 fs8 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff1">F1-scores for selected classifiers </span></div><div class="t m0 x77 ha y2fd ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">50,000+ data points </span></div><div class="t m0 x77 ha y2fe ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">404 features </span></div><div class="t m0 x77 ha y2ff ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">124 cancer and non-cancer variations </span></div><div class="t m0 x77 ha y300 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Cross-validation </span></div><div class="t m0 x77 ha y301 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Hyperparameter-tuning </span></div><div class="t m0 x77 ha y302 ff3 fs9 fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Best Classifier - Gradient Boosting</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf47" class="pf w0 h0" data-page-no="47"><div class="pc pc47 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg47.png"/><div class="t m0 x59 h1e y1ee ff1 fs11 fc1 sc0 ls0 ws0">Results and Evaluation - Summary </div><div class="t m0 x20 h10 y303 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The <span class="_ _4"> </span>Gradient <span class="_ _13"></span>Boosting <span class="_ _7"> </span>Classifier <span class="_ _8"></span>model <span class="_ _4"> </span>is <span class="_ _7"> </span>the <span class="_ _7"> </span>best <span class="_ _7"> </span>performer <span class="_ _7"> </span>of <span class="_ _7"> </span>them <span class="_ _4"> </span>all, <span class="_ _7"> </span>with <span class="_ _7"> </span>a <span class="_ _4"> </span>0.70 <span class="_ _c"></span>f1-score <span class="_ _6"> </span>in <span class="_ _7"> </span>the <span class="_ _7"> </span>testing </span></div><div class="t m0 x60 h10 y304 ff2 fsc fc3 sc0 ls0 ws0">phase <span class="_ _c"></span>for <span class="_ _8"></span>Vectorizing <span class="_ _9"></span>Method <span class="_ _c"></span>01. <span class="_ _b"></span>And <span class="_ _c"></span>it <span class="_ _c"></span>seems <span class="_ _c"></span>all <span class="_ _c"></span>the <span class="_ _c"></span>models <span class="_ _c"></span>are <span class="_ _b"></span>generalized based <span class="_ _8"></span>on <span class="_ _c"></span>the <span class="_ _c"></span>above <span class="_ _b"></span>training <span class="_ _b"></span>&amp; </div><div class="t m0 x60 h10 y305 ff2 fsc fc3 sc0 ls0 ws0">testing measures since there is a very slight difference between training and testing f1-score values.</div><div class="t m0 x20 h10 y306 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The <span class="_ _4"> </span>Gradient <span class="_ _c"></span>Boosting <span class="_ _4"> </span>Classifier <span class="_ _8"></span>model <span class="_ _7"> </span>is <span class="_ _7"> </span>the <span class="_ _7"> </span>best <span class="_ _7"> </span>performer <span class="_ _6"> </span>of <span class="_ _4"> </span>them <span class="_ _7"> </span>all, <span class="_ _7"> </span>with <span class="_ _7"> </span>a <span class="_ _4"> </span>0.68 <span class="_ _13"></span>f1-score <span class="_ _13"> </span>in <span class="_ _4"> </span>the <span class="_ _6"> </span>testing </span></div><div class="t m0 x60 h10 y307 ff2 fsc fc3 sc0 ls0 ws0">phase <span class="_ _c"></span>for <span class="_ _13"></span>Vectorizing <span class="_ _9"></span>Method <span class="_ _13"></span>02. <span class="_ _13"> </span>And <span class="_ _13"> </span>it <span class="_ _13"> </span>seems <span class="_ _13"></span>the <span class="_ _13"></span>KNN <span class="_ _13"></span>model <span class="_ _c"></span>was <span class="_ _13"></span>under- <span class="_ _c"></span>fitted. <span class="_ _b"></span>Because, <span class="_ _c"></span>training <span class="_ _8"></span>achieved </div><div class="t m0 x60 h10 y308 ff2 fsc fc3 sc0 ls0 ws0">0.71 <span class="_ _c"></span>f1-score. <span class="_ _c"></span>But, <span class="_ _7"> </span>testing <span class="_ _13"> </span>achieved <span class="_ _8"></span>0.58 <span class="_ _13"></span>and <span class="_ _6"> </span>it <span class="_ _6"> </span>is <span class="_ _6"> </span>very <span class="_ _13"></span>low <span class="_ _8"></span>compared <span class="_ _8"></span>to <span class="_ _7"> </span>the <span class="_ _6"> </span>value <span class="_ _c"></span>achieved <span class="_ _8"></span>in <span class="_ _6"> </span>the <span class="_ _6"> </span>training. <span class="_ _c"></span>All </div><div class="t m0 x60 h10 y309 ff2 fsc fc3 sc0 ls0 ws0">other models except the KNN seem to be generalized.</div><div class="t m0 x20 h10 y30a ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Gradient <span class="_ _6"> </span>Boosting <span class="_ _4"> </span>was <span class="_ _4"> </span>the <span class="_ _7"> </span>best <span class="_ _4"> </span>performing <span class="_ _7"> </span>algorithm <span class="_ _7"> </span>for <span class="_ _4"> </span>melanoma <span class="_ _6"> </span>diagnosis <span class="_ _4"> </span>utilizing <span class="_ _7"> </span>genetic <span class="_ _4"> </span>features <span class="_ _c"></span>in </span></div><div class="t m0 x60 h10 y30b ff2 fsc fc3 sc0 ls0 ws0">both vectorizing approaches. </div><div class="t m0 x20 h10 y30c ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Based <span class="_ _5"> </span>on <span class="_ _5"> </span>f1-score <span class="_ _4"> </span>values, <span class="_ _4"> </span>it <span class="_ _5"> </span>appears <span class="_ _5"> </span>that <span class="_ _5"> </span>vectorizing <span class="_ _7"> </span>method <span class="_ _5"> </span>01 <span class="_ _4"> </span>is <span class="_ _5"> </span>the <span class="_ _5"> </span>best <span class="_ _5"> </span>methodology <span class="_ _4"> </span>for <span class="_ _5"> </span>representing </span></div><div class="t m0 x60 h10 y30d ff2 fsc fc3 sc0 ls0 ws0">protein <span class="_ _9"></span>primary <span class="_ _b"></span>structure. It also <span class="_ _b"></span>has <span class="_ _a"></span>an <span class="_ _a"></span>advantage in that <span class="_ _b"></span>it <span class="_ _a"></span>only <span class="_ _a"></span>consumes <span class="_ _a"></span>twenty-four features <span class="_ _12"></span>(24) <span class="_ _a"></span>for <span class="_ _a"></span>input </div><div class="t m0 x60 h10 y30e ff2 fsc fc3 sc0 ls0 ws0">buffer. </div><div class="t m0 x20 h10 y30f ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Other <span class="_ _a"></span>than <span class="_ _a"></span>protein <span class="_ _9"></span>primary <span class="_ _a"></span>structure, <span class="_ _9"></span>it <span class="_ _a"></span>appears that tumor <span class="_ _a"></span>origin <span class="_ _a"></span>and <span class="_ _a"></span>age <span class="_ _a"></span>are <span class="_ _9"></span>the <span class="_ _a"></span>most <span class="_ _a"></span>influential factors for </span></div><div class="t m0 x60 h10 y310 ff2 fsc fc3 sc0 ls0 ws0">melanoma identification based on entropy-based feature selection.</div><div class="t m0 x20 h10 y311 ff3 fsc fc3 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Finally, the model was able to detect melanoma patients at the initial stage with overall 0.70 accuracy. </span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf48" class="pf w0 h0" data-page-no="48"><div class="pc pc48 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg48.png"/><div class="t m0 x5a h5 y312 ff1 fs3 fc1 sc0 ls0 ws0">Conclusion</div><div class="t m0 x1a hf y313 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">This <span class="_ _1b"> </span>research <span class="_ _5"> </span>presents  an <span class="_ _1b"> </span>efficient <span class="_ _35"> </span>and <span class="_ _1b"> </span>accurate  method <span class="_ _35"> </span>of <span class="_ _35"> </span>diagnosing <span class="_ _35"> </span>Melanoma  using <span class="_ _35"> </span>computer <span class="_ _35"> </span>vision <span class="_ _35"> </span>and <span class="_ _35"> </span>DNA </span></div><div class="t m0 x1e hf y314 ff2 fsb fc0 sc0 ls0 ws0">sequencing.</div><div class="t m0 x1a hf y315 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">When <span class="_ _a"></span>computer <span class="_ _a"></span>vision <span class="_ _b"></span>is <span class="_ _a"></span>being <span class="_ _a"></span>used, <span class="_ _a"></span>unlike <span class="_ _9"></span>other <span class="_ _b"></span>research <span class="_ _12"></span>projects <span class="_ _9"></span>in <span class="_ _a"></span>this <span class="_ _a"></span>research <span class="_ _12"></span>the <span class="_ _a"></span>features <span class="_ _9"></span>are not <span class="_ _a"></span>limited <span class="_ _a"></span>to <span class="_ _a"></span>the <span class="_ _a"></span>ABCD </span></div><div class="t m0 x1e hf y316 ff2 fsb fc0 sc0 ls0 ws0">features but 12 salient dermoscopic features are being analyzed and extracted to perform the Melanoma diagnosis. </div><div class="t m0 x1a hf y317 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">With <span class="_ _13"> </span>regard <span class="_ _b"></span>to <span class="_ _6"> </span>the <span class="_ _13"> </span>analysis <span class="_ _6"> </span>of <span class="_ _6"> </span>the <span class="_ _13"> </span>DNA <span class="_ _6"> </span>sequences, <span class="_ _13"> </span>this <span class="_ _6"> </span>research <span class="_ _a"></span>this <span class="_ _6"> </span>research <span class="_ _b"></span>the <span class="_ _13"> </span>mutations <span class="_ _6"> </span>are <span class="_ _c"></span>being <span class="_ _6"> </span>identified <span class="_ _b"></span>even <span class="_ _c"></span>in </span></div><div class="t m0 x1e hf y318 ff2 fsb fc0 sc0 ls0 ws0">fragmented <span class="_ _12"></span>DNA sequences and also the DNA mutation is being <span class="_ _a"></span>considered <span class="_ _12"></span>up to its protein functionality level <span class="_ _30"></span>to make <span class="_ _9"></span>sure </div><div class="t m0 x1e hf y319 ff2 fsb fc0 sc0 ls0 ws0">that the DNA mutation detected is a cause for Melanoma.</div><div class="t m0 x1a hf y31a ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">In <span class="_ _8"></span>both <span class="_ _13"></span>these <span class="_ _8"></span>approaches <span class="_ _c"></span>required pre-processing <span class="_ _b"></span>steps <span class="_ _c"></span>are <span class="_ _b"></span>performed <span class="_ _8"></span>to <span class="_ _c"></span>mitigate <span class="_ _c"></span>the <span class="_ _c"></span>noise <span class="_ _c"></span>before <span class="_ _b"></span>processing <span class="_ _8"></span>and <span class="_ _c"></span>then </span></div><div class="t m0 x1e hf y31b ff2 fsb fc0 sc0 ls0 ws0">finally the diagnosis is performed using a classification model.</div><div class="t m0 x1a hf y31c ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Voting-based ensemble <span class="_ _c"></span>method <span class="_ _13"></span>classification <span class="_ _b"></span>is <span class="_ _6"> </span>used <span class="_ _c"></span>with <span class="_ _13"> </span>regard <span class="_ _b"></span>to <span class="_ _13"> </span>the <span class="_ _13"> </span>diagnosis <span class="_ _6"> </span>performed <span class="_ _c"></span>based <span class="_ _c"></span>on <span class="_ _13"> </span>the <span class="_ _13"> </span>dermoscopic </span></div><div class="t m0 x1e hf y31d ff2 fsb fc0 sc0 ls0 ws0">image <span class="_ _4"> </span>and <span class="_ _5"> </span>the <span class="_ _4"> </span>Gradient <span class="_ _7"> </span>Boosting <span class="_ _5"> </span>classifier <span class="_ _13"> </span>is <span class="_ _4"> </span>used <span class="_ _4"> </span>to <span class="_ _5"> </span>build <span class="_ _7"> </span>and <span class="_ _4"> </span>train <span class="_ _4"> </span>the <span class="_ _4"> </span>model <span class="_ _4"> </span>when <span class="_ _4"> </span>diagnosis <span class="_ _5"> </span>is <span class="_ _5"> </span>based <span class="_ _7"> </span>on <span class="_ _5"> </span>the <span class="_ _4"> </span>DNA </div><div class="t m0 x1e hf y31e ff2 fsb fc0 sc0 ls0 ws0">sequence. </div><div class="t m0 x1a hf y31f ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Accuracy, precision, <span class="_ _a"></span>recall <span class="_ _a"></span>and <span class="_ _8"></span>f-measure values <span class="_ _b"></span>were <span class="_ _a"></span>used <span class="_ _b"></span>to <span class="_ _8"></span>evaluate <span class="_ _a"></span>the <span class="_ _8"></span>obtained <span class="_ _b"></span>results <span class="_ _a"></span>and <span class="_ _8"></span>to <span class="_ _8"></span>compare <span class="_ _b"></span>them <span class="_ _8"></span>with <span class="_ _b"></span>the </span></div><div class="t m0 x1e hf y320 ff2 fsb fc0 sc0 ls0 ws0">existing research projects in the considered domain. </div><div class="t m0 x1a hf y321 ff3 fsb fc0 sc0 ls0 ws0">●<span class="_ _3"> </span><span class="ff2">Based <span class="_ _b"></span>on <span class="_ _c"></span>the <span class="_ _8"></span>evaluation <span class="_ _8"></span>results <span class="_ _b"></span>it <span class="_ _8"></span>was <span class="_ _8"></span>visible <span class="_ _8"></span>that <span class="_ _8"></span>the <span class="_ _c"></span>proposal <span class="_ _8"></span>solution <span class="_ _8"></span>in <span class="_ _c"></span>this <span class="_ _8"></span>research is <span class="_ _c"></span>reliable <span class="_ _a"></span>and <span class="_ _8"></span>therefore could <span class="_ _8"></span>be </span></div><div class="t m0 x1e hf y322 ff2 fsb fc0 sc0 ls0 ws0">used <span class="_ _c"></span>for <span class="_ _13"></span>clinical <span class="_ _13"> </span>practice. <span class="_ _c"></span>It <span class="_ _6"> </span>is <span class="_ _13"></span>capable <span class="_ _c"></span>of <span class="_ _13"> </span>diagnosing <span class="_ _6"> </span>Melanoma <span class="_ _13"></span>at <span class="_ _13"> </span>a <span class="_ _13"> </span>considerable <span class="_ _8"></span>accuracy <span class="_ _13"> </span>in <span class="_ _13"> </span>both <span class="_ _6"> </span>instances <span class="_ _13"> </span>where <span class="_ _c"></span>the </div><div class="t m0 x1e hf y323 ff2 fsb fc0 sc0 ls0 ws0">dermoscopic <span class="_ _5"> </span>features <span class="_ _13"> </span>are <span class="_ _4"> </span>visible <span class="_ _7"> </span>on <span class="_ _5"> </span>the <span class="_ _4"> </span>skin <span class="_ _5"> </span>or <span class="_ _4"> </span>even <span class="_ _7"> </span>when <span class="_ _4"> </span>there <span class="_ _4"> </span>are <span class="_ _6"> </span>no <span class="_ _5"> </span>signs <span class="_ _5"> </span>of <span class="_ _4"> </span>Melanoma <span class="_ _4"> </span>that <span class="_ _4"> </span>enables <span class="_ _4"> </span>diagnosis <span class="_ _5"> </span>of </div><div class="t m0 x1e hf y324 ff2 fsb fc0 sc0 ls0 ws0">Melanoma at its very early stages in which it is curable. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf49" class="pf w0 h0" data-page-no="49"><div class="pc pc49 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg49.png"/><div class="t m0 x5a h5 y312 ff1 fs3 fc1 sc0 ls0 ws0">Further Work</div><div class="t m0 x1f h28 y325 ff3 fs9 fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2 fsc">If <span class="_ _a"></span>datasets <span class="_ _a"></span>could <span class="_ _9"></span>be <span class="_ _a"></span>obtained of <span class="_ _a"></span>patients with <span class="_ _a"></span>both <span class="_ _a"></span>dermoscopic images <span class="_ _a"></span>and <span class="_ _a"></span>DNA <span class="_ _a"></span>sequences, then <span class="_ _a"></span>this <span class="_ _a"></span>diagnosis </span></div><div class="t m0 x61 h10 y326 ff2 fsc fc0 sc0 ls0 ws0">performed <span class="_ _9"></span>separately <span class="_ _9"></span>using <span class="_ _a"></span>two methods can <span class="_ _a"></span>be connected <span class="_ _9"></span>as <span class="_ _a"></span>one and <span class="_ _a"></span>that could increase <span class="_ _30"></span>the <span class="_ _a"></span>accuracy <span class="_ _9"></span>and the </div><div class="t m0 x61 h10 y327 ff2 fsc fc0 sc0 ls0 ws0">reliability of the obtained results even better than the already achieved evaluation results. </div><div class="t m0 x59 h10 y328 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">The <span class="_ _8"></span>solution <span class="_ _b"></span>could <span class="_ _a"></span>be <span class="_ _8"></span>further <span class="_ _b"></span>developed <span class="_ _12"></span>to <span class="_ _8"></span>diagnose <span class="_ _b"></span>other <span class="_ _8"></span>types <span class="_ _b"></span>of <span class="_ _8"></span>skin <span class="_ _8"></span>disease <span class="_ _a"></span>as <span class="_ _8"></span>well <span class="_ _a"></span>either <span class="_ _8"></span>by <span class="_ _a"></span>giving <span class="_ _b"></span>concern </span></div><div class="t m0 x61 h10 y329 ff2 fsc fc0 sc0 ls0 ws0">to the dermoscopic image or DNA sequence.</div><div class="t m0 x59 h10 y32a ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">This <span class="_ _e"> </span>system <span class="_ _5"> </span>could <span class="_ _4"> </span>be <span class="_ _5"> </span>further <span class="_ _5"> </span>developed <span class="_ _c"></span>as <span class="_ _e"> </span>a <span class="_ _5"> </span>mobile <span class="_ _4"> </span>application <span class="_ _4"> </span>which <span class="_ _5"> </span>includes <span class="_ _5"> </span>the <span class="_ _5"> </span>entire <span class="_ _7"> </span>screening <span class="_ _7"> </span>process </span></div><div class="t m0 x61 h10 y32b ff2 fsc fc0 sc0 ls0 ws0">starting <span class="_ _6"> </span>from <span class="_ _c"></span>taking <span class="_ _6"> </span>a <span class="_ _6"> </span>picture <span class="_ _c"></span>of <span class="_ _6"> </span>the <span class="_ _6"> </span>lesions <span class="_ _13"></span>to <span class="_ _6"> </span>the <span class="_ _6"> </span>analysis <span class="_ _6"> </span>of <span class="_ _7"> </span>them <span class="_ _6"> </span>while <span class="_ _c"></span>extracting <span class="_ _8"></span>the <span class="_ _6"> </span>dermoscopic <span class="_ _6"> </span>features </div><div class="t m0 x61 h10 y32c ff2 fsc fc0 sc0 ls0 ws0">from <span class="_ _b"></span>them <span class="_ _c"></span>so <span class="_ _c"></span>that <span class="_ _c"></span>the <span class="_ _c"></span>solution. <span class="_ _8"></span>But <span class="_ _c"></span>here <span class="_ _b"></span>the <span class="_ _c"></span>image <span class="_ _c"></span>capturing <span class="_ _8"></span>process <span class="_ _b"></span>needs <span class="_ _b"></span>to <span class="_ _c"></span>be <span class="_ _c"></span>monitored <span class="_ _a"></span>and <span class="_ _c"></span>instructed <span class="_ _b"></span>by </div><div class="t m0 x61 h10 y32d ff2 fsc fc0 sc0 ls0 ws0">the application to ensure that the image is captured accordingly. </div><div class="t m0 x59 h10 y32e ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Four <span class="_ _8"></span>genes <span class="_ _8"></span>are targeted to <span class="_ _8"></span>be <span class="_ _b"></span>checked <span class="_ _9"></span>in <span class="_ _b"></span>this <span class="_ _8"></span>research <span class="_ _12"></span>for <span class="_ _b"></span>mutations <span class="_ _8"></span>that <span class="_ _b"></span>can <span class="_ _8"></span>cause <span class="_ _b"></span>Melanoma. <span class="_ _a"></span>This <span class="_ _8"></span>model <span class="_ _8"></span>can </span></div><div class="t m0 x61 h10 y32f ff2 fsc fc0 sc0 ls0 ws0">be further developed <span class="_ _60"></span>to check <span class="_ _9"></span>for mutations in the <span class="_ _9"></span>other genes as well, when the whole <span class="_ _12"></span>DNA sequence of a patient </div><div class="t m0 x61 h10 y330 ff2 fsc fc0 sc0 ls0 ws0">is entered into the system. </div><div class="t m0 x59 h10 y331 ff3 fsc fc0 sc0 ls0 ws0">●<span class="_ _d"> </span><span class="ff2">Further <span class="_ _8"></span>analysis <span class="_ _8"></span>could <span class="_ _b"></span>be <span class="_ _8"></span>performed <span class="_ _a"></span>on <span class="_ _c"></span>the <span class="_ _8"></span>Secondary, <span class="_ _b"></span>Tertiary <span class="_ _30"></span>and <span class="_ _8"></span>Quaternary <span class="_ _8"></span>structures <span class="_ _a"></span>of <span class="_ _8"></span>proteins <span class="_ _a"></span>and <span class="_ _8"></span>take </span></div><div class="t m0 x61 h10 y332 ff2 fsc fc0 sc0 ls0 ws0">those <span class="_ _c"></span>structures <span class="_ _b"></span>into <span class="_ _c"></span>consideration <span class="_ _b"></span>when <span class="_ _c"></span>detecting <span class="_ _8"></span>Melanoma <span class="_ _8"></span>to <span class="_ _c"></span>improve the <span class="_ _13"></span>accuracy <span class="_ _b"></span>and <span class="_ _c"></span>recall <span class="_ _a"></span>values <span class="_ _c"></span>of <span class="_ _c"></span>the </div><div class="t m0 x61 h10 y333 ff2 fsc fc0 sc0 ls0 ws0">results. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf4a" class="pf w0 h0" data-page-no="4a"><div class="pc pc4a w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg4a.png"/><div class="t m0 x52 h29 y334 ff1 fs15 fc1 sc0 ls0 ws0">Demonstration</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf4b" class="pf w0 h0" data-page-no="4b"><div class="pc pc4b w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg4b.png"/><div class="t m0 x9 h29 y334 ff1 fs15 fc1 sc0 ls0 ws0">Thank You</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
</div>
<div class="loading-indicator">

</div>
</body>
</html>
